\doxysection{scipy.\+stats.\+\_\+stats\+\_\+py Namespace Reference}
\hypertarget{namespacescipy_1_1stats_1_1__stats__py}{}\label{namespacescipy_1_1stats_1_1__stats__py}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection*{Classes}
\begin{DoxyCompactItemize}
\item 
class \mbox{\hyperlink{classscipy_1_1stats_1_1__stats__py_1_1__ParallelP}{\+\_\+\+ParallelP}}
\item 
class \mbox{\hyperlink{classscipy_1_1stats_1_1__stats__py_1_1AlexanderGovernResult}{Alexander\+Govern\+Result}}
\item 
class \mbox{\hyperlink{classscipy_1_1stats_1_1__stats__py_1_1PearsonRResult}{Pearson\+RResult}}
\item 
class \mbox{\hyperlink{classscipy_1_1stats_1_1__stats__py_1_1TtestResult}{Ttest\+Result}}
\end{DoxyCompactItemize}
\doxysubsubsection*{Functions}
\begin{DoxyCompactItemize}
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_aaa8f1f67d10a05713172b6a32169ed91}{\+\_\+chk\+\_\+asarray}} (a, axis)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_aa7b513e10adf4355933dc921bac165b7}{\+\_\+chk2\+\_\+asarray}} (a, b, axis)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_aeb5f31167da564053a0666386c4e28b3}{gmean}} (a, axis=0, dtype=None, weights=None)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_ab8582abb8de89103fad5ce9260fe5f7e}{hmean}} (a, axis=0, dtype=None, \texorpdfstring{$\ast$}{*}, weights=None)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a055d012b1b109ae0f483c925f5d0e0b8}{pmean}} (a, p, \texorpdfstring{$\ast$}{*}, axis=0, dtype=None, weights=None)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a7151d21a20c6f5406310bf931bf4236e}{\+\_\+mode\+\_\+result}} (\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a8c1c3534a15de7f93b96291982b74ac2}{mode}}, count)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a8c1c3534a15de7f93b96291982b74ac2}{mode}} (a, axis=0, nan\+\_\+policy=\textquotesingle{}propagate\textquotesingle{}, keepdims=False)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_aacd16dbbb70282cb051e7b9556d4737f}{\+\_\+mask\+\_\+to\+\_\+limits}} (a, limits, inclusive)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a429c8fac38c7f007f95416b907858513}{tmean}} (a, limits=None, inclusive=(\mbox{\hyperlink{classTrue}{True}}, \mbox{\hyperlink{classTrue}{True}}), axis=None)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a26ec33f46d65449e3c4e90f85e6f3928}{tvar}} (a, limits=None, inclusive=(\mbox{\hyperlink{classTrue}{True}}, \mbox{\hyperlink{classTrue}{True}}), axis=0, ddof=1)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_ad5fced7efb0eda0d00485572e5cd317d}{tmin}} (a, lowerlimit=None, axis=0, inclusive=\mbox{\hyperlink{classTrue}{True}}, nan\+\_\+policy=\textquotesingle{}propagate\textquotesingle{})
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a3a2094871a926390297df45c2c23902e}{tmax}} (a, upperlimit=None, axis=0, inclusive=\mbox{\hyperlink{classTrue}{True}}, nan\+\_\+policy=\textquotesingle{}propagate\textquotesingle{})
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a556a06abd7478c8d2c7500b2992c51c3}{tstd}} (a, limits=None, inclusive=(\mbox{\hyperlink{classTrue}{True}}, \mbox{\hyperlink{classTrue}{True}}), axis=0, ddof=1)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_ad31c6b6ef9f91a6dfc8e174a87c0e190}{tsem}} (a, limits=None, inclusive=(\mbox{\hyperlink{classTrue}{True}}, \mbox{\hyperlink{classTrue}{True}}), axis=0, ddof=1)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a9e755c164ac589dc31d320fecbe127e2}{\+\_\+moment\+\_\+outputs}} (kwds)
\begin{DoxyCompactList}\small\item\em MOMENTS \#. \end{DoxyCompactList}\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a41ef2a077e3c712981b1a282bf3ee032}{\+\_\+moment\+\_\+result\+\_\+object}} (\texorpdfstring{$\ast$}{*}args)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a325ff018bf983292c0e20569bb6cb81f}{moment}} (a, moment=1, axis=0, nan\+\_\+policy=\textquotesingle{}propagate\textquotesingle{}, \texorpdfstring{$\ast$}{*}, center=None)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a4d962bb75618ecb52d24443b5e443ea8}{\+\_\+moment}} (a, \mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a325ff018bf983292c0e20569bb6cb81f}{moment}}, axis, \texorpdfstring{$\ast$}{*}, mean=None)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a9f8cb83384c519dcefbf564456fe81dd}{\+\_\+var}} (x, axis=0, ddof=0, mean=None)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a57e56bc1f28e285175aa52cd9a8dccec}{skew}} (a, axis=0, bias=\mbox{\hyperlink{classTrue}{True}}, nan\+\_\+policy=\textquotesingle{}propagate\textquotesingle{})
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a9fd18adf3aba669d3c24c5d052097f73}{kurtosis}} (a, axis=0, fisher=\mbox{\hyperlink{classTrue}{True}}, bias=\mbox{\hyperlink{classTrue}{True}}, nan\+\_\+policy=\textquotesingle{}propagate\textquotesingle{})
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a9074c49e4ce1d99373dae0a3c48c9d5e}{describe}} (a, axis=0, ddof=1, bias=\mbox{\hyperlink{classTrue}{True}}, nan\+\_\+policy=\textquotesingle{}propagate\textquotesingle{})
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a0e20026493d260d023531ce41e26c269}{\+\_\+normtest\+\_\+finish}} (z, alternative)
\begin{DoxyCompactList}\small\item\em NORMALITY TESTS \#. \end{DoxyCompactList}\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a7ba4fbe391d511f26a5715213741ce19}{skewtest}} (a, axis=0, nan\+\_\+policy=\textquotesingle{}propagate\textquotesingle{}, alternative=\textquotesingle{}two-\/sided\textquotesingle{})
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a8afd0b93401a2e4378a7f325e2b53cf0}{kurtosistest}} (a, axis=0, nan\+\_\+policy=\textquotesingle{}propagate\textquotesingle{}, alternative=\textquotesingle{}two-\/sided\textquotesingle{})
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a04cd8218975e527a625046de8cf8004d}{normaltest}} (a, axis=0, nan\+\_\+policy=\textquotesingle{}propagate\textquotesingle{})
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a052799c289bd2d586b43e9112ec70e62}{jarque\+\_\+bera}} (x, \texorpdfstring{$\ast$}{*}, axis=None)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a50b35c4e412f628c2f7e90a36bdabf25}{scoreatpercentile}} (a, per, limit=(), interpolation\+\_\+method=\textquotesingle{}fraction\textquotesingle{}, axis=None)
\begin{DoxyCompactList}\small\item\em FREQUENCY FUNCTIONS \#. \end{DoxyCompactList}\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a83387627658b0c178785dca865ecd929}{\+\_\+compute\+\_\+qth\+\_\+percentile}} (sorted\+\_\+, per, interpolation\+\_\+method, axis)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_af139493d2638b97ca3345692a7835fd4}{percentileofscore}} (a, score, kind=\textquotesingle{}rank\textquotesingle{}, nan\+\_\+policy=\textquotesingle{}propagate\textquotesingle{})
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_ae82eb3635bf1ad54e142971200bf9710}{\+\_\+histogram}} (a, numbins=10, defaultlimits=None, weights=None, printextras=False)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_ab7cc05458723f566fbbd8bf5c92ac145}{cumfreq}} (a, numbins=10, defaultreallimits=None, weights=None)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_ad241a1e1cd58da4789b3217d42536d0b}{relfreq}} (a, numbins=10, defaultreallimits=None, weights=None)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_aa080af52b52bdc1e1998e5f15eef174d}{obrientransform}} (\texorpdfstring{$\ast$}{*}samples)
\begin{DoxyCompactList}\small\item\em VARIABILITY FUNCTIONS \#. \end{DoxyCompactList}\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_ac92d7d58643573008b4ddc6dfad7447a}{sem}} (a, axis=0, ddof=1, nan\+\_\+policy=\textquotesingle{}propagate\textquotesingle{})
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a5bd539f56514ee1098218fdfb869cea0}{\+\_\+isconst}} (x)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a4b21f46bcd2e74e726ebab41f7c799c8}{\+\_\+quiet\+\_\+nanmean}} (x)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a51738f78d33d4973d9bb92624d6a5bef}{\+\_\+quiet\+\_\+nanstd}} (x, ddof=0)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a40557c1c2fd14506e84aa782a7382f68}{zscore}} (a, axis=0, ddof=0, nan\+\_\+policy=\textquotesingle{}propagate\textquotesingle{})
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a269b1dd104e02d66f4c84e622f9702d1}{gzscore}} (a, \texorpdfstring{$\ast$}{*}, axis=0, ddof=0, nan\+\_\+policy=\textquotesingle{}propagate\textquotesingle{})
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_abecb96afa85e562e14f2ff4f301a0ef9}{zmap}} (scores, compare, axis=0, ddof=0, nan\+\_\+policy=\textquotesingle{}propagate\textquotesingle{})
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a720304ac94e199a283c3cde4e66448f1}{gstd}} (a, axis=0, ddof=1)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_ad8ae1ff6d5c66e636d72a57ad7ea0472}{iqr}} (x, axis=None, rng=(25, 75), scale=1.\+0, nan\+\_\+policy=\textquotesingle{}propagate\textquotesingle{}, interpolation=\textquotesingle{}linear\textquotesingle{}, keepdims=False)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a12bcb107f34707282163dc34932921d5}{\+\_\+mad\+\_\+1d}} (x, center, nan\+\_\+policy)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a3965fb8fb12386cc993ba9ea132af562}{median\+\_\+abs\+\_\+deviation}} (x, axis=0, center=np.\+median, scale=1.\+0, nan\+\_\+policy=\textquotesingle{}propagate\textquotesingle{})
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a40c059cb94ceb8ed4b4573bca8b8024f}{sigmaclip}} (a, low=4., high=4.)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_af1b232b8ef2a078a8bda0f808ca871b0}{trimboth}} (a, proportiontocut, axis=0)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a133a4a1cf3127cc59ab529a6342e4e3b}{trim1}} (a, proportiontocut, tail=\textquotesingle{}right\textquotesingle{}, axis=0)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a692b1677d3e31fb991495d59a7378aff}{trim\+\_\+mean}} (a, proportiontocut, axis=0)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a2e276e13ea5d7dfc68a7ccb7a5dee6dd}{\+\_\+create\+\_\+f\+\_\+oneway\+\_\+nan\+\_\+result}} (shape, axis)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_ace6e79d444d643e8f8c69359dd2e7a76}{\+\_\+first}} (arr, axis)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a7771f3b2db7c5785f3c8395389ec7b76}{f\+\_\+oneway}} (\texorpdfstring{$\ast$}{*}samples, axis=0)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a25a76cb8c69e3c610d9e9c69fc54e9b9}{alexandergovern}} (\texorpdfstring{$\ast$}{*}samples, nan\+\_\+policy=\textquotesingle{}propagate\textquotesingle{})
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_ab1d9eb219fb20aaeb0a65cee954f5001}{\+\_\+alexandergovern\+\_\+input\+\_\+validation}} (samples, nan\+\_\+policy)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_acaf4e4f934f9bc9561d37ca281b850f5}{\+\_\+pearsonr\+\_\+fisher\+\_\+ci}} (r, n, confidence\+\_\+level, alternative)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_ad005971b96b05500e6f701f95ea0c08a}{\+\_\+pearsonr\+\_\+bootstrap\+\_\+ci}} (confidence\+\_\+level, method, x, y, alternative)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a8b002f05757020441e68edf9441f3c7f}{pearsonr}} (x, y, \texorpdfstring{$\ast$}{*}, alternative=\textquotesingle{}two-\/sided\textquotesingle{}, method=None)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_aec971e2362ba3d1144450412cfb4d674}{fisher\+\_\+exact}} (table, alternative=\textquotesingle{}two-\/sided\textquotesingle{})
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a38625835ee2b2814b8dd9ce450b73bde}{spearmanr}} (a, b=None, axis=0, nan\+\_\+policy=\textquotesingle{}propagate\textquotesingle{}, alternative=\textquotesingle{}two-\/sided\textquotesingle{})
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a92e6eba419d8ef505e1e48f9333d995b}{pointbiserialr}} (x, y)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a40ce454d71767fd548c3da8a056cbd35}{kendalltau}} (x, y, initial\+\_\+lexsort=None, nan\+\_\+policy=\textquotesingle{}propagate\textquotesingle{}, method=\textquotesingle{}auto\textquotesingle{}, variant=\textquotesingle{}b\textquotesingle{}, alternative=\textquotesingle{}two-\/sided\textquotesingle{})
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a6ba4662b53e915f8ac0f275bb2199d4f}{weightedtau}} (x, y, rank=\mbox{\hyperlink{classTrue}{True}}, weigher=None, additive=\mbox{\hyperlink{classTrue}{True}})
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_ad0076d6e995299f62317826f85fe2cd6}{\+\_\+perm\+\_\+test}} (x, y, stat, reps=1000, workers=-\/1, random\+\_\+state=None)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_ac06697b36a6f3c2567d9e514a9147d59}{\+\_\+euclidean\+\_\+dist}} (x)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_aa5e6d4b63720d080eab70d14f43d74cd}{multiscale\+\_\+graphcorr}} (x, y, compute\+\_\+distance=\+\_\+euclidean\+\_\+dist, reps=1000, workers=1, is\+\_\+twosamp=False, random\+\_\+state=None)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a16b0aaa0ec15fefa642eb07a77dcf117}{\+\_\+mgc\+\_\+stat}} (distx, disty)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a436f9fb4bc5898afd29daa2f083de1dc}{\+\_\+threshold\+\_\+mgc\+\_\+map}} (stat\+\_\+mgc\+\_\+map, samp\+\_\+size)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a9d928324e2a00e17b80082984cb5fae2}{\+\_\+smooth\+\_\+mgc\+\_\+map}} (sig\+\_\+connect, stat\+\_\+mgc\+\_\+map)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a8b84fb54a29fcfd00875f50715d876dd}{\+\_\+two\+\_\+sample\+\_\+transform}} (u, v)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a921a917ca3f8ea0c04c6bc7f87d33999}{pack\+\_\+\+Ttest\+Result}} (statistic, pvalue, df, alternative, standard\+\_\+error, estimate)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a0baf8c2dbcc1526d7272efa817c2331c}{unpack\+\_\+\+Ttest\+Result}} (res)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a0b40cba4553c34430dc8745cc657d107}{ttest\+\_\+1samp}} (a, popmean, axis=0, nan\+\_\+policy=\textquotesingle{}propagate\textquotesingle{}, alternative="{}two-\/sided"{})
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_af2976e18baddb8943eebd15f581f9e77}{\+\_\+t\+\_\+confidence\+\_\+interval}} (df, t, confidence\+\_\+level, alternative)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a068fc003ca0623460250659b9421b133}{\+\_\+ttest\+\_\+finish}} (df, t, alternative)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a6d068bbda5a330aa2eeb320506a10c1a}{\+\_\+ttest\+\_\+ind\+\_\+from\+\_\+stats}} (mean1, mean2, denom, df, alternative)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a705b5678e9a5ee4406cdfc24418527a7}{\+\_\+unequal\+\_\+var\+\_\+ttest\+\_\+denom}} (v1, n1, v2, n2)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_aa19d5534576fc2aed9f2cf56fd9b8d65}{\+\_\+equal\+\_\+var\+\_\+ttest\+\_\+denom}} (v1, n1, v2, n2)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a3b98b9c4e131fb2b6d72da0ec90092f9}{ttest\+\_\+ind\+\_\+from\+\_\+stats}} (mean1, std1, nobs1, mean2, std2, nobs2, equal\+\_\+var=\mbox{\hyperlink{classTrue}{True}}, alternative="{}two-\/sided"{})
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a1021af752b49ceb9383493994d38eaef}{ttest\+\_\+ind}} (a, b, axis=0, equal\+\_\+var=\mbox{\hyperlink{classTrue}{True}}, nan\+\_\+policy=\textquotesingle{}propagate\textquotesingle{}, permutations=None, random\+\_\+state=None, alternative="{}two-\/sided"{}, trim=0)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_adbe82260a73fb8c647b4f542dd691b1c}{\+\_\+ttest\+\_\+trim\+\_\+var\+\_\+mean\+\_\+len}} (a, trim, axis)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a506544ecd1507c36cadb7b2412e577dd}{\+\_\+calculate\+\_\+winsorized\+\_\+variance}} (a, g, axis)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_ab3be4c10d446d0f0d68e56a854ba225c}{\+\_\+permutation\+\_\+distribution\+\_\+t}} (data, permutations, size\+\_\+a, equal\+\_\+var, random\+\_\+state=None)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a85145ca01f47db09a5cd8137b027c780}{\+\_\+calc\+\_\+t\+\_\+stat}} (a, b, equal\+\_\+var, axis=-\/1)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a76074b3d866305fa36a138f38ce9611b}{\+\_\+permutation\+\_\+ttest}} (a, b, permutations, axis=0, equal\+\_\+var=\mbox{\hyperlink{classTrue}{True}}, nan\+\_\+policy=\textquotesingle{}propagate\textquotesingle{}, random\+\_\+state=None, alternative="{}two-\/sided"{})
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a2686bf2f43feeb0760374ba16a87dd49}{\+\_\+get\+\_\+len}} (a, axis, msg)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a40e1ede0be1d6f44d3aa61aa9e1be55f}{ttest\+\_\+rel}} (a, b, axis=0, nan\+\_\+policy=\textquotesingle{}propagate\textquotesingle{}, alternative="{}two-\/sided"{})
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a0ee99b9e592bd060aa2218c14a26081b}{\+\_\+count}} (a, axis=None)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_aa7ba35ca822e6023cb6ae415de882378}{\+\_\+m\+\_\+broadcast\+\_\+to}} (a, shape)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a13bf3d4c0145bfe43a1b40826651ea3f}{power\+\_\+divergence}} (f\+\_\+obs, f\+\_\+exp=None, ddof=0, axis=0, lambda\+\_\+=None)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a5794f15c32e4b9eaee99f403e23be4c5}{chisquare}} (f\+\_\+obs, f\+\_\+exp=None, ddof=0, axis=0)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a5bb6b6133ae1147438291126f78813fe}{\+\_\+compute\+\_\+dplus}} (cdfvals, x)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a6cf3d17bbee81a6a1875498ae4c31c6f}{\+\_\+compute\+\_\+dminus}} (cdfvals, x)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_afd9c2533435c2d2b47aa429ee290ae43}{ks\+\_\+1samp}} (x, cdf, args=(), alternative=\textquotesingle{}two-\/sided\textquotesingle{}, method=\textquotesingle{}auto\textquotesingle{})
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_ae8b7cadb3896a61b502dcc260b808ddb}{\+\_\+compute\+\_\+prob\+\_\+outside\+\_\+square}} (n, h)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a7203aa48462f055fd04fbfc629597002}{\+\_\+count\+\_\+paths\+\_\+outside\+\_\+method}} (m, n, g, h)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a2c62a9af6e096d5f25b9abb995c59ea4}{\+\_\+attempt\+\_\+exact\+\_\+2kssamp}} (n1, n2, g, d, alternative)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a171a6b4426dbc47cdb093d05f3f1b969}{ks\+\_\+2samp}} (data1, data2, alternative=\textquotesingle{}two-\/sided\textquotesingle{}, method=\textquotesingle{}auto\textquotesingle{})
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_adb3af01a376d251df82e906d556ffaeb}{\+\_\+parse\+\_\+kstest\+\_\+args}} (data1, data2, args, N)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a918a98d18b72d14b9f81de22e21faa56}{kstest}} (rvs, cdf, args=(), N=20, alternative=\textquotesingle{}two-\/sided\textquotesingle{}, method=\textquotesingle{}auto\textquotesingle{})
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_aa1e457e8eecfa8d5e63416bd9fbf04af}{tiecorrect}} (rankvals)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a5c9e3566461d8bb479cb9507e9a1bd2e}{ranksums}} (x, y, alternative=\textquotesingle{}two-\/sided\textquotesingle{})
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a15cf6e865d5b5a8be777b158f295610d}{kruskal}} (\texorpdfstring{$\ast$}{*}samples, nan\+\_\+policy=\textquotesingle{}propagate\textquotesingle{})
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_aa458038daef3b9b7c308d681b36f93ba}{friedmanchisquare}} (\texorpdfstring{$\ast$}{*}samples)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a61540456aa6cd16b909618e175d798f9}{brunnermunzel}} (x, y, alternative="{}two-\/sided"{}, distribution="{}t"{}, nan\+\_\+policy=\textquotesingle{}propagate\textquotesingle{})
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a0b6200c846acee0166b2ecb13abdcba1}{combine\+\_\+pvalues}} (pvalues, method=\textquotesingle{}fisher\textquotesingle{}, weights=None)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a030f7a94530750520b7f11137b0f88c6}{wasserstein\+\_\+distance}} (u\+\_\+values, v\+\_\+values, u\+\_\+weights=None, v\+\_\+weights=None)
\begin{DoxyCompactList}\small\item\em STATISTICAL DISTANCES \#. \end{DoxyCompactList}\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a348efe926c611bfc19391b531685b053}{energy\+\_\+distance}} (u\+\_\+values, v\+\_\+values, u\+\_\+weights=None, v\+\_\+weights=None)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a3f8344423e0575b4b92a28d1aca90279}{\+\_\+cdf\+\_\+distance}} (p, u\+\_\+values, v\+\_\+values, u\+\_\+weights=None, v\+\_\+weights=None)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a5946b7785fb379d1de1cc47b0b338faf}{\+\_\+validate\+\_\+distribution}} (values, weights)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a6394edc619f56ba4b055ea6050b269e6}{find\+\_\+repeats}} (arr)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a35baa494b62135807260b2445335c9f9}{\+\_\+sum\+\_\+of\+\_\+squares}} (a, axis=0)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a4f95272db5e6cd266ad36a67e1353b18}{\+\_\+square\+\_\+of\+\_\+sums}} (a, axis=0)
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_aa3911ef47b2d3dd65d7fc164b15a4799}{rankdata}} (a, method=\textquotesingle{}average\textquotesingle{}, \texorpdfstring{$\ast$}{*}, axis=None, nan\+\_\+policy=\textquotesingle{}propagate\textquotesingle{})
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a5ab039579d581807121a4bd4b1cd14f4}{expectile}} (a, alpha=0.\+5, \texorpdfstring{$\ast$}{*}, weights=None)
\end{DoxyCompactItemize}
\doxysubsubsection*{Variables}
\begin{DoxyCompactItemize}
\item 
list \mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_ab762f81902847b88a6822c3eca3bf77c}{\+\_\+\+\_\+all\+\_\+\+\_\+}}
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a99e444e3ed4cb89551065e638d8e25ba}{Significance\+Result}}
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a8276a1473f84d29df1cfb05a798c6c86}{Mode\+Result}} = \mbox{\hyperlink{classnamedtuple}{namedtuple}}(\textquotesingle{}Mode\+Result\textquotesingle{}, (\textquotesingle{}\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a8c1c3534a15de7f93b96291982b74ac2}{mode}}\textquotesingle{}, \textquotesingle{}count\textquotesingle{}))
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_ad17cbb60a06b3b961b8d3dec3f75f571}{Describe\+Result}}
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a517833de500c51cf8468fe2ccee2b20d}{Skewtest\+Result}} = \mbox{\hyperlink{classnamedtuple}{namedtuple}}(\textquotesingle{}Skewtest\+Result\textquotesingle{}, (\textquotesingle{}statistic\textquotesingle{}, \textquotesingle{}pvalue\textquotesingle{}))
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a4db485cb945393c294a5defd286e49bb}{Kurtosistest\+Result}} = \mbox{\hyperlink{classnamedtuple}{namedtuple}}(\textquotesingle{}Kurtosistest\+Result\textquotesingle{}, (\textquotesingle{}statistic\textquotesingle{}, \textquotesingle{}pvalue\textquotesingle{}))
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a7ecccad49a9bf1d4db7baeba60ed6c54}{Normaltest\+Result}} = \mbox{\hyperlink{classnamedtuple}{namedtuple}}(\textquotesingle{}Normaltest\+Result\textquotesingle{}, (\textquotesingle{}statistic\textquotesingle{}, \textquotesingle{}pvalue\textquotesingle{}))
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a159d3500af93794f6cb4a9f959924bb5}{Histogram\+Result}}
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a1a75604ce103a4f3a8661da14b41ba21}{Cumfreq\+Result}}
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a0739c0c0192b367b27d6d1fc6b076469}{Relfreq\+Result}}
\item 
dict \mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a1d8ab9655add9064cf78ddbfc338cbb2}{\+\_\+scale\+\_\+conversions}}
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_ade3bb71bac397106f36f0ebe2f76ff8e}{Sigmaclip\+Result}} = \mbox{\hyperlink{classnamedtuple}{namedtuple}}(\textquotesingle{}Sigmaclip\+Result\textquotesingle{}, (\textquotesingle{}clipped\textquotesingle{}, \textquotesingle{}lower\textquotesingle{}, \textquotesingle{}upper\textquotesingle{}))
\begin{DoxyCompactList}\small\item\em TRIMMING FUNCTIONS \#. \end{DoxyCompactList}\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a55dd91ad97a133a4f1c4014a850145ea}{F\+\_\+oneway\+Result}} = \mbox{\hyperlink{classnamedtuple}{namedtuple}}(\textquotesingle{}F\+\_\+oneway\+Result\textquotesingle{}, (\textquotesingle{}statistic\textquotesingle{}, \textquotesingle{}pvalue\textquotesingle{}))
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_ae4b1ccd5e913ab32dee4d40b27b90992}{Confidence\+Interval}} = \mbox{\hyperlink{classnamedtuple}{namedtuple}}(\textquotesingle{}Confidence\+Interval\textquotesingle{}, \mbox{[}\textquotesingle{}low\textquotesingle{}, \textquotesingle{}high\textquotesingle{}\mbox{]})
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_ad063c82a38b3f71d713793c4565ca702}{Pearson\+RResult\+Base}}
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a79b67a68f52e541cf43ed09574e13108}{MGCResult}}
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_acf46192a3335b4aa0f1a5fdc412ecc64}{Ttest\+Result\+Base}}
\begin{DoxyCompactList}\small\item\em INFERENTIAL STATISTICS \#. \end{DoxyCompactList}\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_ab6b754f522c03e7fccd72c806de40407}{Ttest\+\_\+ind\+Result}} = \mbox{\hyperlink{classnamedtuple}{namedtuple}}(\textquotesingle{}Ttest\+\_\+ind\+Result\textquotesingle{}, (\textquotesingle{}statistic\textquotesingle{}, \textquotesingle{}pvalue\textquotesingle{}))
\item 
dict \mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a8a1eac21b5e82f9848b2523b6b90995e}{\+\_\+power\+\_\+div\+\_\+lambda\+\_\+names}}
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a0b74f644917a36fcd23aa95f48810815}{Power\+\_\+divergence\+Result}}
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_af6c2563e8e2c6e7d3e19fa32131a6796}{Kstest\+Result}}
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a4c3e4d7762108939709c279fff05dcb4}{Ks\+\_\+2samp\+Result}} = Kstest\+Result
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a13929c3e11b3aea010483ae368fdeeea}{Ranksums\+Result}} = \mbox{\hyperlink{classnamedtuple}{namedtuple}}(\textquotesingle{}Ranksums\+Result\textquotesingle{}, (\textquotesingle{}statistic\textquotesingle{}, \textquotesingle{}pvalue\textquotesingle{}))
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_ac9cba6ca5b344ceda11181bcd5f6adea}{Kruskal\+Result}} = \mbox{\hyperlink{classnamedtuple}{namedtuple}}(\textquotesingle{}Kruskal\+Result\textquotesingle{}, (\textquotesingle{}statistic\textquotesingle{}, \textquotesingle{}pvalue\textquotesingle{}))
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_acf31b33400d148f6ef0e0f3b66f5d2be}{Friedmanchisquare\+Result}}
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a084547535521fc8a4e2ca13da7ccf236}{Brunner\+Munzel\+Result}}
\item 
\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a5abc71d3f4e3d43499eea3985b7d2bd9}{Repeated\+Results}} = \mbox{\hyperlink{classnamedtuple}{namedtuple}}(\textquotesingle{}Repeated\+Results\textquotesingle{}, (\textquotesingle{}values\textquotesingle{}, \textquotesingle{}counts\textquotesingle{}))
\begin{DoxyCompactList}\small\item\em SUPPORT FUNCTIONS \#. \end{DoxyCompactList}\end{DoxyCompactItemize}


\doxysubsection{Detailed Description}
\begin{DoxyVerb}A collection of basic statistical functions for Python.

References
----------
.. [CRCProbStat2000] Zwillinger, D. and Kokoska, S. (2000). CRC Standard
   Probability and Statistics Tables and Formulae. Chapman & Hall: New
   York. 2000.
\end{DoxyVerb}
 

\doxysubsection{Function Documentation}
\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_ab1d9eb219fb20aaeb0a65cee954f5001}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_alexandergovern\_input\_validation@{\_alexandergovern\_input\_validation}}
\index{\_alexandergovern\_input\_validation@{\_alexandergovern\_input\_validation}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_alexandergovern\_input\_validation()}{\_alexandergovern\_input\_validation()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_ab1d9eb219fb20aaeb0a65cee954f5001} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+alexandergovern\+\_\+input\+\_\+validation (\begin{DoxyParamCaption}\item[{}]{samples}{, }\item[{}]{nan\+\_\+policy}{}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}



Definition at line \mbox{\hyperlink{__stats__py_8py_source_l04359}{4359}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a2c62a9af6e096d5f25b9abb995c59ea4}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_attempt\_exact\_2kssamp@{\_attempt\_exact\_2kssamp}}
\index{\_attempt\_exact\_2kssamp@{\_attempt\_exact\_2kssamp}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_attempt\_exact\_2kssamp()}{\_attempt\_exact\_2kssamp()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a2c62a9af6e096d5f25b9abb995c59ea4} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+attempt\+\_\+exact\+\_\+2kssamp (\begin{DoxyParamCaption}\item[{}]{n1}{, }\item[{}]{n2}{, }\item[{}]{g}{, }\item[{}]{d}{, }\item[{}]{alternative}{}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}

\begin{DoxyVerb}Attempts to compute the exact 2sample probability.

n1, n2 are the sample sizes
g is the gcd(n1, n2)
d is the computed max difference in ECDFs

Returns (success, d, probability)
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l08596}{8596}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l08484}{\+\_\+compute\+\_\+prob\+\_\+outside\+\_\+square()}}, and \mbox{\hyperlink{__stats__py_8py_source_l08522}{\+\_\+count\+\_\+paths\+\_\+outside\+\_\+method()}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l08645}{ks\+\_\+2samp()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a85145ca01f47db09a5cd8137b027c780}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_calc\_t\_stat@{\_calc\_t\_stat}}
\index{\_calc\_t\_stat@{\_calc\_t\_stat}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_calc\_t\_stat()}{\_calc\_t\_stat()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a85145ca01f47db09a5cd8137b027c780} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+calc\+\_\+t\+\_\+stat (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{b}{, }\item[{}]{equal\+\_\+var}{, }\item[{}]{axis}{ = {\ttfamily -\/1}}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}

\begin{DoxyVerb}Calculate the t statistic along the given dimension.\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l07579}{7579}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l07542}{\+\_\+permutation\+\_\+distribution\+\_\+t()}}, and \mbox{\hyperlink{__stats__py_8py_source_l07598}{\+\_\+permutation\+\_\+ttest()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a506544ecd1507c36cadb7b2412e577dd}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_calculate\_winsorized\_variance@{\_calculate\_winsorized\_variance}}
\index{\_calculate\_winsorized\_variance@{\_calculate\_winsorized\_variance}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_calculate\_winsorized\_variance()}{\_calculate\_winsorized\_variance()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a506544ecd1507c36cadb7b2412e577dd} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+calculate\+\_\+winsorized\+\_\+variance (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{g}{, }\item[{}]{axis}{}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}

\begin{DoxyVerb}Calculates g-times winsorized variance along specified axis\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l07506}{7506}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l07482}{\+\_\+ttest\+\_\+trim\+\_\+var\+\_\+mean\+\_\+len()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a3f8344423e0575b4b92a28d1aca90279}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_cdf\_distance@{\_cdf\_distance}}
\index{\_cdf\_distance@{\_cdf\_distance}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_cdf\_distance()}{\_cdf\_distance()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a3f8344423e0575b4b92a28d1aca90279} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+cdf\+\_\+distance (\begin{DoxyParamCaption}\item[{}]{p}{, }\item[{}]{u\+\_\+values}{, }\item[{}]{v\+\_\+values}{, }\item[{}]{u\+\_\+weights}{ = {\ttfamily None}, }\item[{}]{v\+\_\+weights}{ = {\ttfamily None}}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}

\begin{DoxyVerb}    Compute, between two one-dimensional distributions :math:`u` and
    :math:`v`, whose respective CDFs are :math:`U` and :math:`V`, the
    statistical distance that is defined as:

    .. math::

        l_p(u, v) = \left( \int_{-\infty}^{+\infty} |U-V|^p \right)^{1/p}

    p is a positive parameter; p = 1 gives the Wasserstein distance, p = 2
    gives the energy distance.

    Parameters
    ----------
    u_values, v_values : array_like
        Values observed in the (empirical) distribution.
    u_weights, v_weights : array_like, optional
        Weight for each value. If unspecified, each value is assigned the same
        weight.
        `u_weights` (resp. `v_weights`) must have the same length as
        `u_values` (resp. `v_values`). If the weight sum differs from 1, it
        must still be positive and finite so that the weights can be normalized
        to sum to 1.

    Returns
    -------
    distance : float
        The computed distance between the distributions.

    Notes
    -----
    The input distributions can be empirical, therefore coming from samples
    whose values are effectively inputs of the function, or they can be seen as
    generalized functions, in which case they are weighted sums of Dirac delta
    functions located at the specified values.

    References
    ----------
    .. [1] Bellemare, Danihelka, Dabney, Mohamed, Lakshminarayanan, Hoyer,
           Munos "The Cramer Distance as a Solution to Biased Wasserstein
           Gradients" (2017). :arXiv:`1705.10743`.
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l09896}{9896}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l09983}{\+\_\+validate\+\_\+distribution()}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l09810}{energy\+\_\+distance()}}, and \mbox{\hyperlink{__stats__py_8py_source_l09733}{wasserstein\+\_\+distance()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_aa7b513e10adf4355933dc921bac165b7}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_chk2\_asarray@{\_chk2\_asarray}}
\index{\_chk2\_asarray@{\_chk2\_asarray}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_chk2\_asarray()}{\_chk2\_asarray()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_aa7b513e10adf4355933dc921bac165b7} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+chk2\+\_\+asarray (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{b}{, }\item[{}]{axis}{}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}



Definition at line \mbox{\hyperlink{__stats__py_8py_source_l00104}{104}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_aaa8f1f67d10a05713172b6a32169ed91}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_chk\_asarray@{\_chk\_asarray}}
\index{\_chk\_asarray@{\_chk\_asarray}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_chk\_asarray()}{\_chk\_asarray()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_aaa8f1f67d10a05713172b6a32169ed91} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+chk\+\_\+asarray (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{axis}{}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}



Definition at line \mbox{\hyperlink{__stats__py_8py_source_l00090}{90}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a6cf3d17bbee81a6a1875498ae4c31c6f}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_compute\_dminus@{\_compute\_dminus}}
\index{\_compute\_dminus@{\_compute\_dminus}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_compute\_dminus()}{\_compute\_dminus()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a6cf3d17bbee81a6a1875498ae4c31c6f} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+compute\+\_\+dminus (\begin{DoxyParamCaption}\item[{}]{cdfvals}{, }\item[{}]{x}{}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}

\begin{DoxyVerb}Computes D- as used in the Kolmogorov-Smirnov test.

Parameters
----------
cdfvals : array_like
    Sorted array of CDF values between 0 and 1
x: array_like
    Sorted array of the stochastic variable itself

Returns
-------
res: Pair with the following elements:
    - Maximum distance of the CDF values above Uniform(0, 1)
    - The location at which the maximum is reached.
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l08286}{8286}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l08310}{ks\+\_\+1samp()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a5bb6b6133ae1147438291126f78813fe}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_compute\_dplus@{\_compute\_dplus}}
\index{\_compute\_dplus@{\_compute\_dplus}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_compute\_dplus()}{\_compute\_dplus()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a5bb6b6133ae1147438291126f78813fe} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+compute\+\_\+dplus (\begin{DoxyParamCaption}\item[{}]{cdfvals}{, }\item[{}]{x}{}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}

\begin{DoxyVerb}Computes D+ as used in the Kolmogorov-Smirnov test.

Parameters
----------
cdfvals : array_like
    Sorted array of CDF values between 0 and 1
x: array_like
    Sorted array of the stochastic variable itself

Returns
-------
res: Pair with the following elements:
    - The maximum distance of the CDF values below Uniform(0, 1).
    - The location at which the maximum is reached.
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l08262}{8262}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l08310}{ks\+\_\+1samp()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_ae8b7cadb3896a61b502dcc260b808ddb}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_compute\_prob\_outside\_square@{\_compute\_prob\_outside\_square}}
\index{\_compute\_prob\_outside\_square@{\_compute\_prob\_outside\_square}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_compute\_prob\_outside\_square()}{\_compute\_prob\_outside\_square()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_ae8b7cadb3896a61b502dcc260b808ddb} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+compute\+\_\+prob\+\_\+outside\+\_\+square (\begin{DoxyParamCaption}\item[{}]{n}{, }\item[{}]{h}{}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}

\begin{DoxyVerb}Compute the proportion of paths that pass outside the two diagonal lines.

Parameters
----------
n : integer
    n > 0
h : integer
    0 <= h <= n

Returns
-------
p : float
    The proportion of paths that pass outside the lines x-y = +/-h.
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l08484}{8484}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l08596}{\+\_\+attempt\+\_\+exact\+\_\+2kssamp()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a83387627658b0c178785dca865ecd929}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_compute\_qth\_percentile@{\_compute\_qth\_percentile}}
\index{\_compute\_qth\_percentile@{\_compute\_qth\_percentile}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_compute\_qth\_percentile()}{\_compute\_qth\_percentile()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a83387627658b0c178785dca865ecd929} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+compute\+\_\+qth\+\_\+percentile (\begin{DoxyParamCaption}\item[{}]{sorted\+\_\+}{, }\item[{}]{per}{, }\item[{}]{interpolation\+\_\+method}{, }\item[{}]{axis}{}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}



Definition at line \mbox{\hyperlink{__stats__py_8py_source_l02249}{2249}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a0ee99b9e592bd060aa2218c14a26081b}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_count@{\_count}}
\index{\_count@{\_count}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_count()}{\_count()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a0ee99b9e592bd060aa2218c14a26081b} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+count (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{axis}{ = {\ttfamily None}}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}

\begin{DoxyVerb}Count the number of non-masked elements of an array.

This function behaves like `np.ma.count`, but is much faster
for ndarrays.
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l07826}{7826}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l07857}{power\+\_\+divergence()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a7203aa48462f055fd04fbfc629597002}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_count\_paths\_outside\_method@{\_count\_paths\_outside\_method}}
\index{\_count\_paths\_outside\_method@{\_count\_paths\_outside\_method}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_count\_paths\_outside\_method()}{\_count\_paths\_outside\_method()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a7203aa48462f055fd04fbfc629597002} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+count\+\_\+paths\+\_\+outside\+\_\+method (\begin{DoxyParamCaption}\item[{}]{m}{, }\item[{}]{n}{, }\item[{}]{g}{, }\item[{}]{h}{}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}

\begin{DoxyVerb}Count the number of paths that pass outside the specified diagonal.

Parameters
----------
m : integer
    m > 0
n : integer
    n > 0
g : integer
    g is greatest common divisor of m and n
h : integer
    0 <= h <= lcm(m,n)

Returns
-------
p : float
    The number of paths that go low.
    The calculation may overflow - check for a finite answer.

Notes
-----
Count the integer lattice paths from (0, 0) to (m, n), which at some
point (x, y) along the path, satisfy:
  m*y <= n*x - h*g
The paths make steps of size +1 in either positive x or positive y
directions.

We generally follow Hodges' treatment of Drion/Gnedenko/Korolyuk.
Hodges, J.L. Jr.,
"The Significance Probability of the Smirnov Two-Sample Test,"
Arkiv fiur Matematik, 3, No. 43 (1958), 469-86.
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l08522}{8522}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l08596}{\+\_\+attempt\+\_\+exact\+\_\+2kssamp()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a2e276e13ea5d7dfc68a7ccb7a5dee6dd}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_create\_f\_oneway\_nan\_result@{\_create\_f\_oneway\_nan\_result}}
\index{\_create\_f\_oneway\_nan\_result@{\_create\_f\_oneway\_nan\_result}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_create\_f\_oneway\_nan\_result()}{\_create\_f\_oneway\_nan\_result()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a2e276e13ea5d7dfc68a7ccb7a5dee6dd} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+create\+\_\+f\+\_\+oneway\+\_\+nan\+\_\+result (\begin{DoxyParamCaption}\item[{}]{shape}{, }\item[{}]{axis}{}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}

\begin{DoxyVerb}This is a helper function for f_oneway for creating the return values
in certain degenerate conditions.  It creates return values that are
all nan with the appropriate shape for the given `shape` and `axis`.
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l03963}{3963}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l03985}{f\+\_\+oneway()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_aa19d5534576fc2aed9f2cf56fd9b8d65}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_equal\_var\_ttest\_denom@{\_equal\_var\_ttest\_denom}}
\index{\_equal\_var\_ttest\_denom@{\_equal\_var\_ttest\_denom}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_equal\_var\_ttest\_denom()}{\_equal\_var\_ttest\_denom()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_aa19d5534576fc2aed9f2cf56fd9b8d65} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+equal\+\_\+var\+\_\+ttest\+\_\+denom (\begin{DoxyParamCaption}\item[{}]{v1}{, }\item[{}]{n1}{, }\item[{}]{v2}{, }\item[{}]{n2}{}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}



Definition at line \mbox{\hyperlink{__stats__py_8py_source_l07020}{7020}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_ac06697b36a6f3c2567d9e514a9147d59}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_euclidean\_dist@{\_euclidean\_dist}}
\index{\_euclidean\_dist@{\_euclidean\_dist}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_euclidean\_dist()}{\_euclidean\_dist()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_ac06697b36a6f3c2567d9e514a9147d59} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+euclidean\+\_\+dist (\begin{DoxyParamCaption}\item[{}]{x}{}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}



Definition at line \mbox{\hyperlink{__stats__py_8py_source_l06216}{6216}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_ace6e79d444d643e8f8c69359dd2e7a76}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_first@{\_first}}
\index{\_first@{\_first}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_first()}{\_first()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_ace6e79d444d643e8f8c69359dd2e7a76} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+first (\begin{DoxyParamCaption}\item[{}]{arr}{, }\item[{}]{axis}{}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}

\begin{DoxyVerb}Return arr[..., 0:1, ...] where 0:1 is in the `axis` position.\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l03980}{3980}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l03985}{f\+\_\+oneway()}}, and \mbox{\hyperlink{__stats__py_8py_source_l03070}{zmap()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a2686bf2f43feeb0760374ba16a87dd49}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_get\_len@{\_get\_len}}
\index{\_get\_len@{\_get\_len}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_get\_len()}{\_get\_len()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a2686bf2f43feeb0760374ba16a87dd49} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+get\+\_\+len (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{axis}{, }\item[{}]{msg}{}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}



Definition at line \mbox{\hyperlink{__stats__py_8py_source_l07671}{7671}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_ae82eb3635bf1ad54e142971200bf9710}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_histogram@{\_histogram}}
\index{\_histogram@{\_histogram}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_histogram()}{\_histogram()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_ae82eb3635bf1ad54e142971200bf9710} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+histogram (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{numbins}{ = {\ttfamily 10}, }\item[{}]{defaultlimits}{ = {\ttfamily None}, }\item[{}]{weights}{ = {\ttfamily None}, }\item[{}]{printextras}{ = {\ttfamily False}}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}

\begin{DoxyVerb}Create a histogram.

Separate the range into several bins and return the number of instances
in each bin.

Parameters
----------
a : array_like
    Array of scores which will be put into bins.
numbins : int, optional
    The number of bins to use for the histogram. Default is 10.
defaultlimits : tuple (lower, upper), optional
    The lower and upper values for the range of the histogram.
    If no value is given, a range slightly larger than the range of the
    values in a is used. Specifically ``(a.min() - s, a.max() + s)``,
    where ``s = (1/2)(a.max() - a.min()) / (numbins - 1)``.
weights : array_like, optional
    The weights for each value in `a`. Default is None, which gives each
    value a weight of 1.0
printextras : bool, optional
    If True, if there are extra points (i.e. the points that fall outside
    the bin limits) a warning is raised saying how many of those points
    there are.  Default is False.

Returns
-------
count : ndarray
    Number of points (or sum of weights) in each bin.
lowerlimit : float
    Lowest value of histogram, the lower limit of the first bin.
binsize : float
    The size of the bins (all bins have the same size).
extrapoints : int
    The number of points outside the range of the histogram.

See Also
--------
numpy.histogram

Notes
-----
This histogram is based on numpy's histogram but has a larger range by
default if default limits is not set.
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l02459}{2459}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l02542}{cumfreq()}}, and \mbox{\hyperlink{__stats__py_8py_source_l02624}{relfreq()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a5bd539f56514ee1098218fdfb869cea0}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_isconst@{\_isconst}}
\index{\_isconst@{\_isconst}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_isconst()}{\_isconst()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a5bd539f56514ee1098218fdfb869cea0} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+isconst (\begin{DoxyParamCaption}\item[{}]{x}{}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}

\begin{DoxyVerb}Check if all values in x are the same.  nans are ignored.

x must be a 1d array.

The return value is a 1d array with length 1, so it can be used
in np.apply_along_axis.
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l02847}{2847}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l03070}{zmap()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_aa7ba35ca822e6023cb6ae415de882378}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_m\_broadcast\_to@{\_m\_broadcast\_to}}
\index{\_m\_broadcast\_to@{\_m\_broadcast\_to}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_m\_broadcast\_to()}{\_m\_broadcast\_to()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_aa7ba35ca822e6023cb6ae415de882378} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+m\+\_\+broadcast\+\_\+to (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{shape}{}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}



Definition at line \mbox{\hyperlink{__stats__py_8py_source_l07846}{7846}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a12bcb107f34707282163dc34932921d5}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_mad\_1d@{\_mad\_1d}}
\index{\_mad\_1d@{\_mad\_1d}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_mad\_1d()}{\_mad\_1d()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a12bcb107f34707282163dc34932921d5} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+mad\+\_\+1d (\begin{DoxyParamCaption}\item[{}]{x}{, }\item[{}]{center}{, }\item[{}]{nan\+\_\+policy}{}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}



Definition at line \mbox{\hyperlink{__stats__py_8py_source_l03464}{3464}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_aacd16dbbb70282cb051e7b9556d4737f}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_mask\_to\_limits@{\_mask\_to\_limits}}
\index{\_mask\_to\_limits@{\_mask\_to\_limits}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_mask\_to\_limits()}{\_mask\_to\_limits()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_aacd16dbbb70282cb051e7b9556d4737f} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+mask\+\_\+to\+\_\+limits (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{limits}{, }\item[{}]{inclusive}{}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}

\begin{DoxyVerb}Mask an array for values outside of given limits.

This is primarily a utility function.

Parameters
----------
a : array
limits : (float or None, float or None)
    A tuple consisting of the (lower limit, upper limit).  Values in the
    input array less than the lower limit or greater than the upper limit
    will be masked out. None implies no limit.
inclusive : (bool, bool)
    A tuple consisting of the (lower flag, upper flag).  These flags
    determine whether values exactly equal to lower or upper are allowed.

Returns
-------
A MaskedArray.

Raises
------
A ValueError if there are no values within the given limits.
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l00522}{522}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l00736}{tmax()}}, \mbox{\hyperlink{__stats__py_8py_source_l00568}{tmean()}}, \mbox{\hyperlink{__stats__py_8py_source_l00673}{tmin()}}, \mbox{\hyperlink{__stats__py_8py_source_l00847}{tsem()}}, and \mbox{\hyperlink{__stats__py_8py_source_l00618}{tvar()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a16b0aaa0ec15fefa642eb07a77dcf117}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_mgc\_stat@{\_mgc\_stat}}
\index{\_mgc\_stat@{\_mgc\_stat}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_mgc\_stat()}{\_mgc\_stat()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a16b0aaa0ec15fefa642eb07a77dcf117} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+mgc\+\_\+stat (\begin{DoxyParamCaption}\item[{}]{distx}{, }\item[{}]{disty}{}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}

\begin{DoxyVerb}Helper function that calculates the MGC stat. See above for use.

Parameters
----------
distx, disty : ndarray
    `distx` and `disty` have shapes `(n, p)` and `(n, q)` or
    `(n, n)` and `(n, n)`
    if distance matrices.

Returns
-------
stat : float
    The sample MGC test statistic within `[-1, 1]`.
stat_dict : dict
    Contains additional useful additional returns containing the following
    keys:

        - stat_mgc_map : ndarray
            MGC-map of the statistics.
        - opt_scale : (float, float)
            The estimated optimal scale as a `(x, y)` pair.
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l06509}{6509}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l06603}{\+\_\+smooth\+\_\+mgc\+\_\+map()}}, and \mbox{\hyperlink{__stats__py_8py_source_l06558}{\+\_\+threshold\+\_\+mgc\+\_\+map()}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l06225}{multiscale\+\_\+graphcorr()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a7151d21a20c6f5406310bf931bf4236e}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_mode\_result@{\_mode\_result}}
\index{\_mode\_result@{\_mode\_result}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_mode\_result()}{\_mode\_result()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a7151d21a20c6f5406310bf931bf4236e} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+mode\+\_\+result (\begin{DoxyParamCaption}\item[{}]{mode}{, }\item[{}]{count}{}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}



Definition at line \mbox{\hyperlink{__stats__py_8py_source_l00428}{428}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a4d962bb75618ecb52d24443b5e443ea8}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_moment@{\_moment}}
\index{\_moment@{\_moment}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_moment()}{\_moment()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a4d962bb75618ecb52d24443b5e443ea8} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+moment (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{moment}{, }\item[{}]{axis}{, }\item[{\texorpdfstring{$\ast$}{*}}]{}{, }\item[{}]{mean}{ = {\ttfamily None}}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}



Definition at line \mbox{\hyperlink{__stats__py_8py_source_l01035}{1035}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a9e755c164ac589dc31d320fecbe127e2}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_moment\_outputs@{\_moment\_outputs}}
\index{\_moment\_outputs@{\_moment\_outputs}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_moment\_outputs()}{\_moment\_outputs()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a9e755c164ac589dc31d320fecbe127e2} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+moment\+\_\+outputs (\begin{DoxyParamCaption}\item[{}]{kwds}{}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}



MOMENTS \#. 



Definition at line \mbox{\hyperlink{__stats__py_8py_source_l00907}{907}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a41ef2a077e3c712981b1a282bf3ee032}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_moment\_result\_object@{\_moment\_result\_object}}
\index{\_moment\_result\_object@{\_moment\_result\_object}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_moment\_result\_object()}{\_moment\_result\_object()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a41ef2a077e3c712981b1a282bf3ee032} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+moment\+\_\+result\+\_\+object (\begin{DoxyParamCaption}\item[{\texorpdfstring{$\ast$}{*}}]{args}{}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}



Definition at line \mbox{\hyperlink{__stats__py_8py_source_l00915}{915}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a0e20026493d260d023531ce41e26c269}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_normtest\_finish@{\_normtest\_finish}}
\index{\_normtest\_finish@{\_normtest\_finish}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_normtest\_finish()}{\_normtest\_finish()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a0e20026493d260d023531ce41e26c269} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+normtest\+\_\+finish (\begin{DoxyParamCaption}\item[{}]{z}{, }\item[{}]{alternative}{}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}



NORMALITY TESTS \#. 

\begin{DoxyVerb}Common code between all the normality-test functions.\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l01415}{1415}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



Referenced by \mbox{\hyperlink{__hypotests_8py_source_l00671}{scipy.\+stats.\+\_\+hypotests.\+\_\+somers\+\_\+d()}}, \mbox{\hyperlink{__morestats_8py_source_l02511}{scipy.\+stats.\+\_\+morestats.\+ansari()}}, \mbox{\hyperlink{__mstats__basic_8py_source_l00777}{scipy.\+stats.\+\_\+mstats\+\_\+basic.\+kendalltau()}}, \mbox{\hyperlink{__stats__py_8py_source_l05595}{kendalltau()}}, \mbox{\hyperlink{__mstats__basic_8py_source_l02967}{scipy.\+stats.\+\_\+mstats\+\_\+basic.\+kurtosistest()}}, \mbox{\hyperlink{__stats__py_8py_source_l01627}{kurtosistest()}}, \mbox{\hyperlink{__morestats_8py_source_l03657}{scipy.\+stats.\+\_\+morestats.\+mood()}}, \mbox{\hyperlink{__stats__py_8py_source_l09133}{ranksums()}}, \mbox{\hyperlink{__mstats__basic_8py_source_l02906}{scipy.\+stats.\+\_\+mstats\+\_\+basic.\+skewtest()}}, and \mbox{\hyperlink{__stats__py_8py_source_l01436}{skewtest()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_adb3af01a376d251df82e906d556ffaeb}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_parse\_kstest\_args@{\_parse\_kstest\_args}}
\index{\_parse\_kstest\_args@{\_parse\_kstest\_args}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_parse\_kstest\_args()}{\_parse\_kstest\_args()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_adb3af01a376d251df82e906d556ffaeb} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+parse\+\_\+kstest\+\_\+args (\begin{DoxyParamCaption}\item[{}]{data1}{, }\item[{}]{data2}{, }\item[{}]{args}{, }\item[{}]{N}{}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}



Definition at line \mbox{\hyperlink{__stats__py_8py_source_l08878}{8878}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_ad005971b96b05500e6f701f95ea0c08a}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_pearsonr\_bootstrap\_ci@{\_pearsonr\_bootstrap\_ci}}
\index{\_pearsonr\_bootstrap\_ci@{\_pearsonr\_bootstrap\_ci}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_pearsonr\_bootstrap\_ci()}{\_pearsonr\_bootstrap\_ci()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_ad005971b96b05500e6f701f95ea0c08a} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+pearsonr\+\_\+bootstrap\+\_\+ci (\begin{DoxyParamCaption}\item[{}]{confidence\+\_\+level}{, }\item[{}]{method}{, }\item[{}]{x}{, }\item[{}]{y}{, }\item[{}]{alternative}{}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}

\begin{DoxyVerb}Compute the confidence interval for Pearson's R using the bootstrap.
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l04426}{4426}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l04533}{pearsonr()}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l04476}{scipy.\+stats.\+\_\+stats\+\_\+py.\+Pearson\+RResult.\+confidence\+\_\+interval()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_acaf4e4f934f9bc9561d37ca281b850f5}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_pearsonr\_fisher\_ci@{\_pearsonr\_fisher\_ci}}
\index{\_pearsonr\_fisher\_ci@{\_pearsonr\_fisher\_ci}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_pearsonr\_fisher\_ci()}{\_pearsonr\_fisher\_ci()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_acaf4e4f934f9bc9561d37ca281b850f5} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+pearsonr\+\_\+fisher\+\_\+ci (\begin{DoxyParamCaption}\item[{}]{r}{, }\item[{}]{n}{, }\item[{}]{confidence\+\_\+level}{, }\item[{}]{alternative}{}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}

\begin{DoxyVerb}Compute the confidence interval for Pearson's R.

Fisher's transformation is used to compute the confidence interval
(https://en.wikipedia.org/wiki/Fisher_transformation).
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l04387}{4387}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l04476}{scipy.\+stats.\+\_\+stats\+\_\+py.\+Pearson\+RResult.\+confidence\+\_\+interval()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_ad0076d6e995299f62317826f85fe2cd6}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_perm\_test@{\_perm\_test}}
\index{\_perm\_test@{\_perm\_test}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_perm\_test()}{\_perm\_test()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_ad0076d6e995299f62317826f85fe2cd6} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+perm\+\_\+test (\begin{DoxyParamCaption}\item[{}]{x}{, }\item[{}]{y}{, }\item[{}]{stat}{, }\item[{}]{reps}{ = {\ttfamily 1000}, }\item[{}]{workers}{ = {\ttfamily -\/1}, }\item[{}]{random\+\_\+state}{ = {\ttfamily None}}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}

\begin{DoxyVerb}Helper function that calculates the p-value. See below for uses.

Parameters
----------
x, y : ndarray
`x` and `y` have shapes `(n, p)` and `(n, q)`.
stat : float
The sample test statistic.
reps : int, optional
The number of replications used to estimate the null when using the
permutation test. The default is 1000 replications.
workers : int or map-like callable, optional
If `workers` is an int the population is subdivided into `workers`
sections and evaluated in parallel (uses
`multiprocessing.Pool <multiprocessing>`). Supply `-1` to use all cores
available to the Process. Alternatively supply a map-like callable,
such as `multiprocessing.Pool.map` for evaluating the population in
parallel. This evaluation is carried out as `workers(func, iterable)`.
Requires that `func` be pickleable.
random_state : {None, int, `numpy.random.Generator`,
            `numpy.random.RandomState`}, optional

If `seed` is None (or `np.random`), the `numpy.random.RandomState`
singleton is used.
If `seed` is an int, a new ``RandomState`` instance is used,
seeded with `seed`.
If `seed` is already a ``Generator`` or ``RandomState`` instance then
that instance is used.

Returns
-------
pvalue : float
The sample test p-value.
null_dist : list
The approximated null distribution.
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l06161}{6161}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l06225}{multiscale\+\_\+graphcorr()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_ab3be4c10d446d0f0d68e56a854ba225c}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_permutation\_distribution\_t@{\_permutation\_distribution\_t}}
\index{\_permutation\_distribution\_t@{\_permutation\_distribution\_t}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_permutation\_distribution\_t()}{\_permutation\_distribution\_t()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_ab3be4c10d446d0f0d68e56a854ba225c} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+permutation\+\_\+distribution\+\_\+t (\begin{DoxyParamCaption}\item[{}]{data}{, }\item[{}]{permutations}{, }\item[{}]{size\+\_\+a}{, }\item[{}]{equal\+\_\+var}{, }\item[{}]{random\+\_\+state}{ = {\ttfamily None}}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}

\begin{DoxyVerb}Generation permutation distribution of t statistic\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l07541}{7541}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__hypotests_8py_source_l00887}{scipy.\+stats.\+\_\+hypotests.\+\_\+all\+\_\+partitions()}}, and \mbox{\hyperlink{__stats__py_8py_source_l07579}{\+\_\+calc\+\_\+t\+\_\+stat()}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l07598}{\+\_\+permutation\+\_\+ttest()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a76074b3d866305fa36a138f38ce9611b}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_permutation\_ttest@{\_permutation\_ttest}}
\index{\_permutation\_ttest@{\_permutation\_ttest}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_permutation\_ttest()}{\_permutation\_ttest()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a76074b3d866305fa36a138f38ce9611b} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+permutation\+\_\+ttest (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{b}{, }\item[{}]{permutations}{, }\item[{}]{axis}{ = {\ttfamily 0}, }\item[{}]{equal\+\_\+var}{ = {\ttfamily \mbox{\hyperlink{classTrue}{True}}}, }\item[{}]{nan\+\_\+policy}{ = {\ttfamily \textquotesingle{}propagate\textquotesingle{}}, }\item[{}]{random\+\_\+state}{ = {\ttfamily None}, }\item[{}]{alternative}{ = {\ttfamily "{}two-\/sided"{}}}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}

\begin{DoxyVerb}Calculates the T-test for the means of TWO INDEPENDENT samples of scores
using permutation methods.

This test is similar to `stats.ttest_ind`, except it doesn't rely on an
approximate normality assumption since it uses a permutation test.
This function is only called from ttest_ind when permutations is not None.

Parameters
----------
a, b : array_like
    The arrays must be broadcastable, except along the dimension
    corresponding to `axis` (the zeroth, by default).
axis : int, optional
    The axis over which to operate on a and b.
permutations : int, optional
    Number of permutations used to calculate p-value. If greater than or
    equal to the number of distinct permutations, perform an exact test.
equal_var : bool, optional
    If False, an equal variance (Welch's) t-test is conducted.  Otherwise,
    an ordinary t-test is conducted.
random_state : {None, int, `numpy.random.Generator`}, optional
    If `seed` is None the `numpy.random.Generator` singleton is used.
    If `seed` is an int, a new ``Generator`` instance is used,
    seeded with `seed`.
    If `seed` is already a ``Generator`` instance then that instance is
    used.
    Pseudorandom number generator state used for generating random
    permutations.

Returns
-------
statistic : float or array
    The calculated t-statistic.
pvalue : float or array
    The p-value.
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l07596}{7596}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l07579}{\+\_\+calc\+\_\+t\+\_\+stat()}}, and \mbox{\hyperlink{__stats__py_8py_source_l07542}{\+\_\+permutation\+\_\+distribution\+\_\+t()}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l07178}{ttest\+\_\+ind()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a4b21f46bcd2e74e726ebab41f7c799c8}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_quiet\_nanmean@{\_quiet\_nanmean}}
\index{\_quiet\_nanmean@{\_quiet\_nanmean}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_quiet\_nanmean()}{\_quiet\_nanmean()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a4b21f46bcd2e74e726ebab41f7c799c8} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+quiet\+\_\+nanmean (\begin{DoxyParamCaption}\item[{}]{x}{}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}

\begin{DoxyVerb}Compute nanmean for the 1d array x, but quietly return nan if x is all nan.

The return value is a 1d array with length 1, so it can be used
in np.apply_along_axis.
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l02863}{2863}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l03070}{zmap()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a51738f78d33d4973d9bb92624d6a5bef}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_quiet\_nanstd@{\_quiet\_nanstd}}
\index{\_quiet\_nanstd@{\_quiet\_nanstd}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_quiet\_nanstd()}{\_quiet\_nanstd()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a51738f78d33d4973d9bb92624d6a5bef} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+quiet\+\_\+nanstd (\begin{DoxyParamCaption}\item[{}]{x}{, }\item[{}]{ddof}{ = {\ttfamily 0}}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}

\begin{DoxyVerb}Compute nanstd for the 1d array x, but quietly return nan if x is all nan.

The return value is a 1d array with length 1, so it can be used
in np.apply_along_axis.
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l02877}{2877}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l03070}{zmap()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a9d928324e2a00e17b80082984cb5fae2}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_smooth\_mgc\_map@{\_smooth\_mgc\_map}}
\index{\_smooth\_mgc\_map@{\_smooth\_mgc\_map}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_smooth\_mgc\_map()}{\_smooth\_mgc\_map()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a9d928324e2a00e17b80082984cb5fae2} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+smooth\+\_\+mgc\+\_\+map (\begin{DoxyParamCaption}\item[{}]{sig\+\_\+connect}{, }\item[{}]{stat\+\_\+mgc\+\_\+map}{}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}

\begin{DoxyVerb}Finds the smoothed maximal within the significant region R.

If area of R is too small it returns the last local correlation. Otherwise,
returns the maximum within significant_connected_region.

Parameters
----------
sig_connect : ndarray
    A binary matrix with 1's indicating the significant region.
stat_mgc_map : ndarray
    All local correlations within `[-1, 1]`.

Returns
-------
stat : float
    The sample MGC statistic within `[-1, 1]`.
opt_scale: (float, float)
    The estimated optimal scale as an `(x, y)` pair.
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l06603}{6603}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l06509}{\+\_\+mgc\+\_\+stat()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a4f95272db5e6cd266ad36a67e1353b18}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_square\_of\_sums@{\_square\_of\_sums}}
\index{\_square\_of\_sums@{\_square\_of\_sums}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_square\_of\_sums()}{\_square\_of\_sums()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a4f95272db5e6cd266ad36a67e1353b18} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+square\+\_\+of\+\_\+sums (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{axis}{ = {\ttfamily 0}}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}

\begin{DoxyVerb}Sum elements of the input array, and return the square(s) of that sum.

Parameters
----------
a : array_like
    Input array.
axis : int or None, optional
    Axis along which to calculate. Default is 0. If None, compute over
    the whole array `a`.

Returns
-------
square_of_sums : float or ndarray
    The square of the sum over `axis`.

See Also
--------
_sum_of_squares : The sum of squares (the opposite of `square_of_sums`).
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l10094}{10094}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l03985}{f\+\_\+oneway()}}, and \mbox{\hyperlink{__stats__py_8py_source_l09215}{kruskal()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a35baa494b62135807260b2445335c9f9}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_sum\_of\_squares@{\_sum\_of\_squares}}
\index{\_sum\_of\_squares@{\_sum\_of\_squares}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_sum\_of\_squares()}{\_sum\_of\_squares()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a35baa494b62135807260b2445335c9f9} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+sum\+\_\+of\+\_\+squares (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{axis}{ = {\ttfamily 0}}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}

\begin{DoxyVerb}Square each element of the input array, and return the sum(s) of that.

Parameters
----------
a : array_like
    Input array.
axis : int or None, optional
    Axis along which to calculate. Default is 0. If None, compute over
    the whole array `a`.

Returns
-------
sum_of_squares : ndarray
    The sum along the given axis for (a**2).

See Also
--------
_square_of_sums : The square(s) of the sum(s) (the opposite of
    `_sum_of_squares`).
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l10068}{10068}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l03985}{f\+\_\+oneway()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_af2976e18baddb8943eebd15f581f9e77}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_t\_confidence\_interval@{\_t\_confidence\_interval}}
\index{\_t\_confidence\_interval@{\_t\_confidence\_interval}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_t\_confidence\_interval()}{\_t\_confidence\_interval()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_af2976e18baddb8943eebd15f581f9e77} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+t\+\_\+confidence\+\_\+interval (\begin{DoxyParamCaption}\item[{}]{df}{, }\item[{}]{t}{, }\item[{}]{confidence\+\_\+level}{, }\item[{}]{alternative}{}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}



Definition at line \mbox{\hyperlink{__stats__py_8py_source_l06946}{6946}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a436f9fb4bc5898afd29daa2f083de1dc}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_threshold\_mgc\_map@{\_threshold\_mgc\_map}}
\index{\_threshold\_mgc\_map@{\_threshold\_mgc\_map}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_threshold\_mgc\_map()}{\_threshold\_mgc\_map()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a436f9fb4bc5898afd29daa2f083de1dc} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+threshold\+\_\+mgc\+\_\+map (\begin{DoxyParamCaption}\item[{}]{stat\+\_\+mgc\+\_\+map}{, }\item[{}]{samp\+\_\+size}{}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}

\begin{DoxyVerb}Finds a connected region of significance in the MGC-map by thresholding.

Parameters
----------
stat_mgc_map : ndarray
All local correlations within `[-1,1]`.
samp_size : int
The sample size of original data.

Returns
-------
sig_connect : ndarray
A binary matrix with 1's indicating the significant region.
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l06558}{6558}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l06509}{\+\_\+mgc\+\_\+stat()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a068fc003ca0623460250659b9421b133}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_ttest\_finish@{\_ttest\_finish}}
\index{\_ttest\_finish@{\_ttest\_finish}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_ttest\_finish()}{\_ttest\_finish()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a068fc003ca0623460250659b9421b133} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+ttest\+\_\+finish (\begin{DoxyParamCaption}\item[{}]{df}{, }\item[{}]{t}{, }\item[{}]{alternative}{}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}

\begin{DoxyVerb}Common code between all 3 t-test functions.\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l06972}{6972}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l06972}{\+\_\+ttest\+\_\+finish()}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l06972}{\+\_\+ttest\+\_\+finish()}}, \mbox{\hyperlink{__stats__mstats__common_8py_source_l00022}{scipy.\+stats.\+\_\+stats\+\_\+mstats\+\_\+common.\+linregress()}}, \mbox{\hyperlink{__mstats__basic_8py_source_l00555}{scipy.\+stats.\+\_\+mstats\+\_\+basic.\+spearmanr()}}, \mbox{\hyperlink{__stats__py_8py_source_l05170}{spearmanr()}}, \mbox{\hyperlink{__mstats__basic_8py_source_l01319}{scipy.\+stats.\+\_\+mstats\+\_\+basic.\+ttest\+\_\+1samp()}}, \mbox{\hyperlink{__stats__py_8py_source_l06766}{ttest\+\_\+1samp()}}, \mbox{\hyperlink{__mstats__basic_8py_source_l01381}{scipy.\+stats.\+\_\+mstats\+\_\+basic.\+ttest\+\_\+ind()}}, \mbox{\hyperlink{__mstats__basic_8py_source_l01462}{scipy.\+stats.\+\_\+mstats\+\_\+basic.\+ttest\+\_\+rel()}}, and \mbox{\hyperlink{__stats__py_8py_source_l07682}{ttest\+\_\+rel()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a6d068bbda5a330aa2eeb320506a10c1a}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_ttest\_ind\_from\_stats@{\_ttest\_ind\_from\_stats}}
\index{\_ttest\_ind\_from\_stats@{\_ttest\_ind\_from\_stats}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_ttest\_ind\_from\_stats()}{\_ttest\_ind\_from\_stats()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a6d068bbda5a330aa2eeb320506a10c1a} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+ttest\+\_\+ind\+\_\+from\+\_\+stats (\begin{DoxyParamCaption}\item[{}]{mean1}{, }\item[{}]{mean2}{, }\item[{}]{denom}{, }\item[{}]{df}{, }\item[{}]{alternative}{}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}



Definition at line \mbox{\hyperlink{__stats__py_8py_source_l06997}{6997}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_adbe82260a73fb8c647b4f542dd691b1c}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_ttest\_trim\_var\_mean\_len@{\_ttest\_trim\_var\_mean\_len}}
\index{\_ttest\_trim\_var\_mean\_len@{\_ttest\_trim\_var\_mean\_len}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_ttest\_trim\_var\_mean\_len()}{\_ttest\_trim\_var\_mean\_len()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_adbe82260a73fb8c647b4f542dd691b1c} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+ttest\+\_\+trim\+\_\+var\+\_\+mean\+\_\+len (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{trim}{, }\item[{}]{axis}{}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}

\begin{DoxyVerb}Variance, mean, and length of winsorized input along specified axis\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l07482}{7482}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l07506}{\+\_\+calculate\+\_\+winsorized\+\_\+variance()}}, and \mbox{\hyperlink{__stats__py_8py_source_l03890}{trim\+\_\+mean()}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l07178}{ttest\+\_\+ind()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a8b84fb54a29fcfd00875f50715d876dd}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_two\_sample\_transform@{\_two\_sample\_transform}}
\index{\_two\_sample\_transform@{\_two\_sample\_transform}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_two\_sample\_transform()}{\_two\_sample\_transform()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a8b84fb54a29fcfd00875f50715d876dd} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+two\+\_\+sample\+\_\+transform (\begin{DoxyParamCaption}\item[{}]{u}{, }\item[{}]{v}{}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}

\begin{DoxyVerb}Helper function that concatenates x and y for two sample MGC stat.

See above for use.

Parameters
----------
u, v : ndarray
    `u` and `v` have shapes `(n, p)` and `(m, p)`.

Returns
-------
x : ndarray
    Concatenate `u` and `v` along the `axis = 0`. `x` thus has shape
    `(2n, p)`.
y : ndarray
    Label matrix for `x` where 0 refers to samples that comes from `u` and
    1 refers to samples that come from `v`. `y` thus has shape `(2n, 1)`.
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l06654}{6654}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l06225}{multiscale\+\_\+graphcorr()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a705b5678e9a5ee4406cdfc24418527a7}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_unequal\_var\_ttest\_denom@{\_unequal\_var\_ttest\_denom}}
\index{\_unequal\_var\_ttest\_denom@{\_unequal\_var\_ttest\_denom}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_unequal\_var\_ttest\_denom()}{\_unequal\_var\_ttest\_denom()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a705b5678e9a5ee4406cdfc24418527a7} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+unequal\+\_\+var\+\_\+ttest\+\_\+denom (\begin{DoxyParamCaption}\item[{}]{v1}{, }\item[{}]{n1}{, }\item[{}]{v2}{, }\item[{}]{n2}{}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}



Definition at line \mbox{\hyperlink{__stats__py_8py_source_l07007}{7007}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a5946b7785fb379d1de1cc47b0b338faf}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_validate\_distribution@{\_validate\_distribution}}
\index{\_validate\_distribution@{\_validate\_distribution}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_validate\_distribution()}{\_validate\_distribution()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a5946b7785fb379d1de1cc47b0b338faf} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+validate\+\_\+distribution (\begin{DoxyParamCaption}\item[{}]{values}{, }\item[{}]{weights}{}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}

\begin{DoxyVerb}Validate the values and weights from a distribution input of `cdf_distance`
and return them as ndarray objects.

Parameters
----------
values : array_like
    Values observed in the (empirical) distribution.
weights : array_like
    Weight for each value.

Returns
-------
values : ndarray
    Values as ndarray.
weights : ndarray
    Weights as ndarray.
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l09983}{9983}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l09896}{\+\_\+cdf\+\_\+distance()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a9f8cb83384c519dcefbf564456fe81dd}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_var@{\_var}}
\index{\_var@{\_var}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_var()}{\_var()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a9f8cb83384c519dcefbf564456fe81dd} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+var (\begin{DoxyParamCaption}\item[{}]{x}{, }\item[{}]{axis}{ = {\ttfamily 0}, }\item[{}]{ddof}{ = {\ttfamily 0}, }\item[{}]{mean}{ = {\ttfamily None}}\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}



Definition at line \mbox{\hyperlink{__stats__py_8py_source_l01098}{1098}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a25a76cb8c69e3c610d9e9c69fc54e9b9}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!alexandergovern@{alexandergovern}}
\index{alexandergovern@{alexandergovern}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{alexandergovern()}{alexandergovern()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a25a76cb8c69e3c610d9e9c69fc54e9b9} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+alexandergovern (\begin{DoxyParamCaption}\item[{\texorpdfstring{$\ast$}{*}}]{samples}{, }\item[{}]{nan\+\_\+policy}{ = {\ttfamily \textquotesingle{}propagate\textquotesingle{}}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Performs the Alexander Govern test.

The Alexander-Govern approximation tests the equality of k independent
means in the face of heterogeneity of variance. The test is applied to
samples from two or more groups, possibly with differing sizes.

Parameters
----------
sample1, sample2, ... : array_like
    The sample measurements for each group.  There must be at least
    two samples.
nan_policy : {'propagate', 'raise', 'omit'}, optional
    Defines how to handle when input contains nan.
    The following options are available (default is 'propagate'):

    * 'propagate': returns nan
    * 'raise': throws an error
    * 'omit': performs the calculations ignoring nan values

Returns
-------
res : AlexanderGovernResult
    An object with attributes:

    statistic : float
        The computed A statistic of the test.
    pvalue : float
        The associated p-value from the chi-squared distribution.

Warns
-----
`~scipy.stats.ConstantInputWarning`
    Raised if an input is a constant array.  The statistic is not defined
    in this case, so ``np.nan`` is returned.

See Also
--------
f_oneway : one-way ANOVA

Notes
-----
The use of this test relies on several assumptions.

1. The samples are independent.
2. Each sample is from a normally distributed population.
3. Unlike `f_oneway`, this test does not assume on homoscedasticity,
   instead relaxing the assumption of equal variances.

Input samples must be finite, one dimensional, and with size greater than
one.

References
----------
.. [1] Alexander, Ralph A., and Diane M. Govern. "A New and Simpler
       Approximation for ANOVA under Variance Heterogeneity." Journal
       of Educational Statistics, vol. 19, no. 2, 1994, pp. 91-101.
       JSTOR, www.jstor.org/stable/1165140. Accessed 12 Sept. 2020.

Examples
--------
>>> from scipy.stats import alexandergovern

Here are some data on annual percentage rate of interest charged on
new car loans at nine of the largest banks in four American cities
taken from the National Institute of Standards and Technology's
ANOVA dataset.

We use `alexandergovern` to test the null hypothesis that all cities
have the same mean APR against the alternative that the cities do not
all have the same mean APR. We decide that a significance level of 5%
is required to reject the null hypothesis in favor of the alternative.

>>> atlanta = [13.75, 13.75, 13.5, 13.5, 13.0, 13.0, 13.0, 12.75, 12.5]
>>> chicago = [14.25, 13.0, 12.75, 12.5, 12.5, 12.4, 12.3, 11.9, 11.9]
>>> houston = [14.0, 14.0, 13.51, 13.5, 13.5, 13.25, 13.0, 12.5, 12.5]
>>> memphis = [15.0, 14.0, 13.75, 13.59, 13.25, 12.97, 12.5, 12.25,
...           11.89]
>>> alexandergovern(atlanta, chicago, houston, memphis)
AlexanderGovernResult(statistic=4.65087071883494,
                      pvalue=0.19922132490385214)

The p-value is 0.1992, indicating a nearly 20% chance of observing
such an extreme value of the test statistic under the null hypothesis.
This exceeds 5%, so we do not reject the null hypothesis in favor of
the alternative.
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l04220}{4220}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a61540456aa6cd16b909618e175d798f9}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!brunnermunzel@{brunnermunzel}}
\index{brunnermunzel@{brunnermunzel}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{brunnermunzel()}{brunnermunzel()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a61540456aa6cd16b909618e175d798f9} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+brunnermunzel (\begin{DoxyParamCaption}\item[{}]{x}{, }\item[{}]{y}{, }\item[{}]{alternative}{ = {\ttfamily "{}two-\/sided"{}}, }\item[{}]{distribution}{ = {\ttfamily "{}t"{}}, }\item[{}]{nan\+\_\+policy}{ = {\ttfamily \textquotesingle{}propagate\textquotesingle{}}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Compute the Brunner-Munzel test on samples x and y.

The Brunner-Munzel test is a nonparametric test of the null hypothesis that
when values are taken one by one from each group, the probabilities of
getting large values in both groups are equal.
Unlike the Wilcoxon-Mann-Whitney's U test, this does not require the
assumption of equivariance of two groups. Note that this does not assume
the distributions are same. This test works on two independent samples,
which may have different sizes.

Parameters
----------
x, y : array_like
    Array of samples, should be one-dimensional.
alternative : {'two-sided', 'less', 'greater'}, optional
    Defines the alternative hypothesis.
    The following options are available (default is 'two-sided'):

      * 'two-sided'
      * 'less': one-sided
      * 'greater': one-sided
distribution : {'t', 'normal'}, optional
    Defines how to get the p-value.
    The following options are available (default is 't'):

      * 't': get the p-value by t-distribution
      * 'normal': get the p-value by standard normal distribution.
nan_policy : {'propagate', 'raise', 'omit'}, optional
    Defines how to handle when input contains nan.
    The following options are available (default is 'propagate'):

      * 'propagate': returns nan
      * 'raise': throws an error
      * 'omit': performs the calculations ignoring nan values

Returns
-------
statistic : float
    The Brunner-Munzer W statistic.
pvalue : float
    p-value assuming an t distribution. One-sided or
    two-sided, depending on the choice of `alternative` and `distribution`.

See Also
--------
mannwhitneyu : Mann-Whitney rank test on two samples.

Notes
-----
Brunner and Munzel recommended to estimate the p-value by t-distribution
when the size of data is 50 or less. If the size is lower than 10, it would
be better to use permuted Brunner Munzel test (see [2]_).

References
----------
.. [1] Brunner, E. and Munzel, U. "The nonparametric Benhrens-Fisher
       problem: Asymptotic theory and a small-sample approximation".
       Biometrical Journal. Vol. 42(2000): 17-25.
.. [2] Neubert, K. and Brunner, E. "A studentized permutation test for the
       non-parametric Behrens-Fisher problem". Computational Statistics and
       Data Analysis. Vol. 51(2007): 5192-5204.

Examples
--------
>>> from scipy import stats
>>> x1 = [1,2,1,1,1,1,1,1,1,1,2,4,1,1]
>>> x2 = [3,3,4,3,1,2,3,1,1,5,4]
>>> w, p_value = stats.brunnermunzel(x1, x2)
>>> w
3.1374674823029505
>>> p_value
0.0057862086661515377
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l09436}{9436}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l10123}{rankdata()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a5794f15c32e4b9eaee99f403e23be4c5}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!chisquare@{chisquare}}
\index{chisquare@{chisquare}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{chisquare()}{chisquare()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a5794f15c32e4b9eaee99f403e23be4c5} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+chisquare (\begin{DoxyParamCaption}\item[{}]{f\+\_\+obs}{, }\item[{}]{f\+\_\+exp}{ = {\ttfamily None}, }\item[{}]{ddof}{ = {\ttfamily 0}, }\item[{}]{axis}{ = {\ttfamily 0}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Calculate a one-way chi-square test.

The chi-square test tests the null hypothesis that the categorical data
has the given frequencies.

Parameters
----------
f_obs : array_like
    Observed frequencies in each category.
f_exp : array_like, optional
    Expected frequencies in each category.  By default the categories are
    assumed to be equally likely.
ddof : int, optional
    "Delta degrees of freedom": adjustment to the degrees of freedom
    for the p-value.  The p-value is computed using a chi-squared
    distribution with ``k - 1 - ddof`` degrees of freedom, where `k`
    is the number of observed frequencies.  The default value of `ddof`
    is 0.
axis : int or None, optional
    The axis of the broadcast result of `f_obs` and `f_exp` along which to
    apply the test.  If axis is None, all values in `f_obs` are treated
    as a single data set.  Default is 0.

Returns
-------
res: Power_divergenceResult
    An object containing attributes:

    chisq : float or ndarray
        The chi-squared test statistic.  The value is a float if `axis` is
        None or `f_obs` and `f_exp` are 1-D.
    pvalue : float or ndarray
        The p-value of the test.  The value is a float if `ddof` and the
        return value `chisq` are scalars.

See Also
--------
scipy.stats.power_divergence
scipy.stats.fisher_exact : Fisher exact test on a 2x2 contingency table.
scipy.stats.barnard_exact : An unconditional exact test. An alternative
    to chi-squared test for small sample sizes.

Notes
-----
This test is invalid when the observed or expected frequencies in each
category are too small.  A typical rule is that all of the observed
and expected frequencies should be at least 5. According to [3]_, the
total number of samples is recommended to be greater than 13,
otherwise exact tests (such as Barnard's Exact test) should be used
because they do not overreject.

Also, the sum of the observed and expected frequencies must be the same
for the test to be valid; `chisquare` raises an error if the sums do not
agree within a relative tolerance of ``1e-8``.

The default degrees of freedom, k-1, are for the case when no parameters
of the distribution are estimated. If p parameters are estimated by
efficient maximum likelihood then the correct degrees of freedom are
k-1-p. If the parameters are estimated in a different way, then the
dof can be between k-1-p and k-1. However, it is also possible that
the asymptotic distribution is not chi-square, in which case this test
is not appropriate.

References
----------
.. [1] Lowry, Richard.  "Concepts and Applications of Inferential
       Statistics". Chapter 8.
       https://web.archive.org/web/20171022032306/http://vassarstats.net:80/textbook/ch8pt1.html
.. [2] "Chi-squared test", https://en.wikipedia.org/wiki/Chi-squared_test
.. [3] Pearson, Karl. "On the criterion that a given system of deviations from the probable
       in the case of a correlated system of variables is such that it can be reasonably
       supposed to have arisen from random sampling", Philosophical Magazine. Series 5. 50
       (1900), pp. 157-175.
.. [4] Mannan, R. William and E. Charles. Meslow. "Bird populations and
       vegetation characteristics in managed and old-growth forests,
       northeastern Oregon." Journal of Wildlife Management
       48, 1219-1238, :doi:`10.2307/3801783`, 1984.

Examples
--------
In [4]_, bird foraging behavior was investigated in an old-growth forest
of Oregon.
In the forest, 44% of the canopy volume was Douglas fir,
24% was ponderosa pine, 29% was grand fir, and 3% was western larch.
The authors observed the behavior of several species of birds, one of
which was the red-breasted nuthatch. They made 189 observations of this
species foraging, recording 43 ("23%") of observations in Douglas fir,
52 ("28%") in ponderosa pine, 54 ("29%") in grand fir, and 40 ("21%") in
western larch.

Using a chi-square test, we can test the null hypothesis that the
proportions of foraging events are equal to the proportions of canopy
volume. The authors of the paper considered a p-value less than 1% to be
significant.

Using the above proportions of canopy volume and observed events, we can
infer expected frequencies.

>>> import numpy as np
>>> f_exp = np.array([44, 24, 29, 3]) / 100 * 189

The observed frequencies of foraging were:

>>> f_obs = np.array([43, 52, 54, 40])

We can now compare the observed frequencies with the expected frequencies.

>>> from scipy.stats import chisquare
>>> chisquare(f_obs=f_obs, f_exp=f_exp)
Power_divergenceResult(statistic=228.23515947653874, pvalue=3.3295585338846486e-49)

The p-value is well below the chosen significance level. Hence, the
authors considered the difference to be significant and concluded
that the relative proportions of foraging events were not the same
as the relative proportions of tree canopy volume.

Following are other generic examples to demonstrate how the other
parameters can be used.

When just `f_obs` is given, it is assumed that the expected frequencies
are uniform and given by the mean of the observed frequencies.

>>> chisquare([16, 18, 16, 14, 12, 12])
Power_divergenceResult(statistic=2.0, pvalue=0.84914503608460956)

With `f_exp` the expected frequencies can be given.

>>> chisquare([16, 18, 16, 14, 12, 12], f_exp=[16, 16, 16, 16, 16, 8])
Power_divergenceResult(statistic=3.5, pvalue=0.62338762774958223)

When `f_obs` is 2-D, by default the test is applied to each column.

>>> obs = np.array([[16, 18, 16, 14, 12, 12], [32, 24, 16, 28, 20, 24]]).T
>>> obs.shape
(6, 2)
>>> chisquare(obs)
Power_divergenceResult(statistic=array([2.        , 6.66666667]), pvalue=array([0.84914504, 0.24663415]))

By setting ``axis=None``, the test is applied to all data in the array,
which is equivalent to applying the test to the flattened array.

>>> chisquare(obs, axis=None)
Power_divergenceResult(statistic=23.31034482758621, pvalue=0.015975692534127565)
>>> chisquare(obs.ravel())
Power_divergenceResult(statistic=23.310344827586206, pvalue=0.01597569253412758)

`ddof` is the change to make to the default degrees of freedom.

>>> chisquare([16, 18, 16, 14, 12, 12], ddof=1)
Power_divergenceResult(statistic=2.0, pvalue=0.7357588823428847)

The calculation of the p-values is done by broadcasting the
chi-squared statistic with `ddof`.

>>> chisquare([16, 18, 16, 14, 12, 12], ddof=[0,1,2])
Power_divergenceResult(statistic=2.0, pvalue=array([0.84914504, 0.73575888, 0.5724067 ]))

`f_obs` and `f_exp` are also broadcast.  In the following, `f_obs` has
shape (6,) and `f_exp` has shape (2, 6), so the result of broadcasting
`f_obs` and `f_exp` has shape (2, 6).  To compute the desired chi-squared
statistics, we use ``axis=1``:

>>> chisquare([16, 18, 16, 14, 12, 12],
...           f_exp=[[16, 16, 16, 16, 16, 8], [8, 20, 20, 16, 12, 12]],
...           axis=1)
Power_divergenceResult(statistic=array([3.5 , 9.25]), pvalue=array([0.62338763, 0.09949846]))
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l08085}{8085}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l07857}{power\+\_\+divergence()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a0b6200c846acee0166b2ecb13abdcba1}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!combine\_pvalues@{combine\_pvalues}}
\index{combine\_pvalues@{combine\_pvalues}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{combine\_pvalues()}{combine\_pvalues()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a0b6200c846acee0166b2ecb13abdcba1} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+combine\+\_\+pvalues (\begin{DoxyParamCaption}\item[{}]{pvalues}{, }\item[{}]{method}{ = {\ttfamily \textquotesingle{}fisher\textquotesingle{}}, }\item[{}]{weights}{ = {\ttfamily None}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Combine p-values from independent tests that bear upon the same hypothesis.

These methods are intended only for combining p-values from hypothesis
tests based upon continuous distributions.

Each method assumes that under the null hypothesis, the p-values are
sampled independently and uniformly from the interval [0, 1]. A test
statistic (different for each method) is computed and a combined
p-value is calculated based upon the distribution of this test statistic
under the null hypothesis.

Parameters
----------
pvalues : array_like, 1-D
    Array of p-values assumed to come from independent tests based on
    continuous distributions.
method : {'fisher', 'pearson', 'tippett', 'stouffer', 'mudholkar_george'}

    Name of method to use to combine p-values.

    The available methods are (see Notes for details):

    * 'fisher': Fisher's method (Fisher's combined probability test)
    * 'pearson': Pearson's method
    * 'mudholkar_george': Mudholkar's and George's method
    * 'tippett': Tippett's method
    * 'stouffer': Stouffer's Z-score method
weights : array_like, 1-D, optional
    Optional array of weights used only for Stouffer's Z-score method.

Returns
-------
res : SignificanceResult
    An object containing attributes:

    statistic : float
        The statistic calculated by the specified method.
    pvalue : float
        The combined p-value.

Notes
-----
If this function is applied to tests with a discrete statistics such as
any rank test or contingency-table test, it will yield systematically
wrong results, e.g. Fisher's method will systematically overestimate the
p-value [1]_. This problem becomes less severe for large sample sizes
when the discrete distributions become approximately continuous.

The differences between the methods can be best illustrated by their
statistics and what aspects of a combination of p-values they emphasise
when considering significance [2]_. For example, methods emphasising large
p-values are more sensitive to strong false and true negatives; conversely
methods focussing on small p-values are sensitive to positives.

* The statistics of Fisher's method (also known as Fisher's combined
  probability test) [3]_ is :math:`-2\\sum_i \\log(p_i)`, which is
  equivalent (as a test statistics) to the product of individual p-values:
  :math:`\\prod_i p_i`. Under the null hypothesis, this statistics follows
  a :math:`\\chi^2` distribution. This method emphasises small p-values.
* Pearson's method uses :math:`-2\\sum_i\\log(1-p_i)`, which is equivalent
  to :math:`\\prod_i \\frac{1}{1-p_i}` [2]_.
  It thus emphasises large p-values.
* Mudholkar and George compromise between Fisher's and Pearson's method by
  averaging their statistics [4]_. Their method emphasises extreme
  p-values, both close to 1 and 0.
* Stouffer's method [5]_ uses Z-scores and the statistic:
  :math:`\\sum_i \\Phi^{-1} (p_i)`, where :math:`\\Phi` is the CDF of the
  standard normal distribution. The advantage of this method is that it is
  straightforward to introduce weights, which can make Stouffer's method
  more powerful than Fisher's method when the p-values are from studies
  of different size [6]_ [7]_.
* Tippett's method uses the smallest p-value as a statistic.
  (Mind that this minimum is not the combined p-value.)

Fisher's method may be extended to combine p-values from dependent tests
[8]_. Extensions such as Brown's method and Kost's method are not currently
implemented.

.. versionadded:: 0.15.0

References
----------
.. [1] Kincaid, W. M., "The Combination of Tests Based on Discrete
       Distributions." Journal of the American Statistical Association 57,
       no. 297 (1962), 10-19.
.. [2] Heard, N. and Rubin-Delanchey, P. "Choosing between methods of
       combining p-values."  Biometrika 105.1 (2018): 239-246.
.. [3] https://en.wikipedia.org/wiki/Fisher%27s_method
.. [4] George, E. O., and G. S. Mudholkar. "On the convolution of logistic
       random variables." Metrika 30.1 (1983): 1-13.
.. [5] https://en.wikipedia.org/wiki/Fisher%27s_method#Relation_to_Stouffer.27s_Z-score_method
.. [6] Whitlock, M. C. "Combining probability from independent tests: the
       weighted Z-method is superior to Fisher's approach." Journal of
       Evolutionary Biology 18, no. 5 (2005): 1368-1373.
.. [7] Zaykin, Dmitri V. "Optimally weighted Z-test is a powerful method
       for combining probabilities in meta-analysis." Journal of
       Evolutionary Biology 24, no. 8 (2011): 1836-1841.
.. [8] https://en.wikipedia.org/wiki/Extensions_of_Fisher%27s_method
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l09583}{9583}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_ab7cc05458723f566fbbd8bf5c92ac145}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!cumfreq@{cumfreq}}
\index{cumfreq@{cumfreq}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{cumfreq()}{cumfreq()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_ab7cc05458723f566fbbd8bf5c92ac145} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+cumfreq (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{numbins}{ = {\ttfamily 10}, }\item[{}]{defaultreallimits}{ = {\ttfamily None}, }\item[{}]{weights}{ = {\ttfamily None}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Return a cumulative frequency histogram, using the histogram function.

A cumulative histogram is a mapping that counts the cumulative number of
observations in all of the bins up to the specified bin.

Parameters
----------
a : array_like
    Input array.
numbins : int, optional
    The number of bins to use for the histogram. Default is 10.
defaultreallimits : tuple (lower, upper), optional
    The lower and upper values for the range of the histogram.
    If no value is given, a range slightly larger than the range of the
    values in `a` is used. Specifically ``(a.min() - s, a.max() + s)``,
    where ``s = (1/2)(a.max() - a.min()) / (numbins - 1)``.
weights : array_like, optional
    The weights for each value in `a`. Default is None, which gives each
    value a weight of 1.0

Returns
-------
cumcount : ndarray
    Binned values of cumulative frequency.
lowerlimit : float
    Lower real limit
binsize : float
    Width of each bin.
extrapoints : int
    Extra points.

Examples
--------
>>> import numpy as np
>>> import matplotlib.pyplot as plt
>>> from scipy import stats
>>> rng = np.random.default_rng()
>>> x = [1, 4, 2, 1, 3, 1]
>>> res = stats.cumfreq(x, numbins=4, defaultreallimits=(1.5, 5))
>>> res.cumcount
array([ 1.,  2.,  3.,  3.])
>>> res.extrapoints
3

Create a normal distribution with 1000 random values

>>> samples = stats.norm.rvs(size=1000, random_state=rng)

Calculate cumulative frequencies

>>> res = stats.cumfreq(samples, numbins=25)

Calculate space of values for x

>>> x = res.lowerlimit + np.linspace(0, res.binsize*res.cumcount.size,
...                                  res.cumcount.size)

Plot histogram and cumulative histogram

>>> fig = plt.figure(figsize=(10, 4))
>>> ax1 = fig.add_subplot(1, 2, 1)
>>> ax2 = fig.add_subplot(1, 2, 2)
>>> ax1.hist(samples, bins=25)
>>> ax1.set_title('Histogram')
>>> ax2.bar(x, res.cumcount, width=res.binsize)
>>> ax2.set_title('Cumulative histogram')
>>> ax2.set_xlim([x.min(), x.max()])

>>> plt.show()
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l02542}{2542}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l02460}{\+\_\+histogram()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a9074c49e4ce1d99373dae0a3c48c9d5e}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!describe@{describe}}
\index{describe@{describe}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{describe()}{describe()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a9074c49e4ce1d99373dae0a3c48c9d5e} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+describe (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{axis}{ = {\ttfamily 0}, }\item[{}]{ddof}{ = {\ttfamily 1}, }\item[{}]{bias}{ = {\ttfamily \mbox{\hyperlink{classTrue}{True}}}, }\item[{}]{nan\+\_\+policy}{ = {\ttfamily \textquotesingle{}propagate\textquotesingle{}}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Compute several descriptive statistics of the passed array.

Parameters
----------
a : array_like
    Input data.
axis : int or None, optional
    Axis along which statistics are calculated. Default is 0.
    If None, compute over the whole array `a`.
ddof : int, optional
    Delta degrees of freedom (only for variance).  Default is 1.
bias : bool, optional
    If False, then the skewness and kurtosis calculations are corrected
    for statistical bias.
nan_policy : {'propagate', 'raise', 'omit'}, optional
    Defines how to handle when input contains nan.
    The following options are available (default is 'propagate'):

    * 'propagate': returns nan
    * 'raise': throws an error
    * 'omit': performs the calculations ignoring nan values

Returns
-------
nobs : int or ndarray of ints
    Number of observations (length of data along `axis`).
    When 'omit' is chosen as nan_policy, the length along each axis
    slice is counted separately.
minmax: tuple of ndarrays or floats
    Minimum and maximum value of `a` along the given axis.
mean : ndarray or float
    Arithmetic mean of `a` along the given axis.
variance : ndarray or float
    Unbiased variance of `a` along the given axis; denominator is number
    of observations minus one.
skewness : ndarray or float
    Skewness of `a` along the given axis, based on moment calculations
    with denominator equal to the number of observations, i.e. no degrees
    of freedom correction.
kurtosis : ndarray or float
    Kurtosis (Fisher) of `a` along the given axis.  The kurtosis is
    normalized so that it is zero for the normal distribution.  No
    degrees of freedom are used.

See Also
--------
skew, kurtosis

Examples
--------
>>> import numpy as np
>>> from scipy import stats
>>> a = np.arange(10)
>>> stats.describe(a)
DescribeResult(nobs=10, minmax=(0, 9), mean=4.5,
               variance=9.166666666666666, skewness=0.0,
               kurtosis=-1.2242424242424244)
>>> b = [[1, 2], [3, 4]]
>>> stats.describe(b)
DescribeResult(nobs=2, minmax=(array([1, 2]), array([3, 4])),
               mean=array([2., 3.]), variance=array([2., 2.]),
               skewness=array([0., 0.]), kurtosis=array([-2., -2.]))
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l01326}{1326}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l01213}{kurtosis()}}, and \mbox{\hyperlink{__stats__py_8py_source_l01110}{skew()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a348efe926c611bfc19391b531685b053}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!energy\_distance@{energy\_distance}}
\index{energy\_distance@{energy\_distance}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{energy\_distance()}{energy\_distance()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a348efe926c611bfc19391b531685b053} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+energy\+\_\+distance (\begin{DoxyParamCaption}\item[{}]{u\+\_\+values}{, }\item[{}]{v\+\_\+values}{, }\item[{}]{u\+\_\+weights}{ = {\ttfamily None}, }\item[{}]{v\+\_\+weights}{ = {\ttfamily None}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Compute the energy distance between two 1D distributions.

    .. versionadded:: 1.0.0

    Parameters
    ----------
    u_values, v_values : array_like
        Values observed in the (empirical) distribution.
    u_weights, v_weights : array_like, optional
        Weight for each value. If unspecified, each value is assigned the same
        weight.
        `u_weights` (resp. `v_weights`) must have the same length as
        `u_values` (resp. `v_values`). If the weight sum differs from 1, it
        must still be positive and finite so that the weights can be normalized
        to sum to 1.

    Returns
    -------
    distance : float
        The computed distance between the distributions.

    Notes
    -----
    The energy distance between two distributions :math:`u` and :math:`v`, whose
    respective CDFs are :math:`U` and :math:`V`, equals to:

    .. math::

        D(u, v) = \left( 2\mathbb E|X - Y| - \mathbb E|X - X'| -
        \mathbb E|Y - Y'| \right)^{1/2}

    where :math:`X` and :math:`X'` (resp. :math:`Y` and :math:`Y'`) are
    independent random variables whose probability distribution is :math:`u`
    (resp. :math:`v`).

    Sometimes the square of this quantity is referred to as the "energy
    distance" (e.g. in [2]_, [4]_), but as noted in [1]_ and [3]_, only the
    definition above satisfies the axioms of a distance function (metric).

    As shown in [2]_, for one-dimensional real-valued variables, the energy
    distance is linked to the non-distribution-free version of the Cramr-von
    Mises distance:

    .. math::

        D(u, v) = \sqrt{2} l_2(u, v) = \left( 2 \int_{-\infty}^{+\infty} (U-V)^2
        \right)^{1/2}

    Note that the common Cramr-von Mises criterion uses the distribution-free
    version of the distance. See [2]_ (section 2), for more details about both
    versions of the distance.

    The input distributions can be empirical, therefore coming from samples
    whose values are effectively inputs of the function, or they can be seen as
    generalized functions, in which case they are weighted sums of Dirac delta
    functions located at the specified values.

    References
    ----------
    .. [1] Rizzo, Szekely "Energy distance." Wiley Interdisciplinary Reviews:
           Computational Statistics, 8(1):27-38 (2015).
    .. [2] Szekely "E-statistics: The energy of statistical samples." Bowling
           Green State University, Department of Mathematics and Statistics,
           Technical Report 02-16 (2002).
    .. [3] "Energy distance", https://en.wikipedia.org/wiki/Energy_distance
    .. [4] Bellemare, Danihelka, Dabney, Mohamed, Lakshminarayanan, Hoyer,
           Munos "The Cramer Distance as a Solution to Biased Wasserstein
           Gradients" (2017). :arXiv:`1705.10743`.

    Examples
    --------
    >>> from scipy.stats import energy_distance
    >>> energy_distance([0], [2])
    2.0000000000000004
    >>> energy_distance([0, 8], [0, 8], [3, 1], [2, 2])
    1.0000000000000002
    >>> energy_distance([0.7, 7.4, 2.4, 6.8], [1.4, 8. ],
    ...                 [2.1, 4.2, 7.4, 8. ], [7.6, 8.8])
    0.88003340976158217
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l09810}{9810}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l09896}{\+\_\+cdf\+\_\+distance()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a5ab039579d581807121a4bd4b1cd14f4}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!expectile@{expectile}}
\index{expectile@{expectile}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{expectile()}{expectile()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a5ab039579d581807121a4bd4b1cd14f4} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+expectile (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{alpha}{ = {\ttfamily 0.5}, }\item[{\texorpdfstring{$\ast$}{*}}]{}{, }\item[{}]{weights}{ = {\ttfamily None}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Compute the expectile at the specified level.

Expectiles are a generalization of the expectation in the same way as
quantiles are a generalization of the median. The expectile at level
`alpha = 0.5` is the mean (average). See Notes for more details.

Parameters
----------
a : array_like
Array containing numbers whose expectile is desired.
alpha : float, default: 0.5
The level of the expectile; e.g., `alpha=0.5` gives the mean.
weights : array_like, optional
An array of weights associated with the values in `a`.
The `weights` must be broadcastable to the same shape as `a`.
Default is None, which gives each value a weight of 1.0.
An integer valued weight element acts like repeating the corresponding
observation in `a` that many times. See Notes for more details.

Returns
-------
expectile : ndarray
The empirical expectile at level `alpha`.

See Also
--------
numpy.mean : Arithmetic average
numpy.quantile : Quantile

Notes
-----
In general, the expectile at level :math:`\alpha` of a random variable
:math:`X` with cumulative distribution function (CDF) :math:`F` is given
by the unique solution :math:`t` of:

.. math::

\alpha E((X - t)_+) = (1 - \alpha) E((t - X)_+) \,.

Here, :math:`(x)_+ = \max(0, x)` is the positive part of :math:`x`.
This equation can be equivalently written as:

.. math::

\alpha \int_t^\infty (x - t)\mathrm{d}F(x)
= (1 - \alpha) \int_{-\infty}^t (t - x)\mathrm{d}F(x) \,.

The empirical expectile at level :math:`\alpha` (`alpha`) of a sample
:math:`a_i` (the array `a`) is defined by plugging in the empirical CDF of
`a`. Given sample or case weights :math:`w` (the array `weights`), it
reads :math:`F_a(x) = \frac{1}{\sum_i w_i} \sum_i w_i 1_{a_i \leq x}`
with indicator function :math:`1_{A}`. This leads to the definition of the
empirical expectile at level `alpha` as the unique solution :math:`t` of:

.. math::

\alpha \sum_{i=1}^n w_i (a_i - t)_+ =
(1 - \alpha) \sum_{i=1}^n w_i (t - a_i)_+ \,.

For :math:`\alpha=0.5`, this simplifies to the weighted average.
Furthermore, the larger :math:`\alpha`, the larger the value of the
expectile.

As a final remark, the expectile at level :math:`\alpha` can also be
written as a minimization problem. One often used choice is

.. math::

\operatorname{argmin}_t
E(\lvert 1_{t\geq X} - \alpha\rvert(t - X)^2) \,.

References
----------
.. [1] W. K. Newey and J. L. Powell (1987), "Asymmetric Least Squares
Estimation and Testing," Econometrica, 55, 819-847.
.. [2] T. Gneiting (2009). "Making and Evaluating Point Forecasts,"
Journal of the American Statistical Association, 106, 746 - 762.
:doi:`10.48550/arXiv.0912.0902`

Examples
--------
>>> import numpy as np
>>> from scipy.stats import expectile
>>> a = [1, 4, 2, -1]
>>> expectile(a, alpha=0.5) == np.mean(a)
True
>>> expectile(a, alpha=0.2)
0.42857142857142855
>>> expectile(a, alpha=0.8)
2.5714285714285716
>>> weights = [1, 3, 1, 1]
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l10270}{10270}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a7771f3b2db7c5785f3c8395389ec7b76}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!f\_oneway@{f\_oneway}}
\index{f\_oneway@{f\_oneway}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{f\_oneway()}{f\_oneway()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a7771f3b2db7c5785f3c8395389ec7b76} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+f\+\_\+oneway (\begin{DoxyParamCaption}\item[{\texorpdfstring{$\ast$}{*}}]{samples}{, }\item[{}]{axis}{ = {\ttfamily 0}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Perform one-way ANOVA.

The one-way ANOVA tests the null hypothesis that two or more groups have
the same population mean.  The test is applied to samples from two or
more groups, possibly with differing sizes.

Parameters
----------
sample1, sample2, ... : array_like
    The sample measurements for each group.  There must be at least
    two arguments.  If the arrays are multidimensional, then all the
    dimensions of the array must be the same except for `axis`.
axis : int, optional
    Axis of the input arrays along which the test is applied.
    Default is 0.

Returns
-------
statistic : float
    The computed F statistic of the test.
pvalue : float
    The associated p-value from the F distribution.

Warns
-----
`~scipy.stats.ConstantInputWarning`
    Raised if all values within each of the input arrays are identical.
    In this case the F statistic is either infinite or isn't defined,
    so ``np.inf`` or ``np.nan`` is returned.

`~scipy.stats.DegenerateDataWarning`
    Raised if the length of any input array is 0, or if all the input
    arrays have length 1.  ``np.nan`` is returned for the F statistic
    and the p-value in these cases.

Notes
-----
The ANOVA test has important assumptions that must be satisfied in order
for the associated p-value to be valid.

1. The samples are independent.
2. Each sample is from a normally distributed population.
3. The population standard deviations of the groups are all equal.  This
   property is known as homoscedasticity.

If these assumptions are not true for a given set of data, it may still
be possible to use the Kruskal-Wallis H-test (`scipy.stats.kruskal`) or
the Alexander-Govern test (`scipy.stats.alexandergovern`) although with
some loss of power.

The length of each group must be at least one, and there must be at
least one group with length greater than one.  If these conditions
are not satisfied, a warning is generated and (``np.nan``, ``np.nan``)
is returned.

If all values in each group are identical, and there exist at least two
groups with different values, the function generates a warning and
returns (``np.inf``, 0).

If all values in all groups are the same, function generates a warning
and returns (``np.nan``, ``np.nan``).

The algorithm is from Heiman [2]_, pp.394-7.

References
----------
.. [1] R. Lowry, "Concepts and Applications of Inferential Statistics",
       Chapter 14, 2014, http://vassarstats.net/textbook/

.. [2] G.W. Heiman, "Understanding research methods and statistics: An
       integrated introduction for psychology", Houghton, Mifflin and
       Company, 2001.

.. [3] G.H. McDonald, "Handbook of Biological Statistics", One-way ANOVA.
       http://www.biostathandbook.com/onewayanova.html

Examples
--------
>>> import numpy as np
>>> from scipy.stats import f_oneway

Here are some data [3]_ on a shell measurement (the length of the anterior
adductor muscle scar, standardized by dividing by length) in the mussel
Mytilus trossulus from five locations: Tillamook, Oregon; Newport, Oregon;
Petersburg, Alaska; Magadan, Russia; and Tvarminne, Finland, taken from a
much larger data set used in McDonald et al. (1991).

>>> tillamook = [0.0571, 0.0813, 0.0831, 0.0976, 0.0817, 0.0859, 0.0735,
...              0.0659, 0.0923, 0.0836]
>>> newport = [0.0873, 0.0662, 0.0672, 0.0819, 0.0749, 0.0649, 0.0835,
...            0.0725]
>>> petersburg = [0.0974, 0.1352, 0.0817, 0.1016, 0.0968, 0.1064, 0.105]
>>> magadan = [0.1033, 0.0915, 0.0781, 0.0685, 0.0677, 0.0697, 0.0764,
...            0.0689]
>>> tvarminne = [0.0703, 0.1026, 0.0956, 0.0973, 0.1039, 0.1045]
>>> f_oneway(tillamook, newport, petersburg, magadan, tvarminne)
F_onewayResult(statistic=7.121019471642447, pvalue=0.0002812242314534544)

`f_oneway` accepts multidimensional input arrays.  When the inputs
are multidimensional and `axis` is not given, the test is performed
along the first axis of the input arrays.  For the following data, the
test is performed three times, once for each column.

>>> a = np.array([[9.87, 9.03, 6.81],
...               [7.18, 8.35, 7.00],
...               [8.39, 7.58, 7.68],
...               [7.45, 6.33, 9.35],
...               [6.41, 7.10, 9.33],
...               [8.00, 8.24, 8.44]])
>>> b = np.array([[6.35, 7.30, 7.16],
...               [6.65, 6.68, 7.63],
...               [5.72, 7.73, 6.72],
...               [7.01, 9.19, 7.41],
...               [7.75, 7.87, 8.30],
...               [6.90, 7.97, 6.97]])
>>> c = np.array([[3.31, 8.77, 1.01],
...               [8.25, 3.24, 3.62],
...               [6.32, 8.81, 5.19],
...               [7.48, 8.83, 8.91],
...               [8.59, 6.01, 6.07],
...               [3.07, 9.72, 7.48]])
>>> F, p = f_oneway(a, b, c)
>>> F
array([1.75676344, 0.03701228, 3.76439349])
>>> p
array([0.20630784, 0.96375203, 0.04733157])
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l03985}{3985}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l03963}{\+\_\+create\+\_\+f\+\_\+oneway\+\_\+nan\+\_\+result()}}, \mbox{\hyperlink{__stats__py_8py_source_l03980}{\+\_\+first()}}, \mbox{\hyperlink{__stats__py_8py_source_l10094}{\+\_\+square\+\_\+of\+\_\+sums()}}, and \mbox{\hyperlink{__stats__py_8py_source_l10068}{\+\_\+sum\+\_\+of\+\_\+squares()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a6394edc619f56ba4b055ea6050b269e6}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!find\_repeats@{find\_repeats}}
\index{find\_repeats@{find\_repeats}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{find\_repeats()}{find\_repeats()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a6394edc619f56ba4b055ea6050b269e6} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+find\+\_\+repeats (\begin{DoxyParamCaption}\item[{}]{arr}{}\end{DoxyParamCaption})}

\begin{DoxyVerb}Find repeats and repeat counts.

Parameters
----------
arr : array_like
    Input array. This is cast to float64.

Returns
-------
values : ndarray
    The unique values from the (flattened) input that are repeated.

counts : ndarray
    Number of times the corresponding 'value' is repeated.

Notes
-----
In numpy >= 1.9 `numpy.unique` provides similar functionality. The main
difference is that `find_repeats` only returns repeated values.

Examples
--------
>>> from scipy import stats
>>> stats.find_repeats([2, 1, 2, 3, 2, 2, 5])
RepeatedResults(values=array([2.]), counts=array([4]))

>>> stats.find_repeats([[10, 20, 1, 2], [5, 5, 4, 4]])
RepeatedResults(values=array([4.,  5.]), counts=array([2, 2]))
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l10033}{10033}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l10030}{Repeated\+Results}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l09338}{friedmanchisquare()}}, and \mbox{\hyperlink{__morestats_8py_source_l03838}{scipy.\+stats.\+\_\+morestats.\+wilcoxon()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_aec971e2362ba3d1144450412cfb4d674}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!fisher\_exact@{fisher\_exact}}
\index{fisher\_exact@{fisher\_exact}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{fisher\_exact()}{fisher\_exact()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_aec971e2362ba3d1144450412cfb4d674} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+fisher\+\_\+exact (\begin{DoxyParamCaption}\item[{}]{table}{, }\item[{}]{alternative}{ = {\ttfamily \textquotesingle{}two-\/sided\textquotesingle{}}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Perform a Fisher exact test on a 2x2 contingency table.

The null hypothesis is that the true odds ratio of the populations
underlying the observations is one, and the observations were sampled
from these populations under a condition: the marginals of the
resulting table must equal those of the observed table. The statistic
returned is the unconditional maximum likelihood estimate of the odds
ratio, and the p-value is the probability under the null hypothesis of
obtaining a table at least as extreme as the one that was actually
observed. There are other possible choices of statistic and two-sided
p-value definition associated with Fisher's exact test; please see the
Notes for more information.

Parameters
----------
table : array_like of ints
    A 2x2 contingency table.  Elements must be non-negative integers.
alternative : {'two-sided', 'less', 'greater'}, optional
    Defines the alternative hypothesis.
    The following options are available (default is 'two-sided'):

    * 'two-sided': the odds ratio of the underlying population is not one
    * 'less': the odds ratio of the underlying population is less than one
    * 'greater': the odds ratio of the underlying population is greater
      than one

    See the Notes for more details.

Returns
-------
res : SignificanceResult
    An object containing attributes:

    statistic : float
        This is the prior odds ratio, not a posterior estimate.
    pvalue : float
        The probability under the null hypothesis of obtaining a
        table at least as extreme as the one that was actually observed.

See Also
--------
chi2_contingency : Chi-square test of independence of variables in a
    contingency table.  This can be used as an alternative to
    `fisher_exact` when the numbers in the table are large.
contingency.odds_ratio : Compute the odds ratio (sample or conditional
    MLE) for a 2x2 contingency table.
barnard_exact : Barnard's exact test, which is a more powerful alternative
    than Fisher's exact test for 2x2 contingency tables.
boschloo_exact : Boschloo's exact test, which is a more powerful
    alternative than Fisher's exact test for 2x2 contingency tables.

Notes
-----
*Null hypothesis and p-values*

The null hypothesis is that the true odds ratio of the populations
underlying the observations is one, and the observations were sampled at
random from these populations under a condition: the marginals of the
resulting table must equal those of the observed table. Equivalently,
the null hypothesis is that the input table is from the hypergeometric
distribution with parameters (as used in `hypergeom`)
``M = a + b + c + d``, ``n = a + b`` and ``N = a + c``, where the
input table is ``[[a, b], [c, d]]``.  This distribution has support
``max(0, N + n - M) <= x <= min(N, n)``, or, in terms of the values
in the input table, ``min(0, a - d) <= x <= a + min(b, c)``.  ``x``
can be interpreted as the upper-left element of a 2x2 table, so the
tables in the distribution have form::

    [  x           n - x     ]
    [N - x    M - (n + N) + x]

For example, if::

    table = [6  2]
            [1  4]

then the support is ``2 <= x <= 7``, and the tables in the distribution
are::

    [2 6]   [3 5]   [4 4]   [5 3]   [6 2]  [7 1]
    [5 0]   [4 1]   [3 2]   [2 3]   [1 4]  [0 5]

The probability of each table is given by the hypergeometric distribution
``hypergeom.pmf(x, M, n, N)``.  For this example, these are (rounded to
three significant digits)::

    x       2      3      4      5       6        7
    p  0.0163  0.163  0.408  0.326  0.0816  0.00466

These can be computed with::

    >>> import numpy as np
    >>> from scipy.stats import hypergeom
    >>> table = np.array([[6, 2], [1, 4]])
    >>> M = table.sum()
    >>> n = table[0].sum()
    >>> N = table[:, 0].sum()
    >>> start, end = hypergeom.support(M, n, N)
    >>> hypergeom.pmf(np.arange(start, end+1), M, n, N)
    array([0.01631702, 0.16317016, 0.40792541, 0.32634033, 0.08158508,
           0.004662  ])

The two-sided p-value is the probability that, under the null hypothesis,
a random table would have a probability equal to or less than the
probability of the input table.  For our example, the probability of
the input table (where ``x = 6``) is 0.0816.  The x values where the
probability does not exceed this are 2, 6 and 7, so the two-sided p-value
is ``0.0163 + 0.0816 + 0.00466 ~= 0.10256``::

    >>> from scipy.stats import fisher_exact
    >>> res = fisher_exact(table, alternative='two-sided')
    >>> res.pvalue
    0.10256410256410257

The one-sided p-value for ``alternative='greater'`` is the probability
that a random table has ``x >= a``, which in our example is ``x >= 6``,
or ``0.0816 + 0.00466 ~= 0.08626``::

    >>> res = fisher_exact(table, alternative='greater')
    >>> res.pvalue
    0.08624708624708627

This is equivalent to computing the survival function of the
distribution at ``x = 5`` (one less than ``x`` from the input table,
because we want to include the probability of ``x = 6`` in the sum)::

    >>> hypergeom.sf(5, M, n, N)
    0.08624708624708627

For ``alternative='less'``, the one-sided p-value is the probability
that a random table has ``x <= a``, (i.e. ``x <= 6`` in our example),
or ``0.0163 + 0.163 + 0.408 + 0.326 + 0.0816 ~= 0.9949``::

    >>> res = fisher_exact(table, alternative='less')
    >>> res.pvalue
    0.9953379953379957

This is equivalent to computing the cumulative distribution function
of the distribution at ``x = 6``:

    >>> hypergeom.cdf(6, M, n, N)
    0.9953379953379957

*Odds ratio*

The calculated odds ratio is different from the value computed by the
R function ``fisher.test``.  This implementation returns the "sample"
or "unconditional" maximum likelihood estimate, while ``fisher.test``
in R uses the conditional maximum likelihood estimate.  To compute the
conditional maximum likelihood estimate of the odds ratio, use
`scipy.stats.contingency.odds_ratio`.

References
----------
.. [1] Fisher, Sir Ronald A, "The Design of Experiments:
       Mathematics of a Lady Tasting Tea." ISBN 978-0-486-41151-4, 1935.
.. [2] "Fisher's exact test",
       https://en.wikipedia.org/wiki/Fisher's_exact_test
.. [3] Emma V. Low et al. "Identifying the lowest effective dose of
       acetazolamide for the prophylaxis of acute mountain sickness:
       systematic review and meta-analysis."
       BMJ, 345, :doi:`10.1136/bmj.e6779`, 2012.

Examples
--------
In [3]_, the effective dose of acetazolamide for the prophylaxis of acute
mountain sickness was investigated. The study notably concluded:

    Acetazolamide 250 mg, 500 mg, and 750 mg daily were all efficacious for
    preventing acute mountain sickness. Acetazolamide 250 mg was the lowest
    effective dose with available evidence for this indication.

The following table summarizes the results of the experiment in which
some participants took a daily dose of acetazolamide 250 mg while others
took a placebo.
Cases of acute mountain sickness were recorded::

                                Acetazolamide   Control/Placebo
    Acute mountain sickness            7           17
    No                                15            5


Is there evidence that the acetazolamide 250 mg reduces the risk of
acute mountain sickness?
We begin by formulating a null hypothesis :math:`H_0`:

    The odds of experiencing acute mountain sickness are the same with
    the acetazolamide treatment as they are with placebo.

Let's assess the plausibility of this hypothesis with
Fisher's test.

>>> from scipy.stats import fisher_exact
>>> res = fisher_exact([[7, 17], [15, 5]], alternative='less')
>>> res.statistic
0.13725490196078433
>>> res.pvalue
0.0028841933752349743

Using a significance level of 5%, we would reject the null hypothesis in
favor of the alternative hypothesis: "The odds of experiencing acute
mountain sickness with acetazolamide treatment are less than the odds of
experiencing acute mountain sickness with placebo."

.. note::

    Because the null distribution of Fisher's exact test is formed under
    the assumption that both row and column sums are fixed, the result of
    the test are conservative when applied to an experiment in which the
    row sums are not fixed.

    In this case, the column sums are fixed; there are 22 subjects in each
    group. But the number of cases of acute mountain sickness is not
    (and cannot be) fixed before conducting the experiment. It is a
    consequence.

    Boschloo's test does not depend on the assumption that the row sums
    are fixed, and consequently, it provides a more powerful test in this
    situation.

    >>> from scipy.stats import boschloo_exact
    >>> res = boschloo_exact([[7, 17], [15, 5]], alternative='less')
    >>> res.statistic
    0.0028841933752349743
    >>> res.pvalue
    0.0015141406667567101

    We verify that the p-value is less than with `fisher_exact`.
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l04873}{4873}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_aa458038daef3b9b7c308d681b36f93ba}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!friedmanchisquare@{friedmanchisquare}}
\index{friedmanchisquare@{friedmanchisquare}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{friedmanchisquare()}{friedmanchisquare()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_aa458038daef3b9b7c308d681b36f93ba} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+friedmanchisquare (\begin{DoxyParamCaption}\item[{\texorpdfstring{$\ast$}{*}}]{samples}{}\end{DoxyParamCaption})}

\begin{DoxyVerb}Compute the Friedman test for repeated samples.

The Friedman test tests the null hypothesis that repeated samples of
the same individuals have the same distribution.  It is often used
to test for consistency among samples obtained in different ways.
For example, if two sampling techniques are used on the same set of
individuals, the Friedman test can be used to determine if the two
sampling techniques are consistent.

Parameters
----------
sample1, sample2, sample3... : array_like
    Arrays of observations.  All of the arrays must have the same number
    of elements.  At least three samples must be given.

Returns
-------
statistic : float
    The test statistic, correcting for ties.
pvalue : float
    The associated p-value assuming that the test statistic has a chi
    squared distribution.

Notes
-----
Due to the assumption that the test statistic has a chi squared
distribution, the p-value is only reliable for n > 10 and more than
6 repeated samples.

References
----------
.. [1] https://en.wikipedia.org/wiki/Friedman_test
.. [2] P. Sprent and N.C. Smeeton, "Applied Nonparametric Statistical
       Methods, Third Edition". Chapter 6, Section 6.3.2.

Examples
--------
In [2]_, the pulse rate (per minute) of a group of seven students was
measured before exercise, immediately after exercise and 5 minutes
after exercise. Is there evidence to suggest that the pulse rates on
these three occasions are similar?

We begin by formulating a null hypothesis :math:`H_0`:

    The pulse rates are identical on these three occasions.

Let's assess the plausibility of this hypothesis with a Friedman test.

>>> from scipy.stats import friedmanchisquare
>>> before = [72, 96, 88, 92, 74, 76, 82]
>>> immediately_after = [120, 120, 132, 120, 101, 96, 112]
>>> five_min_after = [76, 95, 104, 96, 84, 72, 76]
>>> res = friedmanchisquare(before, immediately_after, five_min_after)
>>> res.statistic
10.57142857142857
>>> res.pvalue
0.005063414171757498

Using a significance level of 5%, we would reject the null hypothesis in
favor of the alternative hypothesis: "the pulse rates are different on
these three occasions".
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l09338}{9338}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l10033}{find\+\_\+repeats()}}, and \mbox{\hyperlink{__stats__py_8py_source_l10123}{rankdata()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_aeb5f31167da564053a0666386c4e28b3}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!gmean@{gmean}}
\index{gmean@{gmean}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{gmean()}{gmean()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_aeb5f31167da564053a0666386c4e28b3} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+gmean (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{axis}{ = {\ttfamily 0}, }\item[{}]{dtype}{ = {\ttfamily None}, }\item[{}]{weights}{ = {\ttfamily None}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Compute the weighted geometric mean along the specified axis.

    The weighted geometric mean of the array :math:`a_i` associated to weights
    :math:`w_i` is:

    .. math::

        \exp \left( \frac{ \sum_{i=1}^n w_i \ln a_i }{ \sum_{i=1}^n w_i }
                   \right) \, ,

    and, with equal weights, it gives:

    .. math::

        \sqrt[n]{ \prod_{i=1}^n a_i } \, .

    Parameters
    ----------
    a : array_like
        Input array or object that can be converted to an array.
    axis : int or None, optional
        Axis along which the geometric mean is computed. Default is 0.
        If None, compute over the whole array `a`.
    dtype : dtype, optional
        Type to which the input arrays are cast before the calculation is
        performed.
    weights : array_like, optional
        The `weights` array must be broadcastable to the same shape as `a`.
        Default is None, which gives each value a weight of 1.0.

    Returns
    -------
    gmean : ndarray
        See `dtype` parameter above.

    See Also
    --------
    numpy.mean : Arithmetic average
    numpy.average : Weighted average
    hmean : Harmonic mean

    References
    ----------
    .. [1] "Weighted Geometric Mean", *Wikipedia*,
           https://en.wikipedia.org/wiki/Weighted_geometric_mean.
    .. [2] Grossman, J., Grossman, M., Katz, R., "Averages: A New Approach",
           Archimedes Foundation, 1983

    Examples
    --------
    >>> from scipy.stats import gmean
    >>> gmean([1, 4])
    2.0
    >>> gmean([1, 2, 3, 4, 5, 6, 7])
    3.3800151591412964
    >>> gmean([1, 4, 7], weights=[3, 1, 3])
    2.80668351922014
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l00130}{130}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l00300}{pmean()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a720304ac94e199a283c3cde4e66448f1}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!gstd@{gstd}}
\index{gstd@{gstd}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{gstd()}{gstd()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a720304ac94e199a283c3cde4e66448f1} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+gstd (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{axis}{ = {\ttfamily 0}, }\item[{}]{ddof}{ = {\ttfamily 1}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Calculate the geometric standard deviation of an array.

The geometric standard deviation describes the spread of a set of numbers
where the geometric mean is preferred. It is a multiplicative factor, and
so a dimensionless quantity.

It is defined as the exponent of the standard deviation of ``log(a)``.
Mathematically the population geometric standard deviation can be
evaluated as::

    gstd = exp(std(log(a)))

.. versionadded:: 1.3.0

Parameters
----------
a : array_like
    An array like object containing the sample data.
axis : int, tuple or None, optional
    Axis along which to operate. Default is 0. If None, compute over
    the whole array `a`.
ddof : int, optional
    Degree of freedom correction in the calculation of the
    geometric standard deviation. Default is 1.

Returns
-------
gstd : ndarray or float
    An array of the geometric standard deviation. If `axis` is None or `a`
    is a 1d array a float is returned.

See Also
--------
gmean : Geometric mean
numpy.std : Standard deviation
gzscore : Geometric standard score

Notes
-----
As the calculation requires the use of logarithms the geometric standard
deviation only supports strictly positive values. Any non-positive or
infinite values will raise a `ValueError`.
The geometric standard deviation is sometimes confused with the exponent of
the standard deviation, ``exp(std(a))``. Instead the geometric standard
deviation is ``exp(std(log(a)))``.
The default value for `ddof` is different to the default value (0) used
by other ddof containing functions, such as ``np.std`` and ``np.nanstd``.

References
----------
.. [1] "Geometric standard deviation", *Wikipedia*,
       https://en.wikipedia.org/wiki/Geometric_standard_deviation.
.. [2] Kirkwood, T. B., "Geometric means and measures of dispersion",
       Biometrics, vol. 35, pp. 908-909, 1979

Examples
--------
Find the geometric standard deviation of a log-normally distributed sample.
Note that the standard deviation of the distribution is one, on a
log scale this evaluates to approximately ``exp(1)``.

>>> import numpy as np
>>> from scipy.stats import gstd
>>> rng = np.random.default_rng()
>>> sample = rng.lognormal(mean=0, sigma=1, size=1000)
>>> gstd(sample)
2.810010162475324

Compute the geometric standard deviation of a multidimensional array and
of a given axis.

>>> a = np.arange(1, 25).reshape(2, 3, 4)
>>> gstd(a, axis=None)
2.2944076136018947
>>> gstd(a, axis=2)
array([[1.82424757, 1.22436866, 1.13183117],
       [1.09348306, 1.07244798, 1.05914985]])
>>> gstd(a, axis=(1,2))
array([2.12939215, 1.22120169])

The geometric standard deviation further handles masked arrays.

>>> a = np.arange(1, 25).reshape(2, 3, 4)
>>> ma = np.ma.masked_where(a > 16, a)
>>> ma
masked_array(
  data=[[[1, 2, 3, 4],
         [5, 6, 7, 8],
         [9, 10, 11, 12]],
        [[13, 14, 15, 16],
         [--, --, --, --],
         [--, --, --, --]]],
  mask=[[[False, False, False, False],
         [False, False, False, False],
         [False, False, False, False]],
        [[False, False, False, False],
         [ True,  True,  True,  True],
         [ True,  True,  True,  True]]],
  fill_value=999999)
>>> gstd(ma, axis=2)
masked_array(
  data=[[1.8242475707663655, 1.2243686572447428, 1.1318311657788478],
        [1.0934830582350938, --, --]],
  mask=[[False, False, False],
        [False,  True,  True]],
  fill_value=999999)
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l03152}{3152}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a269b1dd104e02d66f4c84e622f9702d1}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!gzscore@{gzscore}}
\index{gzscore@{gzscore}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{gzscore()}{gzscore()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a269b1dd104e02d66f4c84e622f9702d1} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+gzscore (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{\texorpdfstring{$\ast$}{*}}]{}{, }\item[{}]{axis}{ = {\ttfamily 0}, }\item[{}]{ddof}{ = {\ttfamily 0}, }\item[{}]{nan\+\_\+policy}{ = {\ttfamily \textquotesingle{}propagate\textquotesingle{}}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Compute the geometric standard score.

Compute the geometric z score of each strictly positive value in the
sample, relative to the geometric mean and standard deviation.
Mathematically the geometric z score can be evaluated as::

    gzscore = log(a/gmu) / log(gsigma)

where ``gmu`` (resp. ``gsigma``) is the geometric mean (resp. standard
deviation).

Parameters
----------
a : array_like
    Sample data.
axis : int or None, optional
    Axis along which to operate. Default is 0. If None, compute over
    the whole array `a`.
ddof : int, optional
    Degrees of freedom correction in the calculation of the
    standard deviation. Default is 0.
nan_policy : {'propagate', 'raise', 'omit'}, optional
    Defines how to handle when input contains nan. 'propagate' returns nan,
    'raise' throws an error, 'omit' performs the calculations ignoring nan
    values. Default is 'propagate'.  Note that when the value is 'omit',
    nans in the input also propagate to the output, but they do not affect
    the geometric z scores computed for the non-nan values.

Returns
-------
gzscore : array_like
    The geometric z scores, standardized by geometric mean and geometric
    standard deviation of input array `a`.

See Also
--------
gmean : Geometric mean
gstd : Geometric standard deviation
zscore : Standard score

Notes
-----
This function preserves ndarray subclasses, and works also with
matrices and masked arrays (it uses ``asanyarray`` instead of
``asarray`` for parameters).

.. versionadded:: 1.8

References
----------
.. [1] "Geometric standard score", *Wikipedia*,
       https://en.wikipedia.org/wiki/Geometric_standard_deviation#Geometric_standard_score.

Examples
--------
Draw samples from a log-normal distribution:

>>> import numpy as np
>>> from scipy.stats import zscore, gzscore
>>> import matplotlib.pyplot as plt

>>> rng = np.random.default_rng()
>>> mu, sigma = 3., 1.  # mean and standard deviation
>>> x = rng.lognormal(mu, sigma, size=500)

Display the histogram of the samples:

>>> fig, ax = plt.subplots()
>>> ax.hist(x, 50)
>>> plt.show()

Display the histogram of the samples standardized by the classical zscore.
Distribution is rescaled but its shape is unchanged.

>>> fig, ax = plt.subplots()
>>> ax.hist(zscore(x), 50)
>>> plt.show()

Demonstrate that the distribution of geometric zscores is rescaled and
quasinormal:

>>> fig, ax = plt.subplots()
>>> ax.hist(gzscore(x), 50)
>>> plt.show()
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l02976}{2976}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l02891}{zscore()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_ab8582abb8de89103fad5ce9260fe5f7e}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!hmean@{hmean}}
\index{hmean@{hmean}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{hmean()}{hmean()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_ab8582abb8de89103fad5ce9260fe5f7e} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+hmean (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{axis}{ = {\ttfamily 0}, }\item[{}]{dtype}{ = {\ttfamily None}, }\item[{\texorpdfstring{$\ast$}{*}}]{}{, }\item[{}]{weights}{ = {\ttfamily None}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Calculate the weighted harmonic mean along the specified axis.

    The weighted harmonic mean of the array :math:`a_i` associated to weights
    :math:`w_i` is:

    .. math::

        \frac{ \sum_{i=1}^n w_i }{ \sum_{i=1}^n \frac{w_i}{a_i} } \, ,

    and, with equal weights, it gives:

    .. math::

        \frac{ n }{ \sum_{i=1}^n \frac{1}{a_i} } \, .

    Parameters
    ----------
    a : array_like
        Input array, masked array or object that can be converted to an array.
    axis : int or None, optional
        Axis along which the harmonic mean is computed. Default is 0.
        If None, compute over the whole array `a`.
    dtype : dtype, optional
        Type of the returned array and of the accumulator in which the
        elements are summed. If `dtype` is not specified, it defaults to the
        dtype of `a`, unless `a` has an integer `dtype` with a precision less
        than that of the default platform integer. In that case, the default
        platform integer is used.
    weights : array_like, optional
        The weights array can either be 1-D (in which case its length must be
        the size of `a` along the given `axis`) or of the same shape as `a`.
        Default is None, which gives each value a weight of 1.0.

        .. versionadded:: 1.9

    Returns
    -------
    hmean : ndarray
        See `dtype` parameter above.

    See Also
    --------
    numpy.mean : Arithmetic average
    numpy.average : Weighted average
    gmean : Geometric mean

    Notes
    -----
    The harmonic mean is computed over a single dimension of the input
    array, axis=0 by default, or all values in the array if axis=None.
    float64 intermediate and return values are used for integer inputs.

    References
    ----------
    .. [1] "Weighted Harmonic Mean", *Wikipedia*,
           https://en.wikipedia.org/wiki/Harmonic_mean#Weighted_harmonic_mean
    .. [2] Ferger, F., "The nature and use of the harmonic mean", Journal of
           the American Statistical Association, vol. 26, pp. 36-40, 1931

    Examples
    --------
    >>> from scipy.stats import hmean
    >>> hmean([1, 4])
    1.6000000000000001
    >>> hmean([1, 2, 3, 4, 5, 6, 7])
    2.6997245179063363
    >>> hmean([1, 4, 7], weights=[3, 1, 3])
    1.9029126213592233
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l00205}{205}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_ad8ae1ff6d5c66e636d72a57ad7ea0472}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!iqr@{iqr}}
\index{iqr@{iqr}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{iqr()}{iqr()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_ad8ae1ff6d5c66e636d72a57ad7ea0472} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+iqr (\begin{DoxyParamCaption}\item[{}]{x}{, }\item[{}]{axis}{ = {\ttfamily None}, }\item[{}]{rng}{ = {\ttfamily (25,~75)}, }\item[{}]{scale}{ = {\ttfamily 1.0}, }\item[{}]{nan\+\_\+policy}{ = {\ttfamily \textquotesingle{}propagate\textquotesingle{}}, }\item[{}]{interpolation}{ = {\ttfamily \textquotesingle{}linear\textquotesingle{}}, }\item[{}]{keepdims}{ = {\ttfamily False}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Compute the interquartile range of the data along the specified axis.

The interquartile range (IQR) is the difference between the 75th and
25th percentile of the data. It is a measure of the dispersion
similar to standard deviation or variance, but is much more robust
against outliers [2]_.

The ``rng`` parameter allows this function to compute other
percentile ranges than the actual IQR. For example, setting
``rng=(0, 100)`` is equivalent to `numpy.ptp`.

The IQR of an empty array is `np.nan`.

.. versionadded:: 0.18.0

Parameters
----------
x : array_like
Input array or object that can be converted to an array.
axis : int or sequence of int, optional
Axis along which the range is computed. The default is to
compute the IQR for the entire array.
rng : Two-element sequence containing floats in range of [0,100] optional
Percentiles over which to compute the range. Each must be
between 0 and 100, inclusive. The default is the true IQR:
``(25, 75)``. The order of the elements is not important.
scale : scalar or str, optional
The numerical value of scale will be divided out of the final
result. The following string values are recognized:

* 'raw' : No scaling, just return the raw IQR.
**Deprecated!**  Use ``scale=1`` instead.
* 'normal' : Scale by
:math:`2 \sqrt{2} erf^{-1}(\frac{1}{2}) \approx 1.349`.

The default is 1.0. The use of ``scale='raw'`` is deprecated infavor
of ``scale=1`` and will raise an error in SciPy 1.12.0.
Array-like `scale` is also allowed, as long
as it broadcasts correctly to the output such that
``out / scale`` is a valid operation. The output dimensions
depend on the input array, `x`, the `axis` argument, and the
`keepdims` flag.
nan_policy : {'propagate', 'raise', 'omit'}, optional
Defines how to handle when input contains nan.
The following options are available (default is 'propagate'):

* 'propagate': returns nan
* 'raise': throws an error
* 'omit': performs the calculations ignoring nan values
interpolation : str, optional

Specifies the interpolation method to use when the percentile
boundaries lie between two data points ``i`` and ``j``.
The following options are available (default is 'linear'):

* 'linear': ``i + (j - i)*fraction``, where ``fraction`` is the
fractional part of the index surrounded by ``i`` and ``j``.
* 'lower': ``i``.
* 'higher': ``j``.
* 'nearest': ``i`` or ``j`` whichever is nearest.
* 'midpoint': ``(i + j)/2``.

For NumPy >= 1.22.0, the additional options provided by the ``method``
keyword of `numpy.percentile` are also valid.

keepdims : bool, optional
If this is set to True, the reduced axes are left in the
result as dimensions with size one. With this option, the result
will broadcast correctly against the original array `x`.

Returns
-------
iqr : scalar or ndarray
If ``axis=None``, a scalar is returned. If the input contains
integers or floats of smaller precision than ``np.float64``, then the
output data-type is ``np.float64``. Otherwise, the output data-type is
the same as that of the input.

See Also
--------
numpy.std, numpy.var

References
----------
.. [1] "Interquartile range" https://en.wikipedia.org/wiki/Interquartile_range
.. [2] "Robust measures of scale" https://en.wikipedia.org/wiki/Robust_measures_of_scale
.. [3] "Quantile" https://en.wikipedia.org/wiki/Quantile

Examples
--------
>>> import numpy as np
>>> from scipy.stats import iqr
>>> x = np.array([[10, 7, 4], [3, 2, 1]])
>>> x
array([[10,  7,  4],
[ 3,  2,  1]])
>>> iqr(x)
4.0
>>> iqr(x, axis=0)
array([ 3.5,  2.5,  1.5])
>>> iqr(x, axis=1)
array([ 3.,  1.])
>>> iqr(x, axis=1, keepdims=True)
array([[ 3.],
[ 1.]])
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l03306}{3306}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



Referenced by \mbox{\hyperlink{__hypotests_8py_source_l00028}{scipy.\+stats.\+\_\+hypotests.\+epps\+\_\+singleton\+\_\+2samp()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a052799c289bd2d586b43e9112ec70e62}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!jarque\_bera@{jarque\_bera}}
\index{jarque\_bera@{jarque\_bera}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{jarque\_bera()}{jarque\_bera()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a052799c289bd2d586b43e9112ec70e62} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+jarque\+\_\+bera (\begin{DoxyParamCaption}\item[{}]{x}{, }\item[{\texorpdfstring{$\ast$}{*}}]{}{, }\item[{}]{axis}{ = {\ttfamily None}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Perform the Jarque-Bera goodness of fit test on sample data.

The Jarque-Bera test tests whether the sample data has the skewness and
kurtosis matching a normal distribution.

Note that this test only works for a large enough number of data samples
(>2000) as the test statistic asymptotically has a Chi-squared distribution
with 2 degrees of freedom.

Parameters
----------
x : array_like
    Observations of a random variable.
axis : int or None, default: 0
    If an int, the axis of the input along which to compute the statistic.
    The statistic of each axis-slice (e.g. row) of the input will appear in
    a corresponding element of the output.
    If ``None``, the input will be raveled before computing the statistic.

Returns
-------
result : SignificanceResult
    An object with the following attributes:

    statistic : float
        The test statistic.
    pvalue : float
        The p-value for the hypothesis test.

References
----------
.. [1] Jarque, C. and Bera, A. (1980) "Efficient tests for normality,
       homoscedasticity and serial independence of regression residuals",
       6 Econometric Letters 255-259.
.. [2] Shapiro, S. S., & Wilk, M. B. (1965). An analysis of variance test
       for normality (complete samples). Biometrika, 52(3/4), 591-611.
.. [3] B. Phipson and G. K. Smyth. "Permutation P-values Should Never Be
       Zero: Calculating Exact P-values When Permutations Are Randomly
       Drawn." Statistical Applications in Genetics and Molecular Biology
       9.1 (2010).
.. [4] Panagiotakos, D. B. (2008). The value of p-value in biomedical
       research. The open cardiovascular medicine journal, 2, 97.

Examples
--------
Suppose we wish to infer from measurements whether the weights of adult
human males in a medical study are not normally distributed [2]_.
The weights (lbs) are recorded in the array ``x`` below.

>>> import numpy as np
>>> x = np.array([148, 154, 158, 160, 161, 162, 166, 170, 182, 195, 236])

The Jarque-Bera test begins by computing a statistic based on the sample
skewness and kurtosis.

>>> from scipy import stats
>>> res = stats.jarque_bera(x)
>>> res.statistic
6.982848237344646

Because the normal distribution has zero skewness and zero
("excess" or "Fisher") kurtosis, the value of this statistic tends to be
low for samples drawn from a normal distribution.

The test is performed by comparing the observed value of the statistic
against the null distribution: the distribution of statistic values derived
under the null hypothesis that the weights were drawn from a normal
distribution.
For the Jarque-Bera test, the null distribution for very large samples is
the chi-squared distribution with two degrees of freedom.

>>> import matplotlib.pyplot as plt
>>> dist = stats.chi2(df=2)
>>> jb_val = np.linspace(0, 11, 100)
>>> pdf = dist.pdf(jb_val)
>>> fig, ax = plt.subplots(figsize=(8, 5))
>>> def jb_plot(ax):  # we'll re-use this
...     ax.plot(jb_val, pdf)
...     ax.set_title("Jarque-Bera Null Distribution")
...     ax.set_xlabel("statistic")
...     ax.set_ylabel("probability density")
>>> jb_plot(ax)
>>> plt.show()

The comparison is quantified by the p-value: the proportion of values in
the null distribution greater than or equal to the observed value of the
statistic.

>>> fig, ax = plt.subplots(figsize=(8, 5))
>>> jb_plot(ax)
>>> pvalue = dist.sf(res.statistic)
>>> annotation = (f'p-value={pvalue:.6f}\n(shaded area)')
>>> props = dict(facecolor='black', width=1, headwidth=5, headlength=8)
>>> _ = ax.annotate(annotation, (7.5, 0.01), (8, 0.05), arrowprops=props)
>>> i = jb_val >= res.statistic  # indices of more extreme statistic values
>>> ax.fill_between(jb_val[i], y1=0, y2=pdf[i])
>>> ax.set_xlim(0, 11)
>>> ax.set_ylim(0, 0.3)
>>> plt.show()
>>> res.pvalue
0.03045746622458189

If the p-value is "small" - that is, if there is a low probability of
sampling data from a normally distributed population that produces such an
extreme value of the statistic - this may be taken as evidence against
the null hypothesis in favor of the alternative: the weights were not
drawn from a normal distribution. Note that:

- The inverse is not true; that is, the test is not used to provide
  evidence for the null hypothesis.
- The threshold for values that will be considered "small" is a choice that
  should be made before the data is analyzed [3]_ with consideration of the
  risks of both false positives (incorrectly rejecting the null hypothesis)
  and false negatives (failure to reject a false null hypothesis).

Note that the chi-squared distribution provides an asymptotic approximation
of the null distribution; it is only accurate for samples with many
observations. For small samples like ours, `scipy.stats.monte_carlo_test`
may provide a more accurate, albeit stochastic, approximation of the
exact p-value.

>>> def statistic(x, axis):
...     # underlying calculation of the Jarque Bera statistic
...     s = stats.skew(x, axis=axis)
...     k = stats.kurtosis(x, axis=axis)
...     return x.shape[axis]/6 * (s**2 + k**2/4)
>>> res = stats.monte_carlo_test(x, stats.norm.rvs, statistic,
...                              alternative='greater')
>>> fig, ax = plt.subplots(figsize=(8, 5))
>>> jb_plot(ax)
>>> ax.hist(res.null_distribution, np.linspace(0, 10, 50),
...         density=True)
>>> ax.legend(['aymptotic approximation (many observations)',
...            'Monte Carlo approximation (11 observations)'])
>>> plt.show()
>>> res.pvalue
0.0097  # may vary

Furthermore, despite their stochastic nature, p-values computed in this way
can be used to exactly control the rate of false rejections of the null
hypothesis [4]_.
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l02000}{2000}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l01213}{kurtosis()}}, and \mbox{\hyperlink{__stats__py_8py_source_l01110}{skew()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a40ce454d71767fd548c3da8a056cbd35}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!kendalltau@{kendalltau}}
\index{kendalltau@{kendalltau}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{kendalltau()}{kendalltau()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a40ce454d71767fd548c3da8a056cbd35} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+kendalltau (\begin{DoxyParamCaption}\item[{}]{x}{, }\item[{}]{y}{, }\item[{}]{initial\+\_\+lexsort}{ = {\ttfamily None}, }\item[{}]{nan\+\_\+policy}{ = {\ttfamily \textquotesingle{}propagate\textquotesingle{}}, }\item[{}]{method}{ = {\ttfamily \textquotesingle{}auto\textquotesingle{}}, }\item[{}]{variant}{ = {\ttfamily \textquotesingle{}b\textquotesingle{}}, }\item[{}]{alternative}{ = {\ttfamily \textquotesingle{}two-\/sided\textquotesingle{}}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Calculate Kendall's tau, a correlation measure for ordinal data.

Kendall's tau is a measure of the correspondence between two rankings.
Values close to 1 indicate strong agreement, and values close to -1
indicate strong disagreement. This implements two variants of Kendall's
tau: tau-b (the default) and tau-c (also known as Stuart's tau-c). These
differ only in how they are normalized to lie within the range -1 to 1;
the hypothesis tests (their p-values) are identical. Kendall's original
tau-a is not implemented separately because both tau-b and tau-c reduce
to tau-a in the absence of ties.

Parameters
----------
x, y : array_like
    Arrays of rankings, of the same shape. If arrays are not 1-D, they
    will be flattened to 1-D.
initial_lexsort : bool, optional, deprecated
    This argument is unused.

    .. deprecated:: 1.10.0
       `kendalltau` keyword argument `initial_lexsort` is deprecated as it
       is unused and will be removed in SciPy 1.12.0.
nan_policy : {'propagate', 'raise', 'omit'}, optional
    Defines how to handle when input contains nan.
    The following options are available (default is 'propagate'):

      * 'propagate': returns nan
      * 'raise': throws an error
      * 'omit': performs the calculations ignoring nan values

method : {'auto', 'asymptotic', 'exact'}, optional
    Defines which method is used to calculate the p-value [5]_.
    The following options are available (default is 'auto'):

      * 'auto': selects the appropriate method based on a trade-off
        between speed and accuracy
      * 'asymptotic': uses a normal approximation valid for large samples
      * 'exact': computes the exact p-value, but can only be used if no ties
        are present. As the sample size increases, the 'exact' computation
        time may grow and the result may lose some precision.
variant : {'b', 'c'}, optional
    Defines which variant of Kendall's tau is returned. Default is 'b'.
alternative : {'two-sided', 'less', 'greater'}, optional
    Defines the alternative hypothesis. Default is 'two-sided'.
    The following options are available:

    * 'two-sided': the rank correlation is nonzero
    * 'less': the rank correlation is negative (less than zero)
    * 'greater':  the rank correlation is positive (greater than zero)

Returns
-------
res : SignificanceResult
    An object containing attributes:

    statistic : float
       The tau statistic.
    pvalue : float
       The p-value for a hypothesis test whose null hypothesis is
       an absence of association, tau = 0.

See Also
--------
spearmanr : Calculates a Spearman rank-order correlation coefficient.
theilslopes : Computes the Theil-Sen estimator for a set of points (x, y).
weightedtau : Computes a weighted version of Kendall's tau.

Notes
-----
The definition of Kendall's tau that is used is [2]_::

  tau_b = (P - Q) / sqrt((P + Q + T) * (P + Q + U))

  tau_c = 2 (P - Q) / (n**2 * (m - 1) / m)

where P is the number of concordant pairs, Q the number of discordant
pairs, T the number of ties only in `x`, and U the number of ties only in
`y`.  If a tie occurs for the same pair in both `x` and `y`, it is not
added to either T or U. n is the total number of samples, and m is the
number of unique values in either `x` or `y`, whichever is smaller.

References
----------
.. [1] Maurice G. Kendall, "A New Measure of Rank Correlation", Biometrika
       Vol. 30, No. 1/2, pp. 81-93, 1938.
.. [2] Maurice G. Kendall, "The treatment of ties in ranking problems",
       Biometrika Vol. 33, No. 3, pp. 239-251. 1945.
.. [3] Gottfried E. Noether, "Elements of Nonparametric Statistics", John
       Wiley & Sons, 1967.
.. [4] Peter M. Fenwick, "A new data structure for cumulative frequency
       tables", Software: Practice and Experience, Vol. 24, No. 3,
       pp. 327-336, 1994.
.. [5] Maurice G. Kendall, "Rank Correlation Methods" (4th Edition),
       Charles Griffin & Co., 1970.
.. [6] Kershenobich, D., Fierro, F. J., & Rojkind, M. (1970). The
       relationship between the free pool of proline and collagen content
       in human liver cirrhosis. The Journal of Clinical Investigation,
       49(12), 2246-2249.
.. [7] Hollander, M., Wolfe, D. A., & Chicken, E. (2013). Nonparametric
       statistical methods. John Wiley & Sons.
.. [8] B. Phipson and G. K. Smyth. "Permutation P-values Should Never Be
       Zero: Calculating Exact P-values When Permutations Are Randomly
       Drawn." Statistical Applications in Genetics and Molecular Biology
       9.1 (2010).

Examples
--------
Consider the following data from [6]_, which studied the relationship
between free proline (an amino acid) and total collagen (a protein often
found in connective tissue) in unhealthy human livers.

The ``x`` and ``y`` arrays below record measurements of the two compounds.
The observations are paired: each free proline measurement was taken from
the same liver as the total collagen measurement at the same index.

>>> import numpy as np
>>> # total collagen (mg/g dry weight of liver)
>>> x = np.array([7.1, 7.1, 7.2, 8.3, 9.4, 10.5, 11.4])
>>> # free proline ( mole/g dry weight of liver)
>>> y = np.array([2.8, 2.9, 2.8, 2.6, 3.5, 4.6, 5.0])

These data were analyzed in [7]_ using Spearman's correlation coefficient,
a statistic similar to to Kendall's tau in that it is also sensitive to
ordinal correlation between the samples. Let's perform an analagous study
using Kendall's tau.

>>> from scipy import stats
>>> res = stats.kendalltau(x, y)
>>> res.statistic
0.5499999999999999

The value of this statistic tends to be high (close to 1) for samples with
a strongly positive ordinal correlation, low (close to -1) for samples with
a strongly negative ordinal correlation, and small in magnitude (close to
zero) for samples with weak ordinal correlation.

The test is performed by comparing the observed value of the
statistic against the null distribution: the distribution of statistic
values derived under the null hypothesis that total collagen and free
proline measurements are independent.

For this test, the null distribution for large samples without ties is
approximated as the normal distribution with variance
``(2*(2*n + 5))/(9*n*(n - 1))``, where ``n = len(x)``.

>>> import matplotlib.pyplot as plt
>>> n = len(x)  # len(x) == len(y)
>>> var = (2*(2*n + 5))/(9*n*(n - 1))
>>> dist = stats.norm(scale=np.sqrt(var))
>>> z_vals = np.linspace(-1.25, 1.25, 100)
>>> pdf = dist.pdf(z_vals)
>>> fig, ax = plt.subplots(figsize=(8, 5))
>>> def plot(ax):  # we'll re-use this
...     ax.plot(z_vals, pdf)
...     ax.set_title("Kendall Tau Test Null Distribution")
...     ax.set_xlabel("statistic")
...     ax.set_ylabel("probability density")
>>> plot(ax)
>>> plt.show()

The comparison is quantified by the p-value: the proportion of values in
the null distribution as extreme or more extreme than the observed
value of the statistic. In a two-sided test in which the statistic is
positive, elements of the null distribution greater than the transformed
statistic and elements of the null distribution less than the negative of
the observed statistic are both considered "more extreme".

>>> fig, ax = plt.subplots(figsize=(8, 5))
>>> plot(ax)
>>> pvalue = dist.cdf(-res.statistic) + dist.sf(res.statistic)
>>> annotation = (f'p-value={pvalue:.4f}\n(shaded area)')
>>> props = dict(facecolor='black', width=1, headwidth=5, headlength=8)
>>> _ = ax.annotate(annotation, (0.65, 0.15), (0.8, 0.3), arrowprops=props)
>>> i = z_vals >= res.statistic
>>> ax.fill_between(z_vals[i], y1=0, y2=pdf[i], color='C0')
>>> i = z_vals <= -res.statistic
>>> ax.fill_between(z_vals[i], y1=0, y2=pdf[i], color='C0')
>>> ax.set_xlim(-1.25, 1.25)
>>> ax.set_ylim(0, 0.5)
>>> plt.show()
>>> res.pvalue
0.09108705741631495  # approximate p-value

Note that there is slight disagreement between the shaded area of the curve
and the p-value returned by `kendalltau`. This is because our data has
ties, and we have neglected a tie correction to the null distribution
variance that `kendalltau` performs. For samples without ties, the shaded
areas of our plot and p-value returned by `kendalltau` would match exactly.

If the p-value is "small" - that is, if there is a low probability of
sampling data from independent distributions that produces such an extreme
value of the statistic - this may be taken as evidence against the null
hypothesis in favor of the alternative: the distribution of total collagen
and free proline are *not* independent. Note that:

- The inverse is not true; that is, the test is not used to provide
  evidence for the null hypothesis.
- The threshold for values that will be considered "small" is a choice that
  should be made before the data is analyzed [8]_ with consideration of the
  risks of both false positives (incorrectly rejecting the null hypothesis)
  and false negatives (failure to reject a false null hypothesis).
- Small p-values are not evidence for a *large* effect; rather, they can
  only provide evidence for a "significant" effect, meaning that they are
  unlikely to have occurred under the null hypothesis.

For samples without ties of moderate size, `kendalltau` can compute the
p-value exactly. However, in the presence of ties, `kendalltau` resorts
to an asymptotic approximation. Nonetheles, we can use a permutation test
to compute the null distribution exactly: Under the null hypothesis that
total collagen and free proline are independent, each of the free proline
measurements were equally likely to have been observed with any of the
total collagen measurements. Therefore, we can form an *exact* null
distribution by calculating the statistic under each possible pairing of
elements between ``x`` and ``y``.

>>> def statistic(x):  # explore all possible pairings by permuting `x`
...     return stats.kendalltau(x, y).statistic  # ignore pvalue
>>> ref = stats.permutation_test((x,), statistic,
...                              permutation_type='pairings')
>>> fig, ax = plt.subplots(figsize=(8, 5))
>>> plot(ax)
>>> bins = np.linspace(-1.25, 1.25, 25)
>>> ax.hist(ref.null_distribution, bins=bins, density=True)
>>> ax.legend(['aymptotic approximation\n(many observations)',
...            'exact null distribution'])
>>> plot(ax)
>>> plt.show()
>>> ref.pvalue
0.12222222222222222  # exact p-value

Note that there is significant disagreement between the exact p-value
calculated here and the approximation returned by `kendalltau` above. For
small samples with ties, consider performing a permutation test for more
accurate results.
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l05594}{5594}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l01415}{\+\_\+normtest\+\_\+finish()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a15cf6e865d5b5a8be777b158f295610d}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!kruskal@{kruskal}}
\index{kruskal@{kruskal}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{kruskal()}{kruskal()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a15cf6e865d5b5a8be777b158f295610d} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+kruskal (\begin{DoxyParamCaption}\item[{\texorpdfstring{$\ast$}{*}}]{samples}{, }\item[{}]{nan\+\_\+policy}{ = {\ttfamily \textquotesingle{}propagate\textquotesingle{}}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Compute the Kruskal-Wallis H-test for independent samples.

The Kruskal-Wallis H-test tests the null hypothesis that the population
median of all of the groups are equal.  It is a non-parametric version of
ANOVA.  The test works on 2 or more independent samples, which may have
different sizes.  Note that rejecting the null hypothesis does not
indicate which of the groups differs.  Post hoc comparisons between
groups are required to determine which groups are different.

Parameters
----------
sample1, sample2, ... : array_like
   Two or more arrays with the sample measurements can be given as
   arguments. Samples must be one-dimensional.
nan_policy : {'propagate', 'raise', 'omit'}, optional
    Defines how to handle when input contains nan.
    The following options are available (default is 'propagate'):

      * 'propagate': returns nan
      * 'raise': throws an error
      * 'omit': performs the calculations ignoring nan values

Returns
-------
statistic : float
   The Kruskal-Wallis H statistic, corrected for ties.
pvalue : float
   The p-value for the test using the assumption that H has a chi
   square distribution. The p-value returned is the survival function of
   the chi square distribution evaluated at H.

See Also
--------
f_oneway : 1-way ANOVA.
mannwhitneyu : Mann-Whitney rank test on two samples.
friedmanchisquare : Friedman test for repeated measurements.

Notes
-----
Due to the assumption that H has a chi square distribution, the number
of samples in each group must not be too small.  A typical rule is
that each sample must have at least 5 measurements.

References
----------
.. [1] W. H. Kruskal & W. W. Wallis, "Use of Ranks in
   One-Criterion Variance Analysis", Journal of the American Statistical
   Association, Vol. 47, Issue 260, pp. 583-621, 1952.
.. [2] https://en.wikipedia.org/wiki/Kruskal-Wallis_one-way_analysis_of_variance

Examples
--------
>>> from scipy import stats
>>> x = [1, 3, 5, 7, 9]
>>> y = [2, 4, 6, 8, 10]
>>> stats.kruskal(x, y)
KruskalResult(statistic=0.2727272727272734, pvalue=0.6015081344405895)

>>> x = [1, 1, 1]
>>> y = [2, 2, 2]
>>> z = [2, 2]
>>> stats.kruskal(x, y, z)
KruskalResult(statistic=7.0, pvalue=0.0301973834223185)
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l09215}{9215}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l10094}{\+\_\+square\+\_\+of\+\_\+sums()}}, \mbox{\hyperlink{__stats__py_8py_source_l10123}{rankdata()}}, and \mbox{\hyperlink{__stats__py_8py_source_l09084}{tiecorrect()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_afd9c2533435c2d2b47aa429ee290ae43}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!ks\_1samp@{ks\_1samp}}
\index{ks\_1samp@{ks\_1samp}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{ks\_1samp()}{ks\_1samp()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_afd9c2533435c2d2b47aa429ee290ae43} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+ks\+\_\+1samp (\begin{DoxyParamCaption}\item[{}]{x}{, }\item[{}]{cdf}{, }\item[{}]{args}{ = {\ttfamily ()}, }\item[{}]{alternative}{ = {\ttfamily \textquotesingle{}two-\/sided\textquotesingle{}}, }\item[{}]{method}{ = {\ttfamily \textquotesingle{}auto\textquotesingle{}}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Performs the one-sample Kolmogorov-Smirnov test for goodness of fit.

This test compares the underlying distribution F(x) of a sample
against a given continuous distribution G(x). See Notes for a description
of the available null and alternative hypotheses.

Parameters
----------
x : array_like
    a 1-D array of observations of iid random variables.
cdf : callable
    callable used to calculate the cdf.
args : tuple, sequence, optional
    Distribution parameters, used with `cdf`.
alternative : {'two-sided', 'less', 'greater'}, optional
    Defines the null and alternative hypotheses. Default is 'two-sided'.
    Please see explanations in the Notes below.
method : {'auto', 'exact', 'approx', 'asymp'}, optional
    Defines the distribution used for calculating the p-value.
    The following options are available (default is 'auto'):

      * 'auto' : selects one of the other options.
      * 'exact' : uses the exact distribution of test statistic.
      * 'approx' : approximates the two-sided probability with twice
        the one-sided probability
      * 'asymp': uses asymptotic distribution of test statistic

Returns
-------
res: KstestResult
    An object containing attributes:

    statistic : float
        KS test statistic, either D+, D-, or D (the maximum of the two)
    pvalue : float
        One-tailed or two-tailed p-value.
    statistic_location : float
        Value of `x` corresponding with the KS statistic; i.e., the
        distance between the empirical distribution function and the
        hypothesized cumulative distribution function is measured at this
        observation.
    statistic_sign : int
        +1 if the KS statistic is the maximum positive difference between
        the empirical distribution function and the hypothesized cumulative
        distribution function (D+); -1 if the KS statistic is the maximum
        negative difference (D-).


See Also
--------
ks_2samp, kstest

Notes
-----
There are three options for the null and corresponding alternative
hypothesis that can be selected using the `alternative` parameter.

- `two-sided`: The null hypothesis is that the two distributions are
  identical, F(x)=G(x) for all x; the alternative is that they are not
  identical.

- `less`: The null hypothesis is that F(x) >= G(x) for all x; the
  alternative is that F(x) < G(x) for at least one x.

- `greater`: The null hypothesis is that F(x) <= G(x) for all x; the
  alternative is that F(x) > G(x) for at least one x.

Note that the alternative hypotheses describe the *CDFs* of the
underlying distributions, not the observed values. For example,
suppose x1 ~ F and x2 ~ G. If F(x) > G(x) for all x, the values in
x1 tend to be less than those in x2.

Examples
--------
Suppose we wish to test the null hypothesis that a sample is distributed
according to the standard normal.
We choose a confidence level of 95%; that is, we will reject the null
hypothesis in favor of the alternative if the p-value is less than 0.05.

When testing uniformly distributed data, we would expect the
null hypothesis to be rejected.

>>> import numpy as np
>>> from scipy import stats
>>> rng = np.random.default_rng()
>>> stats.ks_1samp(stats.uniform.rvs(size=100, random_state=rng),
...                stats.norm.cdf)
KstestResult(statistic=0.5001899973268688, pvalue=1.1616392184763533e-23)

Indeed, the p-value is lower than our threshold of 0.05, so we reject the
null hypothesis in favor of the default "two-sided" alternative: the data
are *not* distributed according to the standard normal.

When testing random variates from the standard normal distribution, we
expect the data to be consistent with the null hypothesis most of the time.

>>> x = stats.norm.rvs(size=100, random_state=rng)
>>> stats.ks_1samp(x, stats.norm.cdf)
KstestResult(statistic=0.05345882212970396, pvalue=0.9227159037744717)

As expected, the p-value of 0.92 is not below our threshold of 0.05, so
we cannot reject the null hypothesis.

Suppose, however, that the random variates are distributed according to
a normal distribution that is shifted toward greater values. In this case,
the cumulative density function (CDF) of the underlying distribution tends
to be *less* than the CDF of the standard normal. Therefore, we would
expect the null hypothesis to be rejected with ``alternative='less'``:

>>> x = stats.norm.rvs(size=100, loc=0.5, random_state=rng)
>>> stats.ks_1samp(x, stats.norm.cdf, alternative='less')
KstestResult(statistic=0.17482387821055168, pvalue=0.001913921057766743)

and indeed, with p-value smaller than our threshold, we reject the null
hypothesis in favor of the alternative.
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l08310}{8310}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l08286}{\+\_\+compute\+\_\+dminus()}}, and \mbox{\hyperlink{__stats__py_8py_source_l08262}{\+\_\+compute\+\_\+dplus()}}.



Referenced by \mbox{\hyperlink{__mstats__basic_8py_source_l01639}{scipy.\+stats.\+\_\+mstats\+\_\+basic.\+ks\+\_\+1samp()}}, and \mbox{\hyperlink{__stats__py_8py_source_l08908}{kstest()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a171a6b4426dbc47cdb093d05f3f1b969}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!ks\_2samp@{ks\_2samp}}
\index{ks\_2samp@{ks\_2samp}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{ks\_2samp()}{ks\_2samp()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a171a6b4426dbc47cdb093d05f3f1b969} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+ks\+\_\+2samp (\begin{DoxyParamCaption}\item[{}]{data1}{, }\item[{}]{data2}{, }\item[{}]{alternative}{ = {\ttfamily \textquotesingle{}two-\/sided\textquotesingle{}}, }\item[{}]{method}{ = {\ttfamily \textquotesingle{}auto\textquotesingle{}}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Performs the two-sample Kolmogorov-Smirnov test for goodness of fit.

This test compares the underlying continuous distributions F(x) and G(x)
of two independent samples.  See Notes for a description of the available
null and alternative hypotheses.

Parameters
----------
data1, data2 : array_like, 1-Dimensional
    Two arrays of sample observations assumed to be drawn from a continuous
    distribution, sample sizes can be different.
alternative : {'two-sided', 'less', 'greater'}, optional
    Defines the null and alternative hypotheses. Default is 'two-sided'.
    Please see explanations in the Notes below.
method : {'auto', 'exact', 'asymp'}, optional
    Defines the method used for calculating the p-value.
    The following options are available (default is 'auto'):

      * 'auto' : use 'exact' for small size arrays, 'asymp' for large
      * 'exact' : use exact distribution of test statistic
      * 'asymp' : use asymptotic distribution of test statistic

Returns
-------
res: KstestResult
    An object containing attributes:

    statistic : float
        KS test statistic.
    pvalue : float
        One-tailed or two-tailed p-value.
    statistic_location : float
        Value from `data1` or `data2` corresponding with the KS statistic;
        i.e., the distance between the empirical distribution functions is
        measured at this observation.
    statistic_sign : int
        +1 if the empirical distribution function of `data1` exceeds
        the empirical distribution function of `data2` at
        `statistic_location`, otherwise -1.

See Also
--------
kstest, ks_1samp, epps_singleton_2samp, anderson_ksamp

Notes
-----
There are three options for the null and corresponding alternative
hypothesis that can be selected using the `alternative` parameter.

- `less`: The null hypothesis is that F(x) >= G(x) for all x; the
  alternative is that F(x) < G(x) for at least one x. The statistic
  is the magnitude of the minimum (most negative) difference between the
  empirical distribution functions of the samples.

- `greater`: The null hypothesis is that F(x) <= G(x) for all x; the
  alternative is that F(x) > G(x) for at least one x. The statistic
  is the maximum (most positive) difference between the empirical
  distribution functions of the samples.

- `two-sided`: The null hypothesis is that the two distributions are
  identical, F(x)=G(x) for all x; the alternative is that they are not
  identical. The statistic is the maximum absolute difference between the
  empirical distribution functions of the samples.

Note that the alternative hypotheses describe the *CDFs* of the
underlying distributions, not the observed values of the data. For example,
suppose x1 ~ F and x2 ~ G. If F(x) > G(x) for all x, the values in
x1 tend to be less than those in x2.

If the KS statistic is large, then the p-value will be small, and this may
be taken as evidence against the null hypothesis in favor of the
alternative.

If ``method='exact'``, `ks_2samp` attempts to compute an exact p-value,
that is, the probability under the null hypothesis of obtaining a test
statistic value as extreme as the value computed from the data.
If ``method='asymp'``, the asymptotic Kolmogorov-Smirnov distribution is
used to compute an approximate p-value.
If ``method='auto'``, an exact p-value computation is attempted if both
sample sizes are less than 10000; otherwise, the asymptotic method is used.
In any case, if an exact p-value calculation is attempted and fails, a
warning will be emitted, and the asymptotic p-value will be returned.

The 'two-sided' 'exact' computation computes the complementary probability
and then subtracts from 1.  As such, the minimum probability it can return
is about 1e-16.  While the algorithm itself is exact, numerical
errors may accumulate for large sample sizes.   It is most suited to
situations in which one of the sample sizes is only a few thousand.

We generally follow Hodges' treatment of Drion/Gnedenko/Korolyuk [1]_.

References
----------
.. [1] Hodges, J.L. Jr.,  "The Significance Probability of the Smirnov
       Two-Sample Test," Arkiv fiur Matematik, 3, No. 43 (1958), 469-86.

Examples
--------
Suppose we wish to test the null hypothesis that two samples were drawn
from the same distribution.
We choose a confidence level of 95%; that is, we will reject the null
hypothesis in favor of the alternative if the p-value is less than 0.05.

If the first sample were drawn from a uniform distribution and the second
were drawn from the standard normal, we would expect the null hypothesis
to be rejected.

>>> import numpy as np
>>> from scipy import stats
>>> rng = np.random.default_rng()
>>> sample1 = stats.uniform.rvs(size=100, random_state=rng)
>>> sample2 = stats.norm.rvs(size=110, random_state=rng)
>>> stats.ks_2samp(sample1, sample2)
KstestResult(statistic=0.5454545454545454, pvalue=7.37417839555191e-15)

Indeed, the p-value is lower than our threshold of 0.05, so we reject the
null hypothesis in favor of the default "two-sided" alternative: the data
were *not* drawn from the same distribution.

When both samples are drawn from the same distribution, we expect the data
to be consistent with the null hypothesis most of the time.

>>> sample1 = stats.norm.rvs(size=105, random_state=rng)
>>> sample2 = stats.norm.rvs(size=95, random_state=rng)
>>> stats.ks_2samp(sample1, sample2)
KstestResult(statistic=0.10927318295739348, pvalue=0.5438289009927495)

As expected, the p-value of 0.54 is not below our threshold of 0.05, so
we cannot reject the null hypothesis.

Suppose, however, that the first sample were drawn from
a normal distribution shifted toward greater values. In this case,
the cumulative density function (CDF) of the underlying distribution tends
to be *less* than the CDF underlying the second sample. Therefore, we would
expect the null hypothesis to be rejected with ``alternative='less'``:

>>> sample1 = stats.norm.rvs(size=105, loc=0.5, random_state=rng)
>>> stats.ks_2samp(sample1, sample2, alternative='less')
KstestResult(statistic=0.4055137844611529, pvalue=3.5474563068855554e-08)

and indeed, with p-value smaller than our threshold, we reject the null
hypothesis in favor of the alternative.
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l08645}{8645}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l08596}{\+\_\+attempt\+\_\+exact\+\_\+2kssamp()}}.



Referenced by \mbox{\hyperlink{__mstats__basic_8py_source_l01679}{scipy.\+stats.\+\_\+mstats\+\_\+basic.\+ks\+\_\+2samp()}}, and \mbox{\hyperlink{__stats__py_8py_source_l08908}{kstest()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a918a98d18b72d14b9f81de22e21faa56}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!kstest@{kstest}}
\index{kstest@{kstest}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{kstest()}{kstest()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a918a98d18b72d14b9f81de22e21faa56} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+kstest (\begin{DoxyParamCaption}\item[{}]{rvs}{, }\item[{}]{cdf}{, }\item[{}]{args}{ = {\ttfamily ()}, }\item[{}]{N}{ = {\ttfamily 20}, }\item[{}]{alternative}{ = {\ttfamily \textquotesingle{}two-\/sided\textquotesingle{}}, }\item[{}]{method}{ = {\ttfamily \textquotesingle{}auto\textquotesingle{}}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Performs the (one-sample or two-sample) Kolmogorov-Smirnov test for
goodness of fit.

The one-sample test compares the underlying distribution F(x) of a sample
against a given distribution G(x). The two-sample test compares the
underlying distributions of two independent samples. Both tests are valid
only for continuous distributions.

Parameters
----------
rvs : str, array_like, or callable
    If an array, it should be a 1-D array of observations of random
    variables.
    If a callable, it should be a function to generate random variables;
    it is required to have a keyword argument `size`.
    If a string, it should be the name of a distribution in `scipy.stats`,
    which will be used to generate random variables.
cdf : str, array_like or callable
    If array_like, it should be a 1-D array of observations of random
    variables, and the two-sample test is performed
    (and rvs must be array_like).
    If a callable, that callable is used to calculate the cdf.
    If a string, it should be the name of a distribution in `scipy.stats`,
    which will be used as the cdf function.
args : tuple, sequence, optional
    Distribution parameters, used if `rvs` or `cdf` are strings or
    callables.
N : int, optional
    Sample size if `rvs` is string or callable.  Default is 20.
alternative : {'two-sided', 'less', 'greater'}, optional
    Defines the null and alternative hypotheses. Default is 'two-sided'.
    Please see explanations in the Notes below.
method : {'auto', 'exact', 'approx', 'asymp'}, optional
    Defines the distribution used for calculating the p-value.
    The following options are available (default is 'auto'):

      * 'auto' : selects one of the other options.
      * 'exact' : uses the exact distribution of test statistic.
      * 'approx' : approximates the two-sided probability with twice the
        one-sided probability
      * 'asymp': uses asymptotic distribution of test statistic

Returns
-------
res: KstestResult
    An object containing attributes:

    statistic : float
        KS test statistic, either D+, D-, or D (the maximum of the two)
    pvalue : float
        One-tailed or two-tailed p-value.
    statistic_location : float
        In a one-sample test, this is the value of `rvs`
        corresponding with the KS statistic; i.e., the distance between
        the empirical distribution function and the hypothesized cumulative
        distribution function is measured at this observation.

        In a two-sample test, this is the value from `rvs` or `cdf`
        corresponding with the KS statistic; i.e., the distance between
        the empirical distribution functions is measured at this
        observation.
    statistic_sign : int
        In a one-sample test, this is +1 if the KS statistic is the
        maximum positive difference between the empirical distribution
        function and the hypothesized cumulative distribution function
        (D+); it is -1 if the KS statistic is the maximum negative
        difference (D-).

        In a two-sample test, this is +1 if the empirical distribution
        function of `rvs` exceeds the empirical distribution
        function of `cdf` at `statistic_location`, otherwise -1.

See Also
--------
ks_1samp, ks_2samp

Notes
-----
There are three options for the null and corresponding alternative
hypothesis that can be selected using the `alternative` parameter.

- `two-sided`: The null hypothesis is that the two distributions are
  identical, F(x)=G(x) for all x; the alternative is that they are not
  identical.

- `less`: The null hypothesis is that F(x) >= G(x) for all x; the
  alternative is that F(x) < G(x) for at least one x.

- `greater`: The null hypothesis is that F(x) <= G(x) for all x; the
  alternative is that F(x) > G(x) for at least one x.

Note that the alternative hypotheses describe the *CDFs* of the
underlying distributions, not the observed values. For example,
suppose x1 ~ F and x2 ~ G. If F(x) > G(x) for all x, the values in
x1 tend to be less than those in x2.


Examples
--------
Suppose we wish to test the null hypothesis that a sample is distributed
according to the standard normal.
We choose a confidence level of 95%; that is, we will reject the null
hypothesis in favor of the alternative if the p-value is less than 0.05.

When testing uniformly distributed data, we would expect the
null hypothesis to be rejected.

>>> import numpy as np
>>> from scipy import stats
>>> rng = np.random.default_rng()
>>> stats.kstest(stats.uniform.rvs(size=100, random_state=rng),
...              stats.norm.cdf)
KstestResult(statistic=0.5001899973268688, pvalue=1.1616392184763533e-23)

Indeed, the p-value is lower than our threshold of 0.05, so we reject the
null hypothesis in favor of the default "two-sided" alternative: the data
are *not* distributed according to the standard normal.

When testing random variates from the standard normal distribution, we
expect the data to be consistent with the null hypothesis most of the time.

>>> x = stats.norm.rvs(size=100, random_state=rng)
>>> stats.kstest(x, stats.norm.cdf)
KstestResult(statistic=0.05345882212970396, pvalue=0.9227159037744717)

As expected, the p-value of 0.92 is not below our threshold of 0.05, so
we cannot reject the null hypothesis.

Suppose, however, that the random variates are distributed according to
a normal distribution that is shifted toward greater values. In this case,
the cumulative density function (CDF) of the underlying distribution tends
to be *less* than the CDF of the standard normal. Therefore, we would
expect the null hypothesis to be rejected with ``alternative='less'``:

>>> x = stats.norm.rvs(size=100, loc=0.5, random_state=rng)
>>> stats.kstest(x, stats.norm.cdf, alternative='less')
KstestResult(statistic=0.17482387821055168, pvalue=0.001913921057766743)

and indeed, with p-value smaller than our threshold, we reject the null
hypothesis in favor of the alternative.

For convenience, the previous test can be performed using the name of the
distribution as the second argument.

>>> stats.kstest(x, "norm", alternative='less')
KstestResult(statistic=0.17482387821055168, pvalue=0.001913921057766743)

The examples above have all been one-sample tests identical to those
performed by `ks_1samp`. Note that `kstest` can also perform two-sample
tests identical to those performed by `ks_2samp`. For example, when two
samples are drawn from the same distribution, we expect the data to be
consistent with the null hypothesis most of the time.

>>> sample1 = stats.laplace.rvs(size=105, random_state=rng)
>>> sample2 = stats.laplace.rvs(size=95, random_state=rng)
>>> stats.kstest(sample1, sample2)
KstestResult(statistic=0.11779448621553884, pvalue=0.4494256912629795)

As expected, the p-value of 0.45 is not below our threshold of 0.05, so
we cannot reject the null hypothesis.
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l08908}{8908}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l08310}{ks\+\_\+1samp()}}, and \mbox{\hyperlink{__stats__py_8py_source_l08645}{ks\+\_\+2samp()}}.



Referenced by \mbox{\hyperlink{__mstats__basic_8py_source_l01723}{scipy.\+stats.\+\_\+mstats\+\_\+basic.\+kstest()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a9fd18adf3aba669d3c24c5d052097f73}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!kurtosis@{kurtosis}}
\index{kurtosis@{kurtosis}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{kurtosis()}{kurtosis()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a9fd18adf3aba669d3c24c5d052097f73} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+kurtosis (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{axis}{ = {\ttfamily 0}, }\item[{}]{fisher}{ = {\ttfamily \mbox{\hyperlink{classTrue}{True}}}, }\item[{}]{bias}{ = {\ttfamily \mbox{\hyperlink{classTrue}{True}}}, }\item[{}]{nan\+\_\+policy}{ = {\ttfamily \textquotesingle{}propagate\textquotesingle{}}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Compute the kurtosis (Fisher or Pearson) of a dataset.

Kurtosis is the fourth central moment divided by the square of the
variance. If Fisher's definition is used, then 3.0 is subtracted from
the result to give 0.0 for a normal distribution.

If bias is False then the kurtosis is calculated using k statistics to
eliminate bias coming from biased moment estimators

Use `kurtosistest` to see if result is close enough to normal.

Parameters
----------
a : array
    Data for which the kurtosis is calculated.
axis : int or None, optional
    Axis along which the kurtosis is calculated. Default is 0.
    If None, compute over the whole array `a`.
fisher : bool, optional
    If True, Fisher's definition is used (normal ==> 0.0). If False,
    Pearson's definition is used (normal ==> 3.0).
bias : bool, optional
    If False, then the calculations are corrected for statistical bias.
nan_policy : {'propagate', 'raise', 'omit'}, optional
    Defines how to handle when input contains nan. 'propagate' returns nan,
    'raise' throws an error, 'omit' performs the calculations ignoring nan
    values. Default is 'propagate'.

Returns
-------
kurtosis : array
    The kurtosis of values along an axis, returning NaN where all values
    are equal.

References
----------
.. [1] Zwillinger, D. and Kokoska, S. (2000). CRC Standard
   Probability and Statistics Tables and Formulae. Chapman & Hall: New
   York. 2000.

Examples
--------
In Fisher's definition, the kurtosis of the normal distribution is zero.
In the following example, the kurtosis is close to zero, because it was
calculated from the dataset, not from the continuous distribution.

>>> import numpy as np
>>> from scipy.stats import norm, kurtosis
>>> data = norm.rvs(size=1000, random_state=3)
>>> kurtosis(data)
-0.06928694200380558

The distribution with a higher kurtosis has a heavier tail.
The zero valued kurtosis of the normal distribution in Fisher's definition
can serve as a reference point.

>>> import matplotlib.pyplot as plt
>>> import scipy.stats as stats
>>> from scipy.stats import kurtosis

>>> x = np.linspace(-5, 5, 100)
>>> ax = plt.subplot()
>>> distnames = ['laplace', 'norm', 'uniform']

>>> for distname in distnames:
...     if distname == 'uniform':
...         dist = getattr(stats, distname)(loc=-2, scale=4)
...     else:
...         dist = getattr(stats, distname)
...     data = dist.rvs(size=1000)
...     kur = kurtosis(data, fisher=True)
...     y = dist.pdf(x)
...     ax.plot(x, y, label="{}, {}".format(distname, round(kur, 3)))
...     ax.legend()

The Laplace distribution has a heavier tail than the normal distribution.
The uniform distribution (which has negative kurtosis) has the thinnest
tail.
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l01213}{1213}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l01326}{describe()}}, \mbox{\hyperlink{__stats__py_8py_source_l02000}{jarque\+\_\+bera()}}, and \mbox{\hyperlink{__stats__py_8py_source_l01627}{kurtosistest()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a8afd0b93401a2e4378a7f325e2b53cf0}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!kurtosistest@{kurtosistest}}
\index{kurtosistest@{kurtosistest}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{kurtosistest()}{kurtosistest()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a8afd0b93401a2e4378a7f325e2b53cf0} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+kurtosistest (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{axis}{ = {\ttfamily 0}, }\item[{}]{nan\+\_\+policy}{ = {\ttfamily \textquotesingle{}propagate\textquotesingle{}}, }\item[{}]{alternative}{ = {\ttfamily \textquotesingle{}two-\/sided\textquotesingle{}}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Test whether a dataset has normal kurtosis.

This function tests the null hypothesis that the kurtosis
of the population from which the sample was drawn is that
of the normal distribution.

Parameters
----------
a : array
Array of the sample data.
axis : int or None, optional
Axis along which to compute test. Default is 0. If None,
compute over the whole array `a`.
nan_policy : {'propagate', 'raise', 'omit'}, optional
Defines how to handle when input contains nan.
The following options are available (default is 'propagate'):

* 'propagate': returns nan
* 'raise': throws an error
* 'omit': performs the calculations ignoring nan values

alternative : {'two-sided', 'less', 'greater'}, optional
Defines the alternative hypothesis.
The following options are available (default is 'two-sided'):

* 'two-sided': the kurtosis of the distribution underlying the sample
is different from that of the normal distribution
* 'less': the kurtosis of the distribution underlying the sample
is less than that of the normal distribution
* 'greater': the kurtosis of the distribution underlying the sample
is greater than that of the normal distribution

.. versionadded:: 1.7.0

Returns
-------
statistic : float
The computed z-score for this test.
pvalue : float
The p-value for the hypothesis test.

Notes
-----
Valid only for n>20. This function uses the method described in [1]_.

References
----------
.. [1] see e.g. F. J. Anscombe, W. J. Glynn, "Distribution of the kurtosis
statistic b2 for normal samples", Biometrika, vol. 70, pp. 227-234, 1983.
.. [2] Shapiro, S. S., & Wilk, M. B. (1965). An analysis of variance test
for normality (complete samples). Biometrika, 52(3/4), 591-611.
.. [3] B. Phipson and G. K. Smyth. "Permutation P-values Should Never Be
Zero: Calculating Exact P-values When Permutations Are Randomly
Drawn." Statistical Applications in Genetics and Molecular Biology
9.1 (2010).
.. [4] Panagiotakos, D. B. (2008). The value of p-value in biomedical
research. The open cardiovascular medicine journal, 2, 97.

Examples
--------
Suppose we wish to infer from measurements whether the weights of adult
human males in a medical study are not normally distributed [2]_.
The weights (lbs) are recorded in the array ``x`` below.

>>> import numpy as np
>>> x = np.array([148, 154, 158, 160, 161, 162, 166, 170, 182, 195, 236])

The kurtosis test from [1]_ begins by computing a statistic based on the
sample (excess/Fisher) kurtosis.

>>> from scipy import stats
>>> res = stats.kurtosistest(x)
>>> res.statistic
2.3048235214240873

(The test warns that our sample has too few observations to perform the
test. We'll return to this at the end of the example.)
Because normal distributions have zero excess kurtosis (by definition),
the magnitude of this statistic tends to be low for samples drawn from a
normal distribution.

The test is performed by comparing the observed value of the
statistic against the null distribution: the distribution of statistic
values derived under the null hypothesis that the weights were drawn from
a normal distribution.

For this test, the null distribution of the statistic for very large
samples is the standard normal distribution.

>>> import matplotlib.pyplot as plt
>>> dist = stats.norm()
>>> kt_val = np.linspace(-5, 5, 100)
>>> pdf = dist.pdf(kt_val)
>>> fig, ax = plt.subplots(figsize=(8, 5))
>>> def kt_plot(ax):  # we'll re-use this
...     ax.plot(kt_val, pdf)
...     ax.set_title("Kurtosis Test Null Distribution")
...     ax.set_xlabel("statistic")
...     ax.set_ylabel("probability density")
>>> kt_plot(ax)
>>> plt.show()

The comparison is quantified by the p-value: the proportion of values in
the null distribution as extreme or more extreme than the observed
value of the statistic. In a two-sided test in which the statistic is
positive, elements of the null distribution greater than the observed
statistic and elements of the null distribution less than the negative of
the observed statistic are both considered "more extreme".

>>> fig, ax = plt.subplots(figsize=(8, 5))
>>> kt_plot(ax)
>>> pvalue = dist.cdf(-res.statistic) + dist.sf(res.statistic)
>>> annotation = (f'p-value={pvalue:.3f}\n(shaded area)')
>>> props = dict(facecolor='black', width=1, headwidth=5, headlength=8)
>>> _ = ax.annotate(annotation, (3, 0.005), (3.25, 0.02), arrowprops=props)
>>> i = kt_val >= res.statistic
>>> ax.fill_between(kt_val[i], y1=0, y2=pdf[i], color='C0')
>>> i = kt_val <= -res.statistic
>>> ax.fill_between(kt_val[i], y1=0, y2=pdf[i], color='C0')
>>> ax.set_xlim(-5, 5)
>>> ax.set_ylim(0, 0.1)
>>> plt.show()
>>> res.pvalue
0.0211764592113868

If the p-value is "small" - that is, if there is a low probability of
sampling data from a normally distributed population that produces such an
extreme value of the statistic - this may be taken as evidence against
the null hypothesis in favor of the alternative: the weights were not
drawn from a normal distribution. Note that:

- The inverse is not true; that is, the test is not used to provide
evidence for the null hypothesis.
- The threshold for values that will be considered "small" is a choice that
should be made before the data is analyzed [3]_ with consideration of the
risks of both false positives (incorrectly rejecting the null hypothesis)
and false negatives (failure to reject a false null hypothesis).

Note that the standard normal distribution provides an asymptotic
approximation of the null distribution; it is only accurate for samples
with many observations. This is the reason we received a warning at the
beginning of the example; our sample is quite small. In this case,
`scipy.stats.monte_carlo_test` may provide a more accurate, albeit
stochastic, approximation of the exact p-value.

>>> def statistic(x, axis):
...     # get just the skewtest statistic; ignore the p-value
...     return stats.kurtosistest(x, axis=axis).statistic
>>> res = stats.monte_carlo_test(x, stats.norm.rvs, statistic)
>>> fig, ax = plt.subplots(figsize=(8, 5))
>>> kt_plot(ax)
>>> ax.hist(res.null_distribution, np.linspace(-5, 5, 50),
...         density=True)
>>> ax.legend(['aymptotic approximation\n(many observations)',
...            'Monte Carlo approximation\n(11 observations)'])
>>> plt.show()
>>> res.pvalue
0.0272  # may vary

Furthermore, despite their stochastic nature, p-values computed in this way
can be used to exactly control the rate of false rejections of the null
hypothesis [4]_.
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l01627}{1627}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l01415}{\+\_\+normtest\+\_\+finish()}}, and \mbox{\hyperlink{__stats__py_8py_source_l01213}{kurtosis()}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l01836}{normaltest()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a3965fb8fb12386cc993ba9ea132af562}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!median\_abs\_deviation@{median\_abs\_deviation}}
\index{median\_abs\_deviation@{median\_abs\_deviation}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{median\_abs\_deviation()}{median\_abs\_deviation()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a3965fb8fb12386cc993ba9ea132af562} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+median\+\_\+abs\+\_\+deviation (\begin{DoxyParamCaption}\item[{}]{x}{, }\item[{}]{axis}{ = {\ttfamily 0}, }\item[{}]{center}{ = {\ttfamily np.median}, }\item[{}]{scale}{ = {\ttfamily 1.0}, }\item[{}]{nan\+\_\+policy}{ = {\ttfamily \textquotesingle{}propagate\textquotesingle{}}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Compute the median absolute deviation of the data along the given axis.

The median absolute deviation (MAD, [1]_) computes the median over the
absolute deviations from the median. It is a measure of dispersion
similar to the standard deviation but more robust to outliers [2]_.

The MAD of an empty array is ``np.nan``.

.. versionadded:: 1.5.0

Parameters
----------
x : array_like
    Input array or object that can be converted to an array.
axis : int or None, optional
    Axis along which the range is computed. Default is 0. If None, compute
    the MAD over the entire array.
center : callable, optional
    A function that will return the central value. The default is to use
    np.median. Any user defined function used will need to have the
    function signature ``func(arr, axis)``.
scale : scalar or str, optional
    The numerical value of scale will be divided out of the final
    result. The default is 1.0. The string "normal" is also accepted,
    and results in `scale` being the inverse of the standard normal
    quantile function at 0.75, which is approximately 0.67449.
    Array-like scale is also allowed, as long as it broadcasts correctly
    to the output such that ``out / scale`` is a valid operation. The
    output dimensions depend on the input array, `x`, and the `axis`
    argument.
nan_policy : {'propagate', 'raise', 'omit'}, optional
    Defines how to handle when input contains nan.
    The following options are available (default is 'propagate'):

    * 'propagate': returns nan
    * 'raise': throws an error
    * 'omit': performs the calculations ignoring nan values

Returns
-------
mad : scalar or ndarray
    If ``axis=None``, a scalar is returned. If the input contains
    integers or floats of smaller precision than ``np.float64``, then the
    output data-type is ``np.float64``. Otherwise, the output data-type is
    the same as that of the input.

See Also
--------
numpy.std, numpy.var, numpy.median, scipy.stats.iqr, scipy.stats.tmean,
scipy.stats.tstd, scipy.stats.tvar

Notes
-----
The `center` argument only affects the calculation of the central value
around which the MAD is calculated. That is, passing in ``center=np.mean``
will calculate the MAD around the mean - it will not calculate the *mean*
absolute deviation.

The input array may contain `inf`, but if `center` returns `inf`, the
corresponding MAD for that data will be `nan`.

References
----------
.. [1] "Median absolute deviation",
       https://en.wikipedia.org/wiki/Median_absolute_deviation
.. [2] "Robust measures of scale",
       https://en.wikipedia.org/wiki/Robust_measures_of_scale

Examples
--------
When comparing the behavior of `median_abs_deviation` with ``np.std``,
the latter is affected when we change a single value of an array to have an
outlier value while the MAD hardly changes:

>>> import numpy as np
>>> from scipy import stats
>>> x = stats.norm.rvs(size=100, scale=1, random_state=123456)
>>> x.std()
0.9973906394005013
>>> stats.median_abs_deviation(x)
0.82832610097857
>>> x[0] = 345.6
>>> x.std()
34.42304872314415
>>> stats.median_abs_deviation(x)
0.8323442311590675

Axis handling example:

>>> x = np.array([[10, 7, 4], [3, 2, 1]])
>>> x
array([[10,  7,  4],
       [ 3,  2,  1]])
>>> stats.median_abs_deviation(x)
array([3.5, 2.5, 1.5])
>>> stats.median_abs_deviation(x, axis=None)
2.0

Scale normal example:

>>> x = stats.norm.rvs(size=1000000, scale=2, random_state=123456)
>>> stats.median_abs_deviation(x)
1.3487398527041636
>>> stats.median_abs_deviation(x, scale='normal')
1.9996446978061115
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l03486}{3486}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a8c1c3534a15de7f93b96291982b74ac2}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!mode@{mode}}
\index{mode@{mode}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{mode()}{mode()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a8c1c3534a15de7f93b96291982b74ac2} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+mode (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{axis}{ = {\ttfamily 0}, }\item[{}]{nan\+\_\+policy}{ = {\ttfamily \textquotesingle{}propagate\textquotesingle{}}, }\item[{}]{keepdims}{ = {\ttfamily False}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Return an array of the modal (most common) value in the passed array.

If there is more than one such value, only one is returned.
The bin-count for the modal bins is also returned.

Parameters
----------
a : array_like
Numeric, n-dimensional array of which to find mode(s).
axis : int or None, optional
Axis along which to operate. Default is 0. If None, compute over
the whole array `a`.
nan_policy : {'propagate', 'raise', 'omit'}, optional
Defines how to handle when input contains nan.
The following options are available (default is 'propagate'):

* 'propagate': treats nan as it would treat any other value
* 'raise': throws an error
* 'omit': performs the calculations ignoring nan values
keepdims : bool, optional
If set to ``False``, the `axis` over which the statistic is taken
is consumed (eliminated from the output array). If set to ``True``,
the `axis` is retained with size one, and the result will broadcast
correctly against the input array.

Returns
-------
mode : ndarray
Array of modal values.
count : ndarray
Array of counts for each mode.

Notes
-----
The mode  is calculated using `numpy.unique`.
In NumPy versions 1.21 and after, all NaNs - even those with different
binary representations - are treated as equivalent and counted as separate
instances of the same value.

By convention, the mode of an empty array is NaN, and the associated count
is zero.

Examples
--------
>>> import numpy as np
>>> a = np.array([[3, 0, 3, 7],
...               [3, 2, 6, 2],
...               [1, 7, 2, 8],
...               [3, 0, 6, 1],
...               [3, 2, 5, 5]])
>>> from scipy import stats
>>> stats.mode(a, keepdims=True)
ModeResult(mode=array([[3, 0, 6, 1]]), count=array([[4, 2, 2, 1]]))

To get mode of whole array, specify ``axis=None``:

>>> stats.mode(a, axis=None, keepdims=True)
ModeResult(mode=[[3]], count=[[5]])
>>> stats.mode(a, axis=None, keepdims=False)
ModeResult(mode=3, count=5)
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l00442}{442}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a325ff018bf983292c0e20569bb6cb81f}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!moment@{moment}}
\index{moment@{moment}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{moment()}{moment()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a325ff018bf983292c0e20569bb6cb81f} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+moment (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{moment}{ = {\ttfamily 1}, }\item[{}]{axis}{ = {\ttfamily 0}, }\item[{}]{nan\+\_\+policy}{ = {\ttfamily \textquotesingle{}propagate\textquotesingle{}}, }\item[{\texorpdfstring{$\ast$}{*}}]{}{, }\item[{}]{center}{ = {\ttfamily None}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Calculate the nth moment about the mean for a sample.

A moment is a specific quantitative measure of the shape of a set of
points. It is often used to calculate coefficients of skewness and kurtosis
due to its close relationship with them.

Parameters
----------
a : array_like
Input array.
moment : int or array_like of ints, optional
Order of central moment that is returned. Default is 1.
axis : int or None, optional
Axis along which the central moment is computed. Default is 0.
If None, compute over the whole array `a`.
nan_policy : {'propagate', 'raise', 'omit'}, optional
Defines how to handle when input contains nan.
The following options are available (default is 'propagate'):

* 'propagate': returns nan
* 'raise': throws an error
* 'omit': performs the calculations ignoring nan values

center : float or None, optional
The point about which moments are taken. This can be the sample mean,
the origin, or any other be point. If `None` (default) compute the
center as the sample mean.

Returns
-------
n-th moment about the `center` : ndarray or float
The appropriate moment along the given axis or over all values if axis
is None. The denominator for the moment calculation is the number of
observations, no degrees of freedom correction is done.

See Also
--------
kurtosis, skew, describe

Notes
-----
The k-th moment of a data sample is:

.. math::

m_k = \frac{1}{n} \sum_{i = 1}^n (x_i - c)^k

Where `n` is the number of samples, and `c` is the center around which the
moment is calculated. This function uses exponentiation by squares [1]_ for
efficiency.

Note that, if `a` is an empty array (``a.size == 0``), array `moment` with
one element (`moment.size == 1`) is treated the same as scalar `moment`
(``np.isscalar(moment)``). This might produce arrays of unexpected shape.

References
----------
.. [1] https://eli.thegreenplace.net/2009/03/21/efficient-integer-exponentiation-algorithms

Examples
--------
>>> from scipy.stats import moment
>>> moment([1, 2, 3, 4, 5], moment=1)
0.0
>>> moment([1, 2, 3, 4, 5], moment=2)
2.0
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l00947}{947}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_aa5e6d4b63720d080eab70d14f43d74cd}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!multiscale\_graphcorr@{multiscale\_graphcorr}}
\index{multiscale\_graphcorr@{multiscale\_graphcorr}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{multiscale\_graphcorr()}{multiscale\_graphcorr()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_aa5e6d4b63720d080eab70d14f43d74cd} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+multiscale\+\_\+graphcorr (\begin{DoxyParamCaption}\item[{}]{x}{, }\item[{}]{y}{, }\item[{}]{compute\+\_\+distance}{ = {\ttfamily \+\_\+euclidean\+\_\+dist}, }\item[{}]{reps}{ = {\ttfamily 1000}, }\item[{}]{workers}{ = {\ttfamily 1}, }\item[{}]{is\+\_\+twosamp}{ = {\ttfamily False}, }\item[{}]{random\+\_\+state}{ = {\ttfamily None}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Computes the Multiscale Graph Correlation (MGC) test statistic.

Specifically, for each point, MGC finds the :math:`k`-nearest neighbors for
one property (e.g. cloud density), and the :math:`l`-nearest neighbors for
the other property (e.g. grass wetness) [1]_. This pair :math:`(k, l)` is
called the "scale". A priori, however, it is not know which scales will be
most informative. So, MGC computes all distance pairs, and then efficiently
computes the distance correlations for all scales. The local correlations
illustrate which scales are relatively informative about the relationship.
The key, therefore, to successfully discover and decipher relationships
between disparate data modalities is to adaptively determine which scales
are the most informative, and the geometric implication for the most
informative scales. Doing so not only provides an estimate of whether the
modalities are related, but also provides insight into how the
determination was made. This is especially important in high-dimensional
data, where simple visualizations do not reveal relationships to the
unaided human eye. Characterizations of this implementation in particular
have been derived from and benchmarked within in [2]_.

Parameters
----------
x, y : ndarray
    If ``x`` and ``y`` have shapes ``(n, p)`` and ``(n, q)`` where `n` is
    the number of samples and `p` and `q` are the number of dimensions,
    then the MGC independence test will be run.  Alternatively, ``x`` and
    ``y`` can have shapes ``(n, n)`` if they are distance or similarity
    matrices, and ``compute_distance`` must be sent to ``None``. If ``x``
    and ``y`` have shapes ``(n, p)`` and ``(m, p)``, an unpaired
    two-sample MGC test will be run.
compute_distance : callable, optional
    A function that computes the distance or similarity among the samples
    within each data matrix. Set to ``None`` if ``x`` and ``y`` are
    already distance matrices. The default uses the euclidean norm metric.
    If you are calling a custom function, either create the distance
    matrix before-hand or create a function of the form
    ``compute_distance(x)`` where `x` is the data matrix for which
    pairwise distances are calculated.
reps : int, optional
    The number of replications used to estimate the null when using the
    permutation test. The default is ``1000``.
workers : int or map-like callable, optional
    If ``workers`` is an int the population is subdivided into ``workers``
    sections and evaluated in parallel (uses ``multiprocessing.Pool
    <multiprocessing>``). Supply ``-1`` to use all cores available to the
    Process. Alternatively supply a map-like callable, such as
    ``multiprocessing.Pool.map`` for evaluating the p-value in parallel.
    This evaluation is carried out as ``workers(func, iterable)``.
    Requires that `func` be pickleable. The default is ``1``.
is_twosamp : bool, optional
    If `True`, a two sample test will be run. If ``x`` and ``y`` have
    shapes ``(n, p)`` and ``(m, p)``, this optional will be overridden and
    set to ``True``. Set to ``True`` if ``x`` and ``y`` both have shapes
    ``(n, p)`` and a two sample test is desired. The default is ``False``.
    Note that this will not run if inputs are distance matrices.
random_state : {None, int, `numpy.random.Generator`,
                `numpy.random.RandomState`}, optional

    If `seed` is None (or `np.random`), the `numpy.random.RandomState`
    singleton is used.
    If `seed` is an int, a new ``RandomState`` instance is used,
    seeded with `seed`.
    If `seed` is already a ``Generator`` or ``RandomState`` instance then
    that instance is used.

Returns
-------
res : MGCResult
    An object containing attributes:

    statistic : float
        The sample MGC test statistic within `[-1, 1]`.
    pvalue : float
        The p-value obtained via permutation.
    mgc_dict : dict
        Contains additional useful results:

            - mgc_map : ndarray
                A 2D representation of the latent geometry of the
                relationship.
            - opt_scale : (int, int)
                The estimated optimal scale as a `(x, y)` pair.
            - null_dist : list
                The null distribution derived from the permuted matrices.

See Also
--------
pearsonr : Pearson correlation coefficient and p-value for testing
           non-correlation.
kendalltau : Calculates Kendall's tau.
spearmanr : Calculates a Spearman rank-order correlation coefficient.

Notes
-----
A description of the process of MGC and applications on neuroscience data
can be found in [1]_. It is performed using the following steps:

#. Two distance matrices :math:`D^X` and :math:`D^Y` are computed and
   modified to be mean zero columnwise. This results in two
   :math:`n \times n` distance matrices :math:`A` and :math:`B` (the
   centering and unbiased modification) [3]_.

#. For all values :math:`k` and :math:`l` from :math:`1, ..., n`,

   * The :math:`k`-nearest neighbor and :math:`l`-nearest neighbor graphs
     are calculated for each property. Here, :math:`G_k (i, j)` indicates
     the :math:`k`-smallest values of the :math:`i`-th row of :math:`A`
     and :math:`H_l (i, j)` indicates the :math:`l` smallested values of
     the :math:`i`-th row of :math:`B`

   * Let :math:`\circ` denotes the entry-wise matrix product, then local
     correlations are summed and normalized using the following statistic:

.. math::

    c^{kl} = \frac{\sum_{ij} A G_k B H_l}
                  {\sqrt{\sum_{ij} A^2 G_k \times \sum_{ij} B^2 H_l}}

#. The MGC test statistic is the smoothed optimal local correlation of
   :math:`\{ c^{kl} \}`. Denote the smoothing operation as :math:`R(\cdot)`
   (which essentially set all isolated large correlations) as 0 and
   connected large correlations the same as before, see [3]_.) MGC is,

.. math::

    MGC_n (x, y) = \max_{(k, l)} R \left(c^{kl} \left( x_n, y_n \right)
                                                \right)

The test statistic returns a value between :math:`(-1, 1)` since it is
normalized.

The p-value returned is calculated using a permutation test. This process
is completed by first randomly permuting :math:`y` to estimate the null
distribution and then calculating the probability of observing a test
statistic, under the null, at least as extreme as the observed test
statistic.

MGC requires at least 5 samples to run with reliable results. It can also
handle high-dimensional data sets.
In addition, by manipulating the input data matrices, the two-sample
testing problem can be reduced to the independence testing problem [4]_.
Given sample data :math:`U` and :math:`V` of sizes :math:`p \times n`
:math:`p \times m`, data matrix :math:`X` and :math:`Y` can be created as
follows:

.. math::

    X = [U | V] \in \mathcal{R}^{p \times (n + m)}
    Y = [0_{1 \times n} | 1_{1 \times m}] \in \mathcal{R}^{(n + m)}

Then, the MGC statistic can be calculated as normal. This methodology can
be extended to similar tests such as distance correlation [4]_.

.. versionadded:: 1.4.0

References
----------
.. [1] Vogelstein, J. T., Bridgeford, E. W., Wang, Q., Priebe, C. E.,
       Maggioni, M., & Shen, C. (2019). Discovering and deciphering
       relationships across disparate data modalities. ELife.
.. [2] Panda, S., Palaniappan, S., Xiong, J., Swaminathan, A.,
       Ramachandran, S., Bridgeford, E. W., ... Vogelstein, J. T. (2019).
       mgcpy: A Comprehensive High Dimensional Independence Testing Python
       Package. :arXiv:`1907.02088`
.. [3] Shen, C., Priebe, C.E., & Vogelstein, J. T. (2019). From distance
       correlation to multiscale graph correlation. Journal of the American
       Statistical Association.
.. [4] Shen, C. & Vogelstein, J. T. (2018). The Exact Equivalence of
       Distance and Kernel Methods for Hypothesis Testing.
       :arXiv:`1806.05514`

Examples
--------
>>> import numpy as np
>>> from scipy.stats import multiscale_graphcorr
>>> x = np.arange(100)
>>> y = x
>>> res = multiscale_graphcorr(x, y)
>>> res.statistic, res.pvalue
(1.0, 0.001)

To run an unpaired two-sample test,

>>> x = np.arange(100)
>>> y = np.arange(79)
>>> res = multiscale_graphcorr(x, y)
>>> res.statistic, res.pvalue  # doctest: +SKIP
(0.033258146255703246, 0.023)

or, if shape of the inputs are the same,

>>> x = np.arange(100)
>>> y = x
>>> res = multiscale_graphcorr(x, y, is_twosamp=True)
>>> res.statistic, res.pvalue  # doctest: +SKIP
(-0.008021809890200488, 1.0)
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l06224}{6224}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l06509}{\+\_\+mgc\+\_\+stat()}}, \mbox{\hyperlink{__stats__py_8py_source_l06161}{\+\_\+perm\+\_\+test()}}, and \mbox{\hyperlink{__stats__py_8py_source_l06654}{\+\_\+two\+\_\+sample\+\_\+transform()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a04cd8218975e527a625046de8cf8004d}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!normaltest@{normaltest}}
\index{normaltest@{normaltest}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{normaltest()}{normaltest()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a04cd8218975e527a625046de8cf8004d} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+normaltest (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{axis}{ = {\ttfamily 0}, }\item[{}]{nan\+\_\+policy}{ = {\ttfamily \textquotesingle{}propagate\textquotesingle{}}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Test whether a sample differs from a normal distribution.

This function tests the null hypothesis that a sample comes
from a normal distribution.  It is based on D'Agostino and
Pearson's [1]_, [2]_ test that combines skew and kurtosis to
produce an omnibus test of normality.

Parameters
----------
a : array_like
    The array containing the sample to be tested.
axis : int or None, optional
    Axis along which to compute test. Default is 0. If None,
    compute over the whole array `a`.
nan_policy : {'propagate', 'raise', 'omit'}, optional
    Defines how to handle when input contains nan.
    The following options are available (default is 'propagate'):

      * 'propagate': returns nan
      * 'raise': throws an error
      * 'omit': performs the calculations ignoring nan values

Returns
-------
statistic : float or array
    ``s^2 + k^2``, where ``s`` is the z-score returned by `skewtest` and
    ``k`` is the z-score returned by `kurtosistest`.
pvalue : float or array
   A 2-sided chi squared probability for the hypothesis test.

References
----------
.. [1] D'Agostino, R. B. (1971), "An omnibus test of normality for
       moderate and large sample size", Biometrika, 58, 341-348
.. [2] D'Agostino, R. and Pearson, E. S. (1973), "Tests for departure from
       normality", Biometrika, 60, 613-622
.. [3] Shapiro, S. S., & Wilk, M. B. (1965). An analysis of variance test
       for normality (complete samples). Biometrika, 52(3/4), 591-611.
.. [4] B. Phipson and G. K. Smyth. "Permutation P-values Should Never Be
       Zero: Calculating Exact P-values When Permutations Are Randomly
       Drawn." Statistical Applications in Genetics and Molecular Biology
       9.1 (2010).
.. [5] Panagiotakos, D. B. (2008). The value of p-value in biomedical
       research. The open cardiovascular medicine journal, 2, 97.

Examples
--------
Suppose we wish to infer from measurements whether the weights of adult
human males in a medical study are not normally distributed [3]_.
The weights (lbs) are recorded in the array ``x`` below.

>>> import numpy as np
>>> x = np.array([148, 154, 158, 160, 161, 162, 166, 170, 182, 195, 236])

The normality test of [1]_ and [2]_ begins by computing a statistic based
on the sample skewness and kurtosis.

>>> from scipy import stats
>>> res = stats.normaltest(x)
>>> res.statistic
13.034263121192582

(The test warns that our sample has too few observations to perform the
test. We'll return to this at the end of the example.)
Because the normal distribution has zero skewness and zero
("excess" or "Fisher") kurtosis, the value of this statistic tends to be
low for samples drawn from a normal distribution.

The test is performed by comparing the observed value of the statistic
against the null distribution: the distribution of statistic values derived
under the null hypothesis that the weights were drawn from a normal
distribution.
For this normality test, the null distribution for very large samples is
the chi-squared distribution with two degrees of freedom.

>>> import matplotlib.pyplot as plt
>>> dist = stats.chi2(df=2)
>>> stat_vals = np.linspace(0, 16, 100)
>>> pdf = dist.pdf(stat_vals)
>>> fig, ax = plt.subplots(figsize=(8, 5))
>>> def plot(ax):  # we'll re-use this
...     ax.plot(stat_vals, pdf)
...     ax.set_title("Normality Test Null Distribution")
...     ax.set_xlabel("statistic")
...     ax.set_ylabel("probability density")
>>> plot(ax)
>>> plt.show()

The comparison is quantified by the p-value: the proportion of values in
the null distribution greater than or equal to the observed value of the
statistic.

>>> fig, ax = plt.subplots(figsize=(8, 5))
>>> plot(ax)
>>> pvalue = dist.sf(res.statistic)
>>> annotation = (f'p-value={pvalue:.6f}\n(shaded area)')
>>> props = dict(facecolor='black', width=1, headwidth=5, headlength=8)
>>> _ = ax.annotate(annotation, (13.5, 5e-4), (14, 5e-3), arrowprops=props)
>>> i = stat_vals >= res.statistic  # index more extreme statistic values
>>> ax.fill_between(stat_vals[i], y1=0, y2=pdf[i])
>>> ax.set_xlim(8, 16)
>>> ax.set_ylim(0, 0.01)
>>> plt.show()
>>> res.pvalue
0.0014779023013100172

If the p-value is "small" - that is, if there is a low probability of
sampling data from a normally distributed population that produces such an
extreme value of the statistic - this may be taken as evidence against
the null hypothesis in favor of the alternative: the weights were not
drawn from a normal distribution. Note that:

- The inverse is not true; that is, the test is not used to provide
  evidence for the null hypothesis.
- The threshold for values that will be considered "small" is a choice that
  should be made before the data is analyzed [4]_ with consideration of the
  risks of both false positives (incorrectly rejecting the null hypothesis)
  and false negatives (failure to reject a false null hypothesis).

Note that the chi-squared distribution provides an asymptotic
approximation of the null distribution; it is only accurate for samples
with many observations. This is the reason we received a warning at the
beginning of the example; our sample is quite small. In this case,
`scipy.stats.monte_carlo_test` may provide a more accurate, albeit
stochastic, approximation of the exact p-value.

>>> def statistic(x, axis):
...     # Get only the `normaltest` statistic; ignore approximate p-value
...     return stats.normaltest(x, axis=axis).statistic
>>> res = stats.monte_carlo_test(x, stats.norm.rvs, statistic,
...                              alternative='greater')
>>> fig, ax = plt.subplots(figsize=(8, 5))
>>> plot(ax)
>>> ax.hist(res.null_distribution, np.linspace(0, 25, 50),
...         density=True)
>>> ax.legend(['aymptotic approximation (many observations)',
...            'Monte Carlo approximation (11 observations)'])
>>> ax.set_xlim(0, 14)
>>> plt.show()
>>> res.pvalue
0.0082  # may vary

Furthermore, despite their stochastic nature, p-values computed in this way
can be used to exactly control the rate of false rejections of the null
hypothesis [5]_.
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l01836}{1836}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l01627}{kurtosistest()}}, and \mbox{\hyperlink{__stats__py_8py_source_l01436}{skewtest()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_aa080af52b52bdc1e1998e5f15eef174d}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!obrientransform@{obrientransform}}
\index{obrientransform@{obrientransform}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{obrientransform()}{obrientransform()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_aa080af52b52bdc1e1998e5f15eef174d} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+obrientransform (\begin{DoxyParamCaption}\item[{\texorpdfstring{$\ast$}{*}}]{samples}{}\end{DoxyParamCaption})}



VARIABILITY FUNCTIONS \#. 

\begin{DoxyVerb}Compute the O'Brien transform on input data (any number of arrays).

Used to test for homogeneity of variance prior to running one-way stats.
Each array in ``*samples`` is one level of a factor.
If `f_oneway` is run on the transformed data and found significant,
the variances are unequal.  From Maxwell and Delaney [1]_, p.112.

Parameters
----------
sample1, sample2, ... : array_like
    Any number of arrays.

Returns
-------
obrientransform : ndarray
    Transformed data for use in an ANOVA.  The first dimension
    of the result corresponds to the sequence of transformed
    arrays.  If the arrays given are all 1-D of the same length,
    the return value is a 2-D array; otherwise it is a 1-D array
    of type object, with each element being an ndarray.

References
----------
.. [1] S. E. Maxwell and H. D. Delaney, "Designing Experiments and
       Analyzing Data: A Model Comparison Perspective", Wadsworth, 1990.

Examples
--------
We'll test the following data sets for differences in their variance.

>>> x = [10, 11, 13, 9, 7, 12, 12, 9, 10]
>>> y = [13, 21, 5, 10, 8, 14, 10, 12, 7, 15]

Apply the O'Brien transform to the data.

>>> from scipy.stats import obrientransform
>>> tx, ty = obrientransform(x, y)

Use `scipy.stats.f_oneway` to apply a one-way ANOVA test to the
transformed data.

>>> from scipy.stats import f_oneway
>>> F, p = f_oneway(tx, ty)
>>> p
0.1314139477040335

If we require that ``p < 0.05`` for significance, we cannot conclude
that the variances are different.
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l02704}{2704}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a921a917ca3f8ea0c04c6bc7f87d33999}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!pack\_TtestResult@{pack\_TtestResult}}
\index{pack\_TtestResult@{pack\_TtestResult}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{pack\_TtestResult()}{pack\_TtestResult()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a921a917ca3f8ea0c04c6bc7f87d33999} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+pack\+\_\+\+Ttest\+Result (\begin{DoxyParamCaption}\item[{}]{statistic}{, }\item[{}]{pvalue}{, }\item[{}]{df}{, }\item[{}]{alternative}{, }\item[{}]{standard\+\_\+error}{, }\item[{}]{estimate}{}\end{DoxyParamCaption})}



Definition at line \mbox{\hyperlink{__stats__py_8py_source_l06747}{6747}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a8b002f05757020441e68edf9441f3c7f}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!pearsonr@{pearsonr}}
\index{pearsonr@{pearsonr}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{pearsonr()}{pearsonr()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a8b002f05757020441e68edf9441f3c7f} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+pearsonr (\begin{DoxyParamCaption}\item[{}]{x}{, }\item[{}]{y}{, }\item[{\texorpdfstring{$\ast$}{*}}]{}{, }\item[{}]{alternative}{ = {\ttfamily \textquotesingle{}two-\/sided\textquotesingle{}}, }\item[{}]{method}{ = {\ttfamily None}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Pearson correlation coefficient and p-value for testing non-correlation.

The Pearson correlation coefficient [1]_ measures the linear relationship
between two datasets. Like other correlation
coefficients, this one varies between -1 and +1 with 0 implying no
correlation. Correlations of -1 or +1 imply an exact linear relationship.
Positive correlations imply that as x increases, so does y. Negative
correlations imply that as x increases, y decreases.

This function also performs a test of the null hypothesis that the
distributions underlying the samples are uncorrelated and normally
distributed. (See Kowalski [3]_
for a discussion of the effects of non-normality of the input on the
distribution of the correlation coefficient.)
The p-value roughly indicates the probability of an uncorrelated system
producing datasets that have a Pearson correlation at least as extreme
as the one computed from these datasets.

Parameters
----------
x : (N,) array_like
Input array.
y : (N,) array_like
Input array.
alternative : {'two-sided', 'greater', 'less'}, optional
Defines the alternative hypothesis. Default is 'two-sided'.
The following options are available:

* 'two-sided': the correlation is nonzero
* 'less': the correlation is negative (less than zero)
* 'greater':  the correlation is positive (greater than zero)

.. versionadded:: 1.9.0
method : ResamplingMethod, optional
Defines the method used to compute the p-value. If `method` is an
instance of `PermutationMethod`/`MonteCarloMethod`, the p-value is
computed using
`scipy.stats.permutation_test`/`scipy.stats.monte_carlo_test` with the
provided configuration options and other appropriate settings.
Otherwise, the p-value is computed as documented in the notes.

.. versionadded:: 1.11.0

Returns
-------
result : `~scipy.stats._result_classes.PearsonRResult`
An object with the following attributes:

statistic : float
    Pearson product-moment correlation coefficient.
pvalue : float
    The p-value associated with the chosen alternative.

The object has the following method:

confidence_interval(confidence_level, method)
    This computes the confidence interval of the correlation
    coefficient `statistic` for the given confidence level.
    The confidence interval is returned in a ``namedtuple`` with
    fields `low` and `high`. If `method` is not provided, the
    confidence interval is computed using the Fisher transformation
    [1]_. If `method` is an instance of `BootstrapMethod`, the
    confidence interval is computed using `scipy.stats.bootstrap` with
    the provided configuration options and other appropriate settings.
    In some cases, confidence limits may be NaN due to a degenerate
    resample, and this is typical for very small samples (~6
    observations).

Warns
-----
`~scipy.stats.ConstantInputWarning`
Raised if an input is a constant array.  The correlation coefficient
is not defined in this case, so ``np.nan`` is returned.

`~scipy.stats.NearConstantInputWarning`
Raised if an input is "nearly" constant.  The array ``x`` is considered
nearly constant if ``norm(x - mean(x)) < 1e-13 * abs(mean(x))``.
Numerical errors in the calculation ``x - mean(x)`` in this case might
result in an inaccurate calculation of r.

See Also
--------
spearmanr : Spearman rank-order correlation coefficient.
kendalltau : Kendall's tau, a correlation measure for ordinal data.

Notes
-----
The correlation coefficient is calculated as follows:

.. math::

r = \frac{\sum (x - m_x) (y - m_y)}
         {\sqrt{\sum (x - m_x)^2 \sum (y - m_y)^2}}

where :math:`m_x` is the mean of the vector x and :math:`m_y` is
the mean of the vector y.

Under the assumption that x and y are drawn from
independent normal distributions (so the population correlation coefficient
is 0), the probability density function of the sample correlation
coefficient r is ([1]_, [2]_):

.. math::
f(r) = \frac{{(1-r^2)}^{n/2-2}}{\mathrm{B}(\frac{1}{2},\frac{n}{2}-1)}

where n is the number of samples, and B is the beta function.  This
is sometimes referred to as the exact distribution of r.  This is
the distribution that is used in `pearsonr` to compute the p-value when
the `method` parameter is left at its default value (None).
The distribution is a beta distribution on the interval [-1, 1],
with equal shape parameters a = b = n/2 - 1.  In terms of SciPy's
implementation of the beta distribution, the distribution of r is::

dist = scipy.stats.beta(n/2 - 1, n/2 - 1, loc=-1, scale=2)

The default p-value returned by `pearsonr` is a two-sided p-value. For a
given sample with correlation coefficient r, the p-value is
the probability that abs(r') of a random sample x' and y' drawn from
the population with zero correlation would be greater than or equal
to abs(r). In terms of the object ``dist`` shown above, the p-value
for a given r and length n can be computed as::

p = 2*dist.cdf(-abs(r))

When n is 2, the above continuous distribution is not well-defined.
One can interpret the limit of the beta distribution as the shape
parameters a and b approach a = b = 0 as a discrete distribution with
equal probability masses at r = 1 and r = -1.  More directly, one
can observe that, given the data x = [x1, x2] and y = [y1, y2], and
assuming x1 != x2 and y1 != y2, the only possible values for r are 1
and -1.  Because abs(r') for any sample x' and y' with length 2 will
be 1, the two-sided p-value for a sample of length 2 is always 1.

For backwards compatibility, the object that is returned also behaves
like a tuple of length two that holds the statistic and the p-value.

References
----------
.. [1] "Pearson correlation coefficient", Wikipedia,
   https://en.wikipedia.org/wiki/Pearson_correlation_coefficient
.. [2] Student, "Probable error of a correlation coefficient",
   Biometrika, Volume 6, Issue 2-3, 1 September 1908, pp. 302-310.
.. [3] C. J. Kowalski, "On the Effects of Non-Normality on the Distribution
   of the Sample Product-Moment Correlation Coefficient"
   Journal of the Royal Statistical Society. Series C (Applied
   Statistics), Vol. 21, No. 1 (1972), pp. 1-12.

Examples
--------
>>> import numpy as np
>>> from scipy import stats
>>> x, y = [1, 2, 3, 4, 5, 6, 7], [10, 9, 2.5, 6, 4, 3, 2]
>>> res = stats.pearsonr(x, y)
>>> res
PearsonRResult(statistic=-0.828503883588428, pvalue=0.021280260007523286)

To perform an exact permutation version of the test:

>>> rng = np.random.default_rng(7796654889291491997)
>>> method = stats.PermutationMethod(n_resamples=np.inf, random_state=rng)
>>> stats.pearsonr(x, y, method=method)
PearsonRResult(statistic=-0.828503883588428, pvalue=0.028174603174603175)

To perform the test under the null hypothesis that the data were drawn from
*uniform* distributions:

>>> method = stats.MonteCarloMethod(rvs=(rng.uniform, rng.uniform))
>>> stats.pearsonr(x, y, method=method)
PearsonRResult(statistic=-0.828503883588428, pvalue=0.0188)

To produce an asymptotic 90% confidence interval:

>>> res.confidence_interval(confidence_level=0.9)
ConfidenceInterval(low=-0.9644331982722841, high=-0.3460237473272273)

And for a bootstrap confidence interval:

>>> method = stats.BootstrapMethod(method='BCa', random_state=rng)
>>> res.confidence_interval(confidence_level=0.9, method=method)
ConfidenceInterval(low=-0.9983163756488651, high=-0.22771001702132443)  # may vary

There is a linear dependence between x and y if y = a + b*x + e, where
a,b are constants and e is a random error term, assumed to be independent
of x. For simplicity, assume that x is standard normal, a=0, b=1 and let
e follow a normal distribution with mean zero and standard deviation s>0.

>>> rng = np.random.default_rng()
>>> s = 0.5
>>> x = stats.norm.rvs(size=500, random_state=rng)
>>> e = stats.norm.rvs(scale=s, size=500, random_state=rng)
>>> y = x + e
>>> stats.pearsonr(x, y).statistic
0.9001942438244763

This should be close to the exact value given by

>>> 1/np.sqrt(1 + s**2)
0.8944271909999159

For s=0.5, we observe a high level of correlation. In general, a large
variance of the noise reduces the correlation, while the correlation
approaches one as the variance of the error goes to zero.

It is important to keep in mind that no correlation does not imply
independence unless (x, y) is jointly normal. Correlation can even be zero
when there is a very simple dependence structure: if X follows a
standard normal distribution, let y = abs(x). Note that the correlation
between x and y is zero. Indeed, since the expectation of x is zero,
cov(x, y) = E[x*y]. By definition, this equals E[x*abs(x)] which is zero
by symmetry. The following lines of code illustrate this observation:

>>> y = np.abs(x)
>>> stats.pearsonr(x, y)
PearsonRResult(statistic=-0.05444919272687482, pvalue=0.22422294836207743)

A non-zero correlation coefficient can be misleading. For example, if X has
a standard normal distribution, define y = x if x < 0 and y = 0 otherwise.
A simple calculation shows that corr(x, y) = sqrt(2/Pi) = 0.797...,
implying a high level of correlation:

>>> y = np.where(x < 0, x, 0)
>>> stats.pearsonr(x, y)
PearsonRResult(statistic=0.861985781588, pvalue=4.813432002751103e-149)

This is unintuitive since there is no dependence of x and y if x is larger
than zero which happens in about half of the cases if we sample x and y.
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l04533}{4533}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l04533}{pearsonr()}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l04426}{\+\_\+pearsonr\+\_\+bootstrap\+\_\+ci()}}, \mbox{\hyperlink{__mstats__basic_8py_source_l00382}{scipy.\+stats.\+\_\+mstats\+\_\+basic.\+pearsonr()}}, \mbox{\hyperlink{__stats__py_8py_source_l04533}{pearsonr()}}, and \mbox{\hyperlink{__stats__py_8py_source_l05499}{pointbiserialr()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_af139493d2638b97ca3345692a7835fd4}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!percentileofscore@{percentileofscore}}
\index{percentileofscore@{percentileofscore}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{percentileofscore()}{percentileofscore()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_af139493d2638b97ca3345692a7835fd4} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+percentileofscore (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{score}{, }\item[{}]{kind}{ = {\ttfamily \textquotesingle{}rank\textquotesingle{}}, }\item[{}]{nan\+\_\+policy}{ = {\ttfamily \textquotesingle{}propagate\textquotesingle{}}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Compute the percentile rank of a score relative to a list of scores.

A `percentileofscore` of, for example, 80% means that 80% of the
scores in `a` are below the given score. In the case of gaps or
ties, the exact definition depends on the optional keyword, `kind`.

Parameters
----------
a : array_like
    Array to which `score` is compared.
score : array_like
    Scores to compute percentiles for.
kind : {'rank', 'weak', 'strict', 'mean'}, optional
    Specifies the interpretation of the resulting score.
    The following options are available (default is 'rank'):

      * 'rank': Average percentage ranking of score.  In case of multiple
        matches, average the percentage rankings of all matching scores.
      * 'weak': This kind corresponds to the definition of a cumulative
        distribution function.  A percentileofscore of 80% means that 80%
        of values are less than or equal to the provided score.
      * 'strict': Similar to "weak", except that only values that are
        strictly less than the given score are counted.
      * 'mean': The average of the "weak" and "strict" scores, often used
        in testing.  See https://en.wikipedia.org/wiki/Percentile_rank
nan_policy : {'propagate', 'raise', 'omit'}, optional
    Specifies how to treat `nan` values in `a`.
    The following options are available (default is 'propagate'):

      * 'propagate': returns nan (for each value in `score`).
      * 'raise': throws an error
      * 'omit': performs the calculations ignoring nan values

Returns
-------
pcos : float
    Percentile-position of score (0-100) relative to `a`.

See Also
--------
numpy.percentile
scipy.stats.scoreatpercentile, scipy.stats.rankdata

Examples
--------
Three-quarters of the given values lie below a given score:

>>> import numpy as np
>>> from scipy import stats
>>> stats.percentileofscore([1, 2, 3, 4], 3)
75.0

With multiple matches, note how the scores of the two matches, 0.6
and 0.8 respectively, are averaged:

>>> stats.percentileofscore([1, 2, 3, 3, 4], 3)
70.0

Only 2/5 values are strictly less than 3:

>>> stats.percentileofscore([1, 2, 3, 3, 4], 3, kind='strict')
40.0

But 4/5 values are less than or equal to 3:

>>> stats.percentileofscore([1, 2, 3, 3, 4], 3, kind='weak')
80.0

The average between the weak and the strict scores is:

>>> stats.percentileofscore([1, 2, 3, 3, 4], 3, kind='mean')
60.0

Score arrays (of any dimensionality) are supported:

>>> stats.percentileofscore([1, 2, 3, 3, 4], [2, 3])
array([40., 70.])

The inputs can be infinite:

>>> stats.percentileofscore([-np.inf, 0, 1, np.inf], [1, 2, np.inf])
array([75., 75., 100.])

If `a` is empty, then the resulting percentiles are all `nan`:

>>> stats.percentileofscore([], [1, 2])
array([nan, nan])
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l02292}{2292}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a055d012b1b109ae0f483c925f5d0e0b8}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!pmean@{pmean}}
\index{pmean@{pmean}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{pmean()}{pmean()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a055d012b1b109ae0f483c925f5d0e0b8} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+pmean (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{p}{, }\item[{\texorpdfstring{$\ast$}{*}}]{}{, }\item[{}]{axis}{ = {\ttfamily 0}, }\item[{}]{dtype}{ = {\ttfamily None}, }\item[{}]{weights}{ = {\ttfamily None}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Calculate the weighted power mean along the specified axis.

The weighted power mean of the array :math:`a_i` associated to weights
:math:`w_i` is:

.. math::

\left( \frac{ \sum_{i=1}^n w_i a_i^p }{ \sum_{i=1}^n w_i }
\right)^{ 1 / p } \, ,

and, with equal weights, it gives:

.. math::

\left( \frac{ 1 }{ n } \sum_{i=1}^n a_i^p \right)^{ 1 / p }  \, .

When ``p=0``, it returns the geometric mean.

This mean is also called generalized mean or Hlder mean, and must not be
confused with the Kolmogorov generalized mean, also called
quasi-arithmetic mean or generalized f-mean [3]_.

Parameters
----------
a : array_like
Input array, masked array or object that can be converted to an array.
p : int or float
Exponent.
axis : int or None, optional
Axis along which the power mean is computed. Default is 0.
If None, compute over the whole array `a`.
dtype : dtype, optional
Type of the returned array and of the accumulator in which the
elements are summed. If `dtype` is not specified, it defaults to the
dtype of `a`, unless `a` has an integer `dtype` with a precision less
than that of the default platform integer. In that case, the default
platform integer is used.
weights : array_like, optional
The weights array can either be 1-D (in which case its length must be
the size of `a` along the given `axis`) or of the same shape as `a`.
Default is None, which gives each value a weight of 1.0.

Returns
-------
pmean : ndarray, see `dtype` parameter above.
Output array containing the power mean values.

See Also
--------
numpy.average : Weighted average
gmean : Geometric mean
hmean : Harmonic mean

Notes
-----
The power mean is computed over a single dimension of the input
array, ``axis=0`` by default, or all values in the array if ``axis=None``.
float64 intermediate and return values are used for integer inputs.

.. versionadded:: 1.9

References
----------
.. [1] "Generalized Mean", *Wikipedia*,
https://en.wikipedia.org/wiki/Generalized_mean
.. [2] Norris, N., "Convexity properties of generalized mean value
functions", The Annals of Mathematical Statistics, vol. 8,
pp. 118-120, 1937
.. [3] Bullen, P.S., Handbook of Means and Their Inequalities, 2003

Examples
--------
>>> from scipy.stats import pmean, hmean, gmean
>>> pmean([1, 4], 1.3)
2.639372938300652
>>> pmean([1, 2, 3, 4, 5, 6, 7], 1.3)
4.157111214492084
>>> pmean([1, 4, 7], -2, weights=[3, 1, 3])
1.4969684896631954

For p=-1, power mean is equal to harmonic mean:

>>> pmean([1, 4, 7], -1, weights=[3, 1, 3])
1.9029126213592233
>>> hmean([1, 4, 7], weights=[3, 1, 3])
1.9029126213592233

For p=0, power mean is defined as the geometric mean:

>>> pmean([1, 4, 7], 0, weights=[3, 1, 3])
2.80668351922014
>>> gmean([1, 4, 7], weights=[3, 1, 3])
2.80668351922014
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l00300}{300}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l00130}{gmean()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a92e6eba419d8ef505e1e48f9333d995b}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!pointbiserialr@{pointbiserialr}}
\index{pointbiserialr@{pointbiserialr}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{pointbiserialr()}{pointbiserialr()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a92e6eba419d8ef505e1e48f9333d995b} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+pointbiserialr (\begin{DoxyParamCaption}\item[{}]{x}{, }\item[{}]{y}{}\end{DoxyParamCaption})}

\begin{DoxyVerb}Calculate a point biserial correlation coefficient and its p-value.

The point biserial correlation is used to measure the relationship
between a binary variable, x, and a continuous variable, y. Like other
correlation coefficients, this one varies between -1 and +1 with 0
implying no correlation. Correlations of -1 or +1 imply a determinative
relationship.

This function may be computed using a shortcut formula but produces the
same result as `pearsonr`.

Parameters
----------
x : array_like of bools
    Input array.
y : array_like
    Input array.

Returns
-------
res: SignificanceResult
    An object containing attributes:

    statistic : float
        The R value.
    pvalue : float
        The two-sided p-value.

Notes
-----
`pointbiserialr` uses a t-test with ``n-1`` degrees of freedom.
It is equivalent to `pearsonr`.

The value of the point-biserial correlation can be calculated from:

.. math::

    r_{pb} = \frac{\overline{Y_1} - \overline{Y_0}}
                  {s_y}
             \sqrt{\frac{N_0 N_1}
                        {N (N - 1)}}

Where :math:`\overline{Y_{0}}` and :math:`\overline{Y_{1}}` are means
of the metric observations coded 0 and 1 respectively; :math:`N_{0}` and
:math:`N_{1}` are number of observations coded 0 and 1 respectively;
:math:`N` is the total number of observations and :math:`s_{y}` is the
standard deviation of all the metric observations.

A value of :math:`r_{pb}` that is significantly different from zero is
completely equivalent to a significant difference in means between the two
groups. Thus, an independent groups t Test with :math:`N-2` degrees of
freedom may be used to test whether :math:`r_{pb}` is nonzero. The
relation between the t-statistic for comparing two independent groups and
:math:`r_{pb}` is given by:

.. math::

    t = \sqrt{N - 2}\frac{r_{pb}}{\sqrt{1 - r^{2}_{pb}}}

References
----------
.. [1] J. Lev, "The Point Biserial Coefficient of Correlation", Ann. Math.
       Statist., Vol. 20, no.1, pp. 125-126, 1949.

.. [2] R.F. Tate, "Correlation Between a Discrete and a Continuous
       Variable. Point-Biserial Correlation.", Ann. Math. Statist., Vol. 25,
       np. 3, pp. 603-607, 1954.

.. [3] D. Kornbrot "Point Biserial Correlation", In Wiley StatsRef:
       Statistics Reference Online (eds N. Balakrishnan, et al.), 2014.
       :doi:`10.1002/9781118445112.stat06227`

Examples
--------
>>> import numpy as np
>>> from scipy import stats
>>> a = np.array([0, 0, 0, 1, 1, 1, 1])
>>> b = np.arange(7)
>>> stats.pointbiserialr(a, b)
(0.8660254037844386, 0.011724811003954652)
>>> stats.pearsonr(a, b)
(0.86602540378443871, 0.011724811003954626)
>>> np.corrcoef(a, b)
array([[ 1.       ,  0.8660254],
       [ 0.8660254,  1.       ]])
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l05499}{5499}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l04533}{pearsonr()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a13bf3d4c0145bfe43a1b40826651ea3f}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!power\_divergence@{power\_divergence}}
\index{power\_divergence@{power\_divergence}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{power\_divergence()}{power\_divergence()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a13bf3d4c0145bfe43a1b40826651ea3f} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+power\+\_\+divergence (\begin{DoxyParamCaption}\item[{}]{f\+\_\+obs}{, }\item[{}]{f\+\_\+exp}{ = {\ttfamily None}, }\item[{}]{ddof}{ = {\ttfamily 0}, }\item[{}]{axis}{ = {\ttfamily 0}, }\item[{}]{lambda\+\_\+}{ = {\ttfamily None}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Cressie-Read power divergence statistic and goodness of fit test.

This function tests the null hypothesis that the categorical data
has the given frequencies, using the Cressie-Read power divergence
statistic.

Parameters
----------
f_obs : array_like
    Observed frequencies in each category.
f_exp : array_like, optional
    Expected frequencies in each category.  By default the categories are
    assumed to be equally likely.
ddof : int, optional
    "Delta degrees of freedom": adjustment to the degrees of freedom
    for the p-value.  The p-value is computed using a chi-squared
    distribution with ``k - 1 - ddof`` degrees of freedom, where `k`
    is the number of observed frequencies.  The default value of `ddof`
    is 0.
axis : int or None, optional
    The axis of the broadcast result of `f_obs` and `f_exp` along which to
    apply the test.  If axis is None, all values in `f_obs` are treated
    as a single data set.  Default is 0.
lambda_ : float or str, optional
    The power in the Cressie-Read power divergence statistic.  The default
    is 1.  For convenience, `lambda_` may be assigned one of the following
    strings, in which case the corresponding numerical value is used:

    * ``"pearson"`` (value 1)
        Pearson's chi-squared statistic. In this case, the function is
        equivalent to `chisquare`.
    * ``"log-likelihood"`` (value 0)
        Log-likelihood ratio. Also known as the G-test [3]_.
    * ``"freeman-tukey"`` (value -1/2)
        Freeman-Tukey statistic.
    * ``"mod-log-likelihood"`` (value -1)
        Modified log-likelihood ratio.
    * ``"neyman"`` (value -2)
        Neyman's statistic.
    * ``"cressie-read"`` (value 2/3)
        The power recommended in [5]_.

Returns
-------
res: Power_divergenceResult
    An object containing attributes:

    statistic : float or ndarray
        The Cressie-Read power divergence test statistic.  The value is
        a float if `axis` is None or if` `f_obs` and `f_exp` are 1-D.
    pvalue : float or ndarray
        The p-value of the test.  The value is a float if `ddof` and the
        return value `stat` are scalars.

See Also
--------
chisquare

Notes
-----
This test is invalid when the observed or expected frequencies in each
category are too small.  A typical rule is that all of the observed
and expected frequencies should be at least 5.

Also, the sum of the observed and expected frequencies must be the same
for the test to be valid; `power_divergence` raises an error if the sums
do not agree within a relative tolerance of ``1e-8``.

When `lambda_` is less than zero, the formula for the statistic involves
dividing by `f_obs`, so a warning or error may be generated if any value
in `f_obs` is 0.

Similarly, a warning or error may be generated if any value in `f_exp` is
zero when `lambda_` >= 0.

The default degrees of freedom, k-1, are for the case when no parameters
of the distribution are estimated. If p parameters are estimated by
efficient maximum likelihood then the correct degrees of freedom are
k-1-p. If the parameters are estimated in a different way, then the
dof can be between k-1-p and k-1. However, it is also possible that
the asymptotic distribution is not a chisquare, in which case this
test is not appropriate.

This function handles masked arrays.  If an element of `f_obs` or `f_exp`
is masked, then data at that position is ignored, and does not count
towards the size of the data set.

.. versionadded:: 0.13.0

References
----------
.. [1] Lowry, Richard.  "Concepts and Applications of Inferential
       Statistics". Chapter 8.
       https://web.archive.org/web/20171015035606/http://faculty.vassar.edu/lowry/ch8pt1.html
.. [2] "Chi-squared test", https://en.wikipedia.org/wiki/Chi-squared_test
.. [3] "G-test", https://en.wikipedia.org/wiki/G-test
.. [4] Sokal, R. R. and Rohlf, F. J. "Biometry: the principles and
       practice of statistics in biological research", New York: Freeman
       (1981)
.. [5] Cressie, N. and Read, T. R. C., "Multinomial Goodness-of-Fit
       Tests", J. Royal Stat. Soc. Series B, Vol. 46, No. 3 (1984),
       pp. 440-464.

Examples
--------
(See `chisquare` for more examples.)

When just `f_obs` is given, it is assumed that the expected frequencies
are uniform and given by the mean of the observed frequencies.  Here we
perform a G-test (i.e. use the log-likelihood ratio statistic):

>>> import numpy as np
>>> from scipy.stats import power_divergence
>>> power_divergence([16, 18, 16, 14, 12, 12], lambda_='log-likelihood')
(2.006573162632538, 0.84823476779463769)

The expected frequencies can be given with the `f_exp` argument:

>>> power_divergence([16, 18, 16, 14, 12, 12],
...                  f_exp=[16, 16, 16, 16, 16, 8],
...                  lambda_='log-likelihood')
(3.3281031458963746, 0.6495419288047497)

When `f_obs` is 2-D, by default the test is applied to each column.

>>> obs = np.array([[16, 18, 16, 14, 12, 12], [32, 24, 16, 28, 20, 24]]).T
>>> obs.shape
(6, 2)
>>> power_divergence(obs, lambda_="log-likelihood")
(array([ 2.00657316,  6.77634498]), array([ 0.84823477,  0.23781225]))

By setting ``axis=None``, the test is applied to all data in the array,
which is equivalent to applying the test to the flattened array.

>>> power_divergence(obs, axis=None)
(23.31034482758621, 0.015975692534127565)
>>> power_divergence(obs.ravel())
(23.31034482758621, 0.015975692534127565)

`ddof` is the change to make to the default degrees of freedom.

>>> power_divergence([16, 18, 16, 14, 12, 12], ddof=1)
(2.0, 0.73575888234288467)

The calculation of the p-values is done by broadcasting the
test statistic with `ddof`.

>>> power_divergence([16, 18, 16, 14, 12, 12], ddof=[0,1,2])
(2.0, array([ 0.84914504,  0.73575888,  0.5724067 ]))

`f_obs` and `f_exp` are also broadcast.  In the following, `f_obs` has
shape (6,) and `f_exp` has shape (2, 6), so the result of broadcasting
`f_obs` and `f_exp` has shape (2, 6).  To compute the desired chi-squared
statistics, we must use ``axis=1``:

>>> power_divergence([16, 18, 16, 14, 12, 12],
...                  f_exp=[[16, 16, 16, 16, 16, 8],
...                         [8, 20, 20, 16, 12, 12]],
...                  axis=1)
(array([ 3.5 ,  9.25]), array([ 0.62338763,  0.09949846]))
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l07857}{7857}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l07826}{\+\_\+count()}}.



Referenced by \mbox{\hyperlink{contingency_8py_source_l00144}{scipy.\+stats.\+contingency.\+chi2\+\_\+contingency()}}, and \mbox{\hyperlink{__stats__py_8py_source_l08085}{chisquare()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_aa3911ef47b2d3dd65d7fc164b15a4799}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!rankdata@{rankdata}}
\index{rankdata@{rankdata}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{rankdata()}{rankdata()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_aa3911ef47b2d3dd65d7fc164b15a4799} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+rankdata (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{method}{ = {\ttfamily \textquotesingle{}average\textquotesingle{}}, }\item[{\texorpdfstring{$\ast$}{*}}]{}{, }\item[{}]{axis}{ = {\ttfamily None}, }\item[{}]{nan\+\_\+policy}{ = {\ttfamily \textquotesingle{}propagate\textquotesingle{}}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Assign ranks to data, dealing with ties appropriately.

By default (``axis=None``), the data array is first flattened, and a flat
array of ranks is returned. Separately reshape the rank array to the
shape of the data array if desired (see Examples).

Ranks begin at 1.  The `method` argument controls how ranks are assigned
to equal values.  See [1]_ for further discussion of ranking methods.

Parameters
----------
a : array_like
    The array of values to be ranked.
method : {'average', 'min', 'max', 'dense', 'ordinal'}, optional
    The method used to assign ranks to tied elements.
    The following methods are available (default is 'average'):

      * 'average': The average of the ranks that would have been assigned to
        all the tied values is assigned to each value.
      * 'min': The minimum of the ranks that would have been assigned to all
        the tied values is assigned to each value.  (This is also
        referred to as "competition" ranking.)
      * 'max': The maximum of the ranks that would have been assigned to all
        the tied values is assigned to each value.
      * 'dense': Like 'min', but the rank of the next highest element is
        assigned the rank immediately after those assigned to the tied
        elements.
      * 'ordinal': All values are given a distinct rank, corresponding to
        the order that the values occur in `a`.
axis : {None, int}, optional
    Axis along which to perform the ranking. If ``None``, the data array
    is first flattened.
nan_policy : {'propagate', 'omit', 'raise'}, optional
    Defines how to handle when input contains nan.
    The following options are available (default is 'propagate'):

      * 'propagate': propagates nans through the rank calculation
      * 'omit': performs the calculations ignoring nan values
      * 'raise': raises an error

    .. note::

        When `nan_policy` is 'propagate', the output is an array of *all*
        nans because ranks relative to nans in the input are undefined.
        When `nan_policy` is 'omit', nans in `a` are ignored when ranking
        the other values, and the corresponding locations of the output
        are nan.

    .. versionadded:: 1.10

Returns
-------
ranks : ndarray
     An array of size equal to the size of `a`, containing rank
     scores.

References
----------
.. [1] "Ranking", https://en.wikipedia.org/wiki/Ranking

Examples
--------
>>> import numpy as np
>>> from scipy.stats import rankdata
>>> rankdata([0, 2, 3, 2])
array([ 1. ,  2.5,  4. ,  2.5])
>>> rankdata([0, 2, 3, 2], method='min')
array([ 1,  2,  4,  2])
>>> rankdata([0, 2, 3, 2], method='max')
array([ 1,  3,  4,  3])
>>> rankdata([0, 2, 3, 2], method='dense')
array([ 1,  2,  3,  2])
>>> rankdata([0, 2, 3, 2], method='ordinal')
array([ 1,  2,  4,  3])
>>> rankdata([[0, 2], [3, 2]]).reshape(2,2)
array([[1. , 2.5],
      [4. , 2.5]])
>>> rankdata([[0, 2, 2], [3, 2, 5]], axis=1)
array([[1. , 2.5, 2.5],
       [2. , 1. , 3. ]])
>>> rankdata([0, 2, 3, np.nan, -2, np.nan], nan_policy="propagate")
array([nan, nan, nan, nan, nan, nan])
>>> rankdata([0, 2, 3, np.nan, -2, np.nan], nan_policy="omit")
array([ 2.,  3.,  4., nan,  1., nan])
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l10123}{10123}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l09437}{brunnermunzel()}}, \mbox{\hyperlink{__stats__py_8py_source_l09338}{friedmanchisquare()}}, \mbox{\hyperlink{__stats__py_8py_source_l09215}{kruskal()}}, \mbox{\hyperlink{__stats__py_8py_source_l09133}{ranksums()}}, \mbox{\hyperlink{scipy_2stats_2tests_2test__rank_8py_source_l00093}{scipy.\+stats.\+tests.\+test\+\_\+rank.\+Test\+Rank\+Data.\+test\+\_\+basic()}}, \mbox{\hyperlink{scipy_2stats_2tests_2test__rank_8py_source_l00031}{scipy.\+stats.\+tests.\+test\+\_\+rank.\+Test\+Tie\+Correct.\+test\+\_\+basic()}}, \mbox{\hyperlink{scipy_2stats_2tests_2test__rank_8py_source_l00076}{scipy.\+stats.\+tests.\+test\+\_\+rank.\+Test\+Rank\+Data.\+test\+\_\+empty()}}, and \mbox{\hyperlink{scipy_2stats_2tests_2test__rank_8py_source_l00084}{scipy.\+stats.\+tests.\+test\+\_\+rank.\+Test\+Rank\+Data.\+test\+\_\+one()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a5c9e3566461d8bb479cb9507e9a1bd2e}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!ranksums@{ranksums}}
\index{ranksums@{ranksums}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{ranksums()}{ranksums()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a5c9e3566461d8bb479cb9507e9a1bd2e} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+ranksums (\begin{DoxyParamCaption}\item[{}]{x}{, }\item[{}]{y}{, }\item[{}]{alternative}{ = {\ttfamily \textquotesingle{}two-\/sided\textquotesingle{}}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Compute the Wilcoxon rank-sum statistic for two samples.

The Wilcoxon rank-sum test tests the null hypothesis that two sets
of measurements are drawn from the same distribution.  The alternative
hypothesis is that values in one sample are more likely to be
larger than the values in the other sample.

This test should be used to compare two samples from continuous
distributions.  It does not handle ties between measurements
in x and y.  For tie-handling and an optional continuity correction
see `scipy.stats.mannwhitneyu`.

Parameters
----------
x,y : array_like
    The data from the two samples.
alternative : {'two-sided', 'less', 'greater'}, optional
    Defines the alternative hypothesis. Default is 'two-sided'.
    The following options are available:

    * 'two-sided': one of the distributions (underlying `x` or `y`) is
      stochastically greater than the other.
    * 'less': the distribution underlying `x` is stochastically less
      than the distribution underlying `y`.
    * 'greater': the distribution underlying `x` is stochastically greater
      than the distribution underlying `y`.

    .. versionadded:: 1.7.0

Returns
-------
statistic : float
    The test statistic under the large-sample approximation that the
    rank sum statistic is normally distributed.
pvalue : float
    The p-value of the test.

References
----------
.. [1] https://en.wikipedia.org/wiki/Wilcoxon_rank-sum_test

Examples
--------
We can test the hypothesis that two independent unequal-sized samples are
drawn from the same distribution with computing the Wilcoxon rank-sum
statistic.

>>> import numpy as np
>>> from scipy.stats import ranksums
>>> rng = np.random.default_rng()
>>> sample1 = rng.uniform(-1, 1, 200)
>>> sample2 = rng.uniform(-0.5, 1.5, 300) # a shifted distribution
>>> ranksums(sample1, sample2)
RanksumsResult(statistic=-7.887059, pvalue=3.09390448e-15)  # may vary
>>> ranksums(sample1, sample2, alternative='less')
RanksumsResult(statistic=-7.750585297581713, pvalue=4.573497606342543e-15) # may vary
>>> ranksums(sample1, sample2, alternative='greater')
RanksumsResult(statistic=-7.750585297581713, pvalue=0.9999999999999954) # may vary

The p-value of less than ``0.05`` indicates that this test rejects the
hypothesis at the 5% significance level.
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l09133}{9133}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l01415}{\+\_\+normtest\+\_\+finish()}}, and \mbox{\hyperlink{__stats__py_8py_source_l10123}{rankdata()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_ad241a1e1cd58da4789b3217d42536d0b}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!relfreq@{relfreq}}
\index{relfreq@{relfreq}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{relfreq()}{relfreq()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_ad241a1e1cd58da4789b3217d42536d0b} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+relfreq (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{numbins}{ = {\ttfamily 10}, }\item[{}]{defaultreallimits}{ = {\ttfamily None}, }\item[{}]{weights}{ = {\ttfamily None}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Return a relative frequency histogram, using the histogram function.

A relative frequency  histogram is a mapping of the number of
observations in each of the bins relative to the total of observations.

Parameters
----------
a : array_like
    Input array.
numbins : int, optional
    The number of bins to use for the histogram. Default is 10.
defaultreallimits : tuple (lower, upper), optional
    The lower and upper values for the range of the histogram.
    If no value is given, a range slightly larger than the range of the
    values in a is used. Specifically ``(a.min() - s, a.max() + s)``,
    where ``s = (1/2)(a.max() - a.min()) / (numbins - 1)``.
weights : array_like, optional
    The weights for each value in `a`. Default is None, which gives each
    value a weight of 1.0

Returns
-------
frequency : ndarray
    Binned values of relative frequency.
lowerlimit : float
    Lower real limit.
binsize : float
    Width of each bin.
extrapoints : int
    Extra points.

Examples
--------
>>> import numpy as np
>>> import matplotlib.pyplot as plt
>>> from scipy import stats
>>> rng = np.random.default_rng()
>>> a = np.array([2, 4, 1, 2, 3, 2])
>>> res = stats.relfreq(a, numbins=4)
>>> res.frequency
array([ 0.16666667, 0.5       , 0.16666667,  0.16666667])
>>> np.sum(res.frequency)  # relative frequencies should add up to 1
1.0

Create a normal distribution with 1000 random values

>>> samples = stats.norm.rvs(size=1000, random_state=rng)

Calculate relative frequencies

>>> res = stats.relfreq(samples, numbins=25)

Calculate space of values for x

>>> x = res.lowerlimit + np.linspace(0, res.binsize*res.frequency.size,
...                                  res.frequency.size)

Plot relative frequency histogram

>>> fig = plt.figure(figsize=(5, 4))
>>> ax = fig.add_subplot(1, 1, 1)
>>> ax.bar(x, res.frequency, width=res.binsize)
>>> ax.set_title('Relative frequency histogram')
>>> ax.set_xlim([x.min(), x.max()])

>>> plt.show()
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l02624}{2624}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l02460}{\+\_\+histogram()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a50b35c4e412f628c2f7e90a36bdabf25}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!scoreatpercentile@{scoreatpercentile}}
\index{scoreatpercentile@{scoreatpercentile}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{scoreatpercentile()}{scoreatpercentile()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a50b35c4e412f628c2f7e90a36bdabf25} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+scoreatpercentile (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{per}{, }\item[{}]{limit}{ = {\ttfamily ()}, }\item[{}]{interpolation\+\_\+method}{ = {\ttfamily \textquotesingle{}fraction\textquotesingle{}}, }\item[{}]{axis}{ = {\ttfamily None}}\end{DoxyParamCaption})}



FREQUENCY FUNCTIONS \#. 

\begin{DoxyVerb}Calculate the score at a given percentile of the input sequence.

For example, the score at `per=50` is the median. If the desired quantile
lies between two data points, we interpolate between them, according to
the value of `interpolation`. If the parameter `limit` is provided, it
should be a tuple (lower, upper) of two values.

Parameters
----------
a : array_like
    A 1-D array of values from which to extract score.
per : array_like
    Percentile(s) at which to extract score.  Values should be in range
    [0,100].
limit : tuple, optional
    Tuple of two scalars, the lower and upper limits within which to
    compute the percentile. Values of `a` outside
    this (closed) interval will be ignored.
interpolation_method : {'fraction', 'lower', 'higher'}, optional
    Specifies the interpolation method to use,
    when the desired quantile lies between two data points `i` and `j`
    The following options are available (default is 'fraction'):

      * 'fraction': ``i + (j - i) * fraction`` where ``fraction`` is the
        fractional part of the index surrounded by ``i`` and ``j``
      * 'lower': ``i``
      * 'higher': ``j``

axis : int, optional
    Axis along which the percentiles are computed. Default is None. If
    None, compute over the whole array `a`.

Returns
-------
score : float or ndarray
    Score at percentile(s).

See Also
--------
percentileofscore, numpy.percentile

Notes
-----
This function will become obsolete in the future.
For NumPy 1.9 and higher, `numpy.percentile` provides all the functionality
that `scoreatpercentile` provides.  And it's significantly faster.
Therefore it's recommended to use `numpy.percentile` for users that have
numpy >= 1.9.

Examples
--------
>>> import numpy as np
>>> from scipy import stats
>>> a = np.arange(100)
>>> stats.scoreatpercentile(a, 50)
49.5
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l02168}{2168}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_ac92d7d58643573008b4ddc6dfad7447a}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!sem@{sem}}
\index{sem@{sem}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{sem()}{sem()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_ac92d7d58643573008b4ddc6dfad7447a} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+sem (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{axis}{ = {\ttfamily 0}, }\item[{}]{ddof}{ = {\ttfamily 1}, }\item[{}]{nan\+\_\+policy}{ = {\ttfamily \textquotesingle{}propagate\textquotesingle{}}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Compute standard error of the mean.

Calculate the standard error of the mean (or standard error of
measurement) of the values in the input array.

Parameters
----------
a : array_like
    An array containing the values for which the standard error is
    returned.
axis : int or None, optional
    Axis along which to operate. Default is 0. If None, compute over
    the whole array `a`.
ddof : int, optional
    Delta degrees-of-freedom. How many degrees of freedom to adjust
    for bias in limited samples relative to the population estimate
    of variance. Defaults to 1.
nan_policy : {'propagate', 'raise', 'omit'}, optional
    Defines how to handle when input contains nan.
    The following options are available (default is 'propagate'):

      * 'propagate': returns nan
      * 'raise': throws an error
      * 'omit': performs the calculations ignoring nan values

Returns
-------
s : ndarray or float
    The standard error of the mean in the sample(s), along the input axis.

Notes
-----
The default value for `ddof` is different to the default (0) used by other
ddof containing routines, such as np.std and np.nanstd.

Examples
--------
Find standard error along the first axis:

>>> import numpy as np
>>> from scipy import stats
>>> a = np.arange(20).reshape(5,4)
>>> stats.sem(a)
array([ 2.8284,  2.8284,  2.8284,  2.8284])

Find standard error across the whole array, using n degrees of freedom:

>>> stats.sem(a, axis=None, ddof=0)
1.2893796958227628
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l02790}{2790}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a40c059cb94ceb8ed4b4573bca8b8024f}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!sigmaclip@{sigmaclip}}
\index{sigmaclip@{sigmaclip}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{sigmaclip()}{sigmaclip()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a40c059cb94ceb8ed4b4573bca8b8024f} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+sigmaclip (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{low}{ = {\ttfamily 4.}, }\item[{}]{high}{ = {\ttfamily 4.}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Perform iterative sigma-clipping of array elements.

Starting from the full sample, all elements outside the critical range are
removed, i.e. all elements of the input array `c` that satisfy either of
the following conditions::

    c < mean(c) - std(c)*low
    c > mean(c) + std(c)*high

The iteration continues with the updated sample until no
elements are outside the (updated) range.

Parameters
----------
a : array_like
    Data array, will be raveled if not 1-D.
low : float, optional
    Lower bound factor of sigma clipping. Default is 4.
high : float, optional
    Upper bound factor of sigma clipping. Default is 4.

Returns
-------
clipped : ndarray
    Input array with clipped elements removed.
lower : float
    Lower threshold value use for clipping.
upper : float
    Upper threshold value use for clipping.

Examples
--------
>>> import numpy as np
>>> from scipy.stats import sigmaclip
>>> a = np.concatenate((np.linspace(9.5, 10.5, 31),
...                     np.linspace(0, 20, 5)))
>>> fact = 1.5
>>> c, low, upp = sigmaclip(a, fact, fact)
>>> c
array([  9.96666667,  10.        ,  10.03333333,  10.        ])
>>> c.var(), c.std()
(0.00055555555555555165, 0.023570226039551501)
>>> low, c.mean() - fact*c.std(), c.min()
(9.9646446609406727, 9.9646446609406727, 9.9666666666666668)
>>> upp, c.mean() + fact*c.std(), c.max()
(10.035355339059327, 10.035355339059327, 10.033333333333333)

>>> a = np.concatenate((np.linspace(9.5, 10.5, 11),
...                     np.linspace(-100, -50, 3)))
>>> c, low, upp = sigmaclip(a, 1.8, 1.8)
>>> (c == np.linspace(9.5, 10.5, 11)).all()
True
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l03648}{3648}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l03645}{Sigmaclip\+Result}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a57e56bc1f28e285175aa52cd9a8dccec}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!skew@{skew}}
\index{skew@{skew}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{skew()}{skew()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a57e56bc1f28e285175aa52cd9a8dccec} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+skew (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{axis}{ = {\ttfamily 0}, }\item[{}]{bias}{ = {\ttfamily \mbox{\hyperlink{classTrue}{True}}}, }\item[{}]{nan\+\_\+policy}{ = {\ttfamily \textquotesingle{}propagate\textquotesingle{}}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Compute the sample skewness of a data set.

For normally distributed data, the skewness should be about zero. For
unimodal continuous distributions, a skewness value greater than zero means
that there is more weight in the right tail of the distribution. The
function `skewtest` can be used to determine if the skewness value
is close enough to zero, statistically speaking.

Parameters
----------
a : ndarray
    Input array.
axis : int or None, optional
    Axis along which skewness is calculated. Default is 0.
    If None, compute over the whole array `a`.
bias : bool, optional
    If False, then the calculations are corrected for statistical bias.
nan_policy : {'propagate', 'raise', 'omit'}, optional
    Defines how to handle when input contains nan.
    The following options are available (default is 'propagate'):

      * 'propagate': returns nan
      * 'raise': throws an error
      * 'omit': performs the calculations ignoring nan values

Returns
-------
skewness : ndarray
    The skewness of values along an axis, returning NaN where all values
    are equal.

Notes
-----
The sample skewness is computed as the Fisher-Pearson coefficient
of skewness, i.e.

.. math::

    g_1=\frac{m_3}{m_2^{3/2}}

where

.. math::

    m_i=\frac{1}{N}\sum_{n=1}^N(x[n]-\bar{x})^i

is the biased sample :math:`i\texttt{th}` central moment, and
:math:`\bar{x}` is
the sample mean.  If ``bias`` is False, the calculations are
corrected for bias and the value computed is the adjusted
Fisher-Pearson standardized moment coefficient, i.e.

.. math::

    G_1=\frac{k_3}{k_2^{3/2}}=
        \frac{\sqrt{N(N-1)}}{N-2}\frac{m_3}{m_2^{3/2}}.

References
----------
.. [1] Zwillinger, D. and Kokoska, S. (2000). CRC Standard
   Probability and Statistics Tables and Formulae. Chapman & Hall: New
   York. 2000.
   Section 2.2.24.1

Examples
--------
>>> from scipy.stats import skew
>>> skew([1, 2, 3, 4, 5])
0.0
>>> skew([2, 8, 0, 4, 1, 9, 9, 0])
0.2650554122698573
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l01110}{1110}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l01326}{describe()}}, \mbox{\hyperlink{__continuous__distns_8py_source_l02475}{scipy.\+stats.\+\_\+continuous\+\_\+distns.\+weibull\+\_\+min\+\_\+gen.\+fit()}}, \mbox{\hyperlink{__stats__py_8py_source_l02000}{jarque\+\_\+bera()}}, and \mbox{\hyperlink{__stats__py_8py_source_l01436}{skewtest()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a7ba4fbe391d511f26a5715213741ce19}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!skewtest@{skewtest}}
\index{skewtest@{skewtest}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{skewtest()}{skewtest()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a7ba4fbe391d511f26a5715213741ce19} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+skewtest (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{axis}{ = {\ttfamily 0}, }\item[{}]{nan\+\_\+policy}{ = {\ttfamily \textquotesingle{}propagate\textquotesingle{}}, }\item[{}]{alternative}{ = {\ttfamily \textquotesingle{}two-\/sided\textquotesingle{}}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Test whether the skew is different from the normal distribution.

This function tests the null hypothesis that the skewness of
the population that the sample was drawn from is the same
as that of a corresponding normal distribution.

Parameters
----------
a : array
The data to be tested.
axis : int or None, optional
Axis along which statistics are calculated. Default is 0.
If None, compute over the whole array `a`.
nan_policy : {'propagate', 'raise', 'omit'}, optional
Defines how to handle when input contains nan.
The following options are available (default is 'propagate'):

* 'propagate': returns nan
* 'raise': throws an error
* 'omit': performs the calculations ignoring nan values

alternative : {'two-sided', 'less', 'greater'}, optional
Defines the alternative hypothesis. Default is 'two-sided'.
The following options are available:

* 'two-sided': the skewness of the distribution underlying the sample
is different from that of the normal distribution (i.e. 0)
* 'less': the skewness of the distribution underlying the sample
is less than that of the normal distribution
* 'greater': the skewness of the distribution underlying the sample
is greater than that of the normal distribution

.. versionadded:: 1.7.0

Returns
-------
statistic : float
The computed z-score for this test.
pvalue : float
The p-value for the hypothesis test.

Notes
-----
The sample size must be at least 8.

References
----------
.. [1] R. B. D'Agostino, A. J. Belanger and R. B. D'Agostino Jr.,
"A suggestion for using powerful and informative tests of
normality", American Statistician 44, pp. 316-321, 1990.
.. [2] Shapiro, S. S., & Wilk, M. B. (1965). An analysis of variance test
for normality (complete samples). Biometrika, 52(3/4), 591-611.
.. [3] B. Phipson and G. K. Smyth. "Permutation P-values Should Never Be
Zero: Calculating Exact P-values When Permutations Are Randomly
Drawn." Statistical Applications in Genetics and Molecular Biology
9.1 (2010).

Examples
--------
Suppose we wish to infer from measurements whether the weights of adult
human males in a medical study are not normally distributed [2]_.
The weights (lbs) are recorded in the array ``x`` below.

>>> import numpy as np
>>> x = np.array([148, 154, 158, 160, 161, 162, 166, 170, 182, 195, 236])

The skewness test from [1]_ begins by computing a statistic based on the
sample skewness.

>>> from scipy import stats
>>> res = stats.skewtest(x)
>>> res.statistic
2.7788579769903414

Because normal distributions have zero skewness, the magnitude of this
statistic tends to be low for samples drawn from a normal distribution.

The test is performed by comparing the observed value of the
statistic against the null distribution: the distribution of statistic
values derived under the null hypothesis that the weights were drawn from
a normal distribution.

For this test, the null distribution of the statistic for very large
samples is the standard normal distribution.

>>> import matplotlib.pyplot as plt
>>> dist = stats.norm()
>>> st_val = np.linspace(-5, 5, 100)
>>> pdf = dist.pdf(st_val)
>>> fig, ax = plt.subplots(figsize=(8, 5))
>>> def st_plot(ax):  # we'll re-use this
...     ax.plot(st_val, pdf)
...     ax.set_title("Skew Test Null Distribution")
...     ax.set_xlabel("statistic")
...     ax.set_ylabel("probability density")
>>> st_plot(ax)
>>> plt.show()

The comparison is quantified by the p-value: the proportion of values in
the null distribution as extreme or more extreme than the observed
value of the statistic. In a two-sided test, elements of the null
distribution greater than the observed statistic and elements of the null
distribution less than the negative of the observed statistic are both
considered "more extreme".

>>> fig, ax = plt.subplots(figsize=(8, 5))
>>> st_plot(ax)
>>> pvalue = dist.cdf(-res.statistic) + dist.sf(res.statistic)
>>> annotation = (f'p-value={pvalue:.3f}\n(shaded area)')
>>> props = dict(facecolor='black', width=1, headwidth=5, headlength=8)
>>> _ = ax.annotate(annotation, (3, 0.005), (3.25, 0.02), arrowprops=props)
>>> i = st_val >= res.statistic
>>> ax.fill_between(st_val[i], y1=0, y2=pdf[i], color='C0')
>>> i = st_val <= -res.statistic
>>> ax.fill_between(st_val[i], y1=0, y2=pdf[i], color='C0')
>>> ax.set_xlim(-5, 5)
>>> ax.set_ylim(0, 0.1)
>>> plt.show()
>>> res.pvalue
0.005455036974740185

If the p-value is "small" - that is, if there is a low probability of
sampling data from a normally distributed population that produces such an
extreme value of the statistic - this may be taken as evidence against
the null hypothesis in favor of the alternative: the weights were not
drawn from a normal distribution. Note that:

- The inverse is not true; that is, the test is not used to provide
evidence for the null hypothesis.
- The threshold for values that will be considered "small" is a choice that
should be made before the data is analyzed [3]_ with consideration of the
risks of both false positives (incorrectly rejecting the null hypothesis)
and false negatives (failure to reject a false null hypothesis).

Note that the standard normal distribution provides an asymptotic
approximation of the null distribution; it is only accurate for samples
with many observations. For small samples like ours,
`scipy.stats.monte_carlo_test` may provide a more accurate, albeit
stochastic, approximation of the exact p-value.

>>> def statistic(x, axis):
...     # get just the skewtest statistic; ignore the p-value
...     return stats.skewtest(x, axis=axis).statistic
>>> res = stats.monte_carlo_test(x, stats.norm.rvs, statistic)
>>> fig, ax = plt.subplots(figsize=(8, 5))
>>> st_plot(ax)
>>> ax.hist(res.null_distribution, np.linspace(-5, 5, 50),
...         density=True)
>>> ax.legend(['aymptotic approximation\n(many observations)',
...            'Monte Carlo approximation\n(11 observations)'])
>>> plt.show()
>>> res.pvalue
0.0062  # may vary

In this case, the asymptotic approximation and Monte Carlo approximation
agree fairly closely, even for our small sample.
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l01436}{1436}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l01415}{\+\_\+normtest\+\_\+finish()}}, and \mbox{\hyperlink{__stats__py_8py_source_l01110}{skew()}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l01836}{normaltest()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a38625835ee2b2814b8dd9ce450b73bde}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!spearmanr@{spearmanr}}
\index{spearmanr@{spearmanr}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{spearmanr()}{spearmanr()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a38625835ee2b2814b8dd9ce450b73bde} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+spearmanr (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{b}{ = {\ttfamily None}, }\item[{}]{axis}{ = {\ttfamily 0}, }\item[{}]{nan\+\_\+policy}{ = {\ttfamily \textquotesingle{}propagate\textquotesingle{}}, }\item[{}]{alternative}{ = {\ttfamily \textquotesingle{}two-\/sided\textquotesingle{}}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Calculate a Spearman correlation coefficient with associated p-value.

The Spearman rank-order correlation coefficient is a nonparametric measure
of the monotonicity of the relationship between two datasets.
Like other correlation coefficients,
this one varies between -1 and +1 with 0 implying no correlation.
Correlations of -1 or +1 imply an exact monotonic relationship. Positive
correlations imply that as x increases, so does y. Negative correlations
imply that as x increases, y decreases.

The p-value roughly indicates the probability of an uncorrelated system
producing datasets that have a Spearman correlation at least as extreme
as the one computed from these datasets. Although calculation of the
p-value does not make strong assumptions about the distributions underlying
the samples, it is only accurate for very large samples (>500
observations). For smaller sample sizes, consider a permutation test (see
Examples section below).

Parameters
----------
a, b : 1D or 2D array_like, b is optional
One or two 1-D or 2-D arrays containing multiple variables and
observations. When these are 1-D, each represents a vector of
observations of a single variable. For the behavior in the 2-D case,
see under ``axis``, below.
Both arrays need to have the same length in the ``axis`` dimension.
axis : int or None, optional
If axis=0 (default), then each column represents a variable, with
observations in the rows. If axis=1, the relationship is transposed:
each row represents a variable, while the columns contain observations.
If axis=None, then both arrays will be raveled.
nan_policy : {'propagate', 'raise', 'omit'}, optional
Defines how to handle when input contains nan.
The following options are available (default is 'propagate'):

* 'propagate': returns nan
* 'raise': throws an error
* 'omit': performs the calculations ignoring nan values

alternative : {'two-sided', 'less', 'greater'}, optional
Defines the alternative hypothesis. Default is 'two-sided'.
The following options are available:

* 'two-sided': the correlation is nonzero
* 'less': the correlation is negative (less than zero)
* 'greater':  the correlation is positive (greater than zero)

.. versionadded:: 1.7.0

Returns
-------
res : SignificanceResult
An object containing attributes:

statistic : float or ndarray (2-D square)
    Spearman correlation matrix or correlation coefficient (if only 2
    variables are given as parameters). Correlation matrix is square
    with length equal to total number of variables (columns or rows) in
    ``a`` and ``b`` combined.
pvalue : float
    The p-value for a hypothesis test whose null hypothesis
    is that two samples have no ordinal correlation. See
    `alternative` above for alternative hypotheses. `pvalue` has the
    same shape as `statistic`.

Warns
-----
`~scipy.stats.ConstantInputWarning`
Raised if an input is a constant array.  The correlation coefficient
is not defined in this case, so ``np.nan`` is returned.

References
----------
.. [1] Zwillinger, D. and Kokoska, S. (2000). CRC Standard
Probability and Statistics Tables and Formulae. Chapman & Hall: New
York. 2000.
Section  14.7
.. [2] Kendall, M. G. and Stuart, A. (1973).
The Advanced Theory of Statistics, Volume 2: Inference and Relationship.
Griffin. 1973.
Section 31.18
.. [3] Kershenobich, D., Fierro, F. J., & Rojkind, M. (1970). The
relationship between the free pool of proline and collagen content in
human liver cirrhosis. The Journal of Clinical Investigation, 49(12),
2246-2249.
.. [4] Hollander, M., Wolfe, D. A., & Chicken, E. (2013). Nonparametric
statistical methods. John Wiley & Sons.
.. [5] B. Phipson and G. K. Smyth. "Permutation P-values Should Never Be
Zero: Calculating Exact P-values When Permutations Are Randomly Drawn."
Statistical Applications in Genetics and Molecular Biology 9.1 (2010).
.. [6] Ludbrook, J., & Dudley, H. (1998). Why permutation tests are
superior to t and F tests in biomedical research. The American
Statistician, 52(2), 127-132.

Examples
--------
Consider the following data from [3]_, which studied the relationship
between free proline (an amino acid) and total collagen (a protein often
found in connective tissue) in unhealthy human livers.

The ``x`` and ``y`` arrays below record measurements of the two compounds.
The observations are paired: each free proline measurement was taken from
the same liver as the total collagen measurement at the same index.

>>> import numpy as np
>>> # total collagen (mg/g dry weight of liver)
>>> x = np.array([7.1, 7.1, 7.2, 8.3, 9.4, 10.5, 11.4])
>>> # free proline ( mole/g dry weight of liver)
>>> y = np.array([2.8, 2.9, 2.8, 2.6, 3.5, 4.6, 5.0])

These data were analyzed in [4]_ using Spearman's correlation coefficient,
a statistic sensitive to monotonic correlation between the samples.

>>> from scipy import stats
>>> res = stats.spearmanr(x, y)
>>> res.statistic
0.7000000000000001

The value of this statistic tends to be high (close to 1) for samples with
a strongly positive ordinal correlation, low (close to -1) for samples with
a strongly negative ordinal correlation, and small in magnitude (close to
zero) for samples with weak ordinal correlation.

The test is performed by comparing the observed value of the
statistic against the null distribution: the distribution of statistic
values derived under the null hypothesis that total collagen and free
proline measurements are independent.

For this test, the statistic can be transformed such that the null
distribution for large samples is Student's t distribution with
``len(x) - 2`` degrees of freedom.

>>> import matplotlib.pyplot as plt
>>> dof = len(x)-2  # len(x) == len(y)
>>> dist = stats.t(df=dof)
>>> t_vals = np.linspace(-5, 5, 100)
>>> pdf = dist.pdf(t_vals)
>>> fig, ax = plt.subplots(figsize=(8, 5))
>>> def plot(ax):  # we'll re-use this
...     ax.plot(t_vals, pdf)
...     ax.set_title("Spearman's Rho Test Null Distribution")
...     ax.set_xlabel("statistic")
...     ax.set_ylabel("probability density")
>>> plot(ax)
>>> plt.show()

The comparison is quantified by the p-value: the proportion of values in
the null distribution as extreme or more extreme than the observed
value of the statistic. In a two-sided test in which the statistic is
positive, elements of the null distribution greater than the transformed
statistic and elements of the null distribution less than the negative of
the observed statistic are both considered "more extreme".

>>> fig, ax = plt.subplots(figsize=(8, 5))
>>> plot(ax)
>>> rs = res.statistic  # original statistic
>>> transformed = rs * np.sqrt(dof / ((rs+1.0)*(1.0-rs)))
>>> pvalue = dist.cdf(-transformed) + dist.sf(transformed)
>>> annotation = (f'p-value={pvalue:.4f}\n(shaded area)')
>>> props = dict(facecolor='black', width=1, headwidth=5, headlength=8)
>>> _ = ax.annotate(annotation, (2.7, 0.025), (3, 0.03), arrowprops=props)
>>> i = t_vals >= transformed
>>> ax.fill_between(t_vals[i], y1=0, y2=pdf[i], color='C0')
>>> i = t_vals <= -transformed
>>> ax.fill_between(t_vals[i], y1=0, y2=pdf[i], color='C0')
>>> ax.set_xlim(-5, 5)
>>> ax.set_ylim(0, 0.1)
>>> plt.show()
>>> res.pvalue
0.07991669030889909  # two-sided p-value

If the p-value is "small" - that is, if there is a low probability of
sampling data from independent distributions that produces such an extreme
value of the statistic - this may be taken as evidence against the null
hypothesis in favor of the alternative: the distribution of total collagen
and free proline are *not* independent. Note that:

- The inverse is not true; that is, the test is not used to provide
evidence for the null hypothesis.
- The threshold for values that will be considered "small" is a choice that
should be made before the data is analyzed [5]_ with consideration of the
risks of both false positives (incorrectly rejecting the null hypothesis)
and false negatives (failure to reject a false null hypothesis).
- Small p-values are not evidence for a *large* effect; rather, they can
only provide evidence for a "significant" effect, meaning that they are
unlikely to have occurred under the null hypothesis.

Suppose that before performing the experiment, the authors had reason
to predict a positive correlation between the total collagen and free
proline measurements, and that they had chosen to assess the plausibility
of the null hypothesis against a one-sided alternative: free proline has a
positive ordinal correlation with total collagen. In this case, only those
values in the null distribution that are as great or greater than the
observed statistic are considered to be more extreme.

>>> res = stats.spearmanr(x, y, alternative='greater')
>>> res.statistic
0.7000000000000001  # same statistic
>>> fig, ax = plt.subplots(figsize=(8, 5))
>>> plot(ax)
>>> pvalue = dist.sf(transformed)
>>> annotation = (f'p-value={pvalue:.6f}\n(shaded area)')
>>> props = dict(facecolor='black', width=1, headwidth=5, headlength=8)
>>> _ = ax.annotate(annotation, (3, 0.018), (3.5, 0.03), arrowprops=props)
>>> i = t_vals >= transformed
>>> ax.fill_between(t_vals[i], y1=0, y2=pdf[i], color='C0')
>>> ax.set_xlim(1, 5)
>>> ax.set_ylim(0, 0.1)
>>> plt.show()
>>> res.pvalue
0.03995834515444954  # one-sided p-value; half of the two-sided p-value

Note that the t-distribution provides an asymptotic approximation of the
null distribution; it is only accurate for samples with many observations.
For small samples, it may be more appropriate to perform a permutation
test: Under the null hypothesis that total collagen and free proline are
independent, each of the free proline measurements were equally likely to
have been observed with any of the total collagen measurements. Therefore,
we can form an *exact* null distribution by calculating the statistic under
each possible pairing of elements between ``x`` and ``y``.

>>> def statistic(x):  # explore all possible pairings by permuting `x`
...     rs = stats.spearmanr(x, y).statistic  # ignore pvalue
...     transformed = rs * np.sqrt(dof / ((rs+1.0)*(1.0-rs)))
...     return transformed
>>> ref = stats.permutation_test((x,), statistic, alternative='greater',
...                              permutation_type='pairings')
>>> fig, ax = plt.subplots(figsize=(8, 5))
>>> plot(ax)
>>> ax.hist(ref.null_distribution, np.linspace(-5, 5, 26),
...         density=True)
>>> ax.legend(['aymptotic approximation\n(many observations)',
...            f'exact \n({len(ref.null_distribution)} permutations)'])
>>> plt.show()
>>> ref.pvalue
0.04563492063492063  # exact one-sided p-value
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l05169}{5169}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l06972}{\+\_\+ttest\+\_\+finish()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_aa1e457e8eecfa8d5e63416bd9fbf04af}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!tiecorrect@{tiecorrect}}
\index{tiecorrect@{tiecorrect}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{tiecorrect()}{tiecorrect()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_aa1e457e8eecfa8d5e63416bd9fbf04af} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+tiecorrect (\begin{DoxyParamCaption}\item[{}]{rankvals}{}\end{DoxyParamCaption})}

\begin{DoxyVerb}Tie correction factor for Mann-Whitney U and Kruskal-Wallis H tests.

Parameters
----------
rankvals : array_like
    A 1-D sequence of ranks.  Typically this will be the array
    returned by `~scipy.stats.rankdata`.

Returns
-------
factor : float
    Correction factor for U or H.

See Also
--------
rankdata : Assign ranks to the data
mannwhitneyu : Mann-Whitney rank test
kruskal : Kruskal-Wallis H test

References
----------
.. [1] Siegel, S. (1956) Nonparametric Statistics for the Behavioral
       Sciences.  New York: McGraw-Hill.

Examples
--------
>>> from scipy.stats import tiecorrect, rankdata
>>> tiecorrect([1, 2.5, 2.5, 4])
0.9
>>> ranks = rankdata([1, 3, 2, 4, 5, 7, 2, 8, 4])
>>> ranks
array([ 1. ,  4. ,  2.5,  5.5,  7. ,  8. ,  2.5,  9. ,  5.5])
>>> tiecorrect(ranks)
0.9833333333333333
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l09084}{9084}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l09215}{kruskal()}}, \mbox{\hyperlink{scipy_2stats_2tests_2test__rank_8py_source_l00031}{scipy.\+stats.\+tests.\+test\+\_\+rank.\+Test\+Tie\+Correct.\+test\+\_\+basic()}}, \mbox{\hyperlink{scipy_2stats_2tests_2test__rank_8py_source_l00010}{scipy.\+stats.\+tests.\+test\+\_\+rank.\+Test\+Tie\+Correct.\+test\+\_\+empty()}}, \mbox{\hyperlink{scipy_2stats_2tests_2test__rank_8py_source_l00022}{scipy.\+stats.\+tests.\+test\+\_\+rank.\+Test\+Tie\+Correct.\+test\+\_\+no\+\_\+correction()}}, and \mbox{\hyperlink{scipy_2stats_2tests_2test__rank_8py_source_l00016}{scipy.\+stats.\+tests.\+test\+\_\+rank.\+Test\+Tie\+Correct.\+test\+\_\+one()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a3a2094871a926390297df45c2c23902e}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!tmax@{tmax}}
\index{tmax@{tmax}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{tmax()}{tmax()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a3a2094871a926390297df45c2c23902e} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+tmax (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{upperlimit}{ = {\ttfamily None}, }\item[{}]{axis}{ = {\ttfamily 0}, }\item[{}]{inclusive}{ = {\ttfamily \mbox{\hyperlink{classTrue}{True}}}, }\item[{}]{nan\+\_\+policy}{ = {\ttfamily \textquotesingle{}propagate\textquotesingle{}}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Compute the trimmed maximum.

This function computes the maximum value of an array along a given axis,
while ignoring values larger than a specified upper limit.

Parameters
----------
a : array_like
    Array of values.
upperlimit : None or float, optional
    Values in the input array greater than the given limit will be ignored.
    When upperlimit is None, then all values are used. The default value
    is None.
axis : int or None, optional
    Axis along which to operate. Default is 0. If None, compute over the
    whole array `a`.
inclusive : {True, False}, optional
    This flag determines whether values exactly equal to the upper limit
    are included.  The default value is True.
nan_policy : {'propagate', 'raise', 'omit'}, optional
    Defines how to handle when input contains nan.
    The following options are available (default is 'propagate'):

      * 'propagate': returns nan
      * 'raise': throws an error
      * 'omit': performs the calculations ignoring nan values

Returns
-------
tmax : float, int or ndarray
    Trimmed maximum.

Examples
--------
>>> import numpy as np
>>> from scipy import stats
>>> x = np.arange(20)
>>> stats.tmax(x)
19

>>> stats.tmax(x, 13)
13

>>> stats.tmax(x, 13, inclusive=False)
12
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l00736}{736}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l00522}{\+\_\+mask\+\_\+to\+\_\+limits()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a429c8fac38c7f007f95416b907858513}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!tmean@{tmean}}
\index{tmean@{tmean}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{tmean()}{tmean()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a429c8fac38c7f007f95416b907858513} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+tmean (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{limits}{ = {\ttfamily None}, }\item[{}]{inclusive}{ = {\ttfamily (\mbox{\hyperlink{classTrue}{True}},~\mbox{\hyperlink{classTrue}{True}})}, }\item[{}]{axis}{ = {\ttfamily None}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Compute the trimmed mean.

This function finds the arithmetic mean of given values, ignoring values
outside the given `limits`.

Parameters
----------
a : array_like
    Array of values.
limits : None or (lower limit, upper limit), optional
    Values in the input array less than the lower limit or greater than the
    upper limit will be ignored.  When limits is None (default), then all
    values are used.  Either of the limit values in the tuple can also be
    None representing a half-open interval.
inclusive : (bool, bool), optional
    A tuple consisting of the (lower flag, upper flag).  These flags
    determine whether values exactly equal to the lower or upper limits
    are included.  The default value is (True, True).
axis : int or None, optional
    Axis along which to compute test. Default is None.

Returns
-------
tmean : ndarray
    Trimmed mean.

See Also
--------
trim_mean : Returns mean after trimming a proportion from both tails.

Examples
--------
>>> import numpy as np
>>> from scipy import stats
>>> x = np.arange(20)
>>> stats.tmean(x)
9.5
>>> stats.tmean(x, (3,17))
10.0
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l00568}{568}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l00522}{\+\_\+mask\+\_\+to\+\_\+limits()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_ad5fced7efb0eda0d00485572e5cd317d}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!tmin@{tmin}}
\index{tmin@{tmin}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{tmin()}{tmin()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_ad5fced7efb0eda0d00485572e5cd317d} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+tmin (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{lowerlimit}{ = {\ttfamily None}, }\item[{}]{axis}{ = {\ttfamily 0}, }\item[{}]{inclusive}{ = {\ttfamily \mbox{\hyperlink{classTrue}{True}}}, }\item[{}]{nan\+\_\+policy}{ = {\ttfamily \textquotesingle{}propagate\textquotesingle{}}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Compute the trimmed minimum.

This function finds the miminum value of an array `a` along the
specified axis, but only considering values greater than a specified
lower limit.

Parameters
----------
a : array_like
    Array of values.
lowerlimit : None or float, optional
    Values in the input array less than the given limit will be ignored.
    When lowerlimit is None, then all values are used. The default value
    is None.
axis : int or None, optional
    Axis along which to operate. Default is 0. If None, compute over the
    whole array `a`.
inclusive : {True, False}, optional
    This flag determines whether values exactly equal to the lower limit
    are included.  The default value is True.
nan_policy : {'propagate', 'raise', 'omit'}, optional
    Defines how to handle when input contains nan.
    The following options are available (default is 'propagate'):

      * 'propagate': returns nan
      * 'raise': throws an error
      * 'omit': performs the calculations ignoring nan values

Returns
-------
tmin : float, int or ndarray
    Trimmed minimum.

Examples
--------
>>> import numpy as np
>>> from scipy import stats
>>> x = np.arange(20)
>>> stats.tmin(x)
0

>>> stats.tmin(x, 13)
13

>>> stats.tmin(x, 13, inclusive=False)
14
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l00673}{673}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l00522}{\+\_\+mask\+\_\+to\+\_\+limits()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a133a4a1cf3127cc59ab529a6342e4e3b}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!trim1@{trim1}}
\index{trim1@{trim1}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{trim1()}{trim1()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a133a4a1cf3127cc59ab529a6342e4e3b} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+trim1 (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{proportiontocut}{, }\item[{}]{tail}{ = {\ttfamily \textquotesingle{}right\textquotesingle{}}, }\item[{}]{axis}{ = {\ttfamily 0}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Slice off a proportion from ONE end of the passed array distribution.

If `proportiontocut` = 0.1, slices off 'leftmost' or 'rightmost'
10% of scores. The lowest or highest values are trimmed (depending on
the tail).
Slice off less if proportion results in a non-integer slice index
(i.e. conservatively slices off `proportiontocut` ).

Parameters
----------
a : array_like
    Input array.
proportiontocut : float
    Fraction to cut off of 'left' or 'right' of distribution.
tail : {'left', 'right'}, optional
    Defaults to 'right'.
axis : int or None, optional
    Axis along which to trim data. Default is 0. If None, compute over
    the whole array `a`.

Returns
-------
trim1 : ndarray
    Trimmed version of array `a`. The order of the trimmed content is
    undefined.

Examples
--------
Create an array of 10 values and trim 20% of its lowest values:

>>> import numpy as np
>>> from scipy import stats
>>> a = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
>>> stats.trim1(a, 0.2, 'left')
array([2, 4, 3, 5, 6, 7, 8, 9])

Note that the elements of the input array are trimmed by value, but the
output array is not necessarily sorted.

The proportion to trim is rounded down to the nearest integer. For
instance, trimming 25% of the values from an array of 10 values will
return an array of 8 values:

>>> b = np.arange(10)
>>> stats.trim1(b, 1/4).shape
(8,)

Multidimensional arrays can be trimmed along any axis or across the entire
array:

>>> c = [2, 4, 6, 8, 0, 1, 3, 5, 7, 9]
>>> d = np.array([a, b, c])
>>> stats.trim1(d, 0.8, axis=0).shape
(1, 10)
>>> stats.trim1(d, 0.8, axis=1).shape
(3, 2)
>>> stats.trim1(d, 0.8, axis=None).shape
(6,)
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l03803}{3803}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a692b1677d3e31fb991495d59a7378aff}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!trim\_mean@{trim\_mean}}
\index{trim\_mean@{trim\_mean}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{trim\_mean()}{trim\_mean()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a692b1677d3e31fb991495d59a7378aff} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+trim\+\_\+mean (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{proportiontocut}{, }\item[{}]{axis}{ = {\ttfamily 0}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Return mean of array after trimming distribution from both tails.

If `proportiontocut` = 0.1, slices off 'leftmost' and 'rightmost' 10% of
scores. The input is sorted before slicing. Slices off less if proportion
results in a non-integer slice index (i.e., conservatively slices off
`proportiontocut` ).

Parameters
----------
a : array_like
    Input array.
proportiontocut : float
    Fraction to cut off of both tails of the distribution.
axis : int or None, optional
    Axis along which the trimmed means are computed. Default is 0.
    If None, compute over the whole array `a`.

Returns
-------
trim_mean : ndarray
    Mean of trimmed array.

See Also
--------
trimboth
tmean : Compute the trimmed mean ignoring values outside given `limits`.

Examples
--------
>>> import numpy as np
>>> from scipy import stats
>>> x = np.arange(20)
>>> stats.trim_mean(x, 0.1)
9.5
>>> x2 = x.reshape(5, 4)
>>> x2
array([[ 0,  1,  2,  3],
       [ 4,  5,  6,  7],
       [ 8,  9, 10, 11],
       [12, 13, 14, 15],
       [16, 17, 18, 19]])
>>> stats.trim_mean(x2, 0.25)
array([  8.,   9.,  10.,  11.])
>>> stats.trim_mean(x2, 0.25, axis=1)
array([  1.5,   5.5,   9.5,  13.5,  17.5])
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l03890}{3890}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l07482}{\+\_\+ttest\+\_\+trim\+\_\+var\+\_\+mean\+\_\+len()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_af1b232b8ef2a078a8bda0f808ca871b0}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!trimboth@{trimboth}}
\index{trimboth@{trimboth}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{trimboth()}{trimboth()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_af1b232b8ef2a078a8bda0f808ca871b0} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+trimboth (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{proportiontocut}{, }\item[{}]{axis}{ = {\ttfamily 0}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Slice off a proportion of items from both ends of an array.

Slice off the passed proportion of items from both ends of the passed
array (i.e., with `proportiontocut` = 0.1, slices leftmost 10% **and**
rightmost 10% of scores). The trimmed values are the lowest and
highest ones.
Slice off less if proportion results in a non-integer slice index (i.e.
conservatively slices off `proportiontocut`).

Parameters
----------
a : array_like
    Data to trim.
proportiontocut : float
    Proportion (in range 0-1) of total data set to trim of each end.
axis : int or None, optional
    Axis along which to trim data. Default is 0. If None, compute over
    the whole array `a`.

Returns
-------
out : ndarray
    Trimmed version of array `a`. The order of the trimmed content
    is undefined.

See Also
--------
trim_mean

Examples
--------
Create an array of 10 values and trim 10% of those values from each end:

>>> import numpy as np
>>> from scipy import stats
>>> a = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
>>> stats.trimboth(a, 0.1)
array([1, 3, 2, 4, 5, 6, 7, 8])

Note that the elements of the input array are trimmed by value, but the
output array is not necessarily sorted.

The proportion to trim is rounded down to the nearest integer. For
instance, trimming 25% of the values from each end of an array of 10
values will return an array of 6 values:

>>> b = np.arange(10)
>>> stats.trimboth(b, 1/4).shape
(6,)

Multidimensional arrays can be trimmed along any axis or across the entire
array:

>>> c = [2, 4, 6, 8, 0, 1, 3, 5, 7, 9]
>>> d = np.array([a, b, c])
>>> stats.trimboth(d, 0.4, axis=0).shape
(1, 10)
>>> stats.trimboth(d, 0.4, axis=1).shape
(3, 2)
>>> stats.trimboth(d, 0.4, axis=None).shape
(6,)
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l03717}{3717}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_ad31c6b6ef9f91a6dfc8e174a87c0e190}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!tsem@{tsem}}
\index{tsem@{tsem}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{tsem()}{tsem()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_ad31c6b6ef9f91a6dfc8e174a87c0e190} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+tsem (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{limits}{ = {\ttfamily None}, }\item[{}]{inclusive}{ = {\ttfamily (\mbox{\hyperlink{classTrue}{True}},~\mbox{\hyperlink{classTrue}{True}})}, }\item[{}]{axis}{ = {\ttfamily 0}, }\item[{}]{ddof}{ = {\ttfamily 1}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Compute the trimmed standard error of the mean.

This function finds the standard error of the mean for given
values, ignoring values outside the given `limits`.

Parameters
----------
a : array_like
    Array of values.
limits : None or (lower limit, upper limit), optional
    Values in the input array less than the lower limit or greater than the
    upper limit will be ignored. When limits is None, then all values are
    used. Either of the limit values in the tuple can also be None
    representing a half-open interval.  The default value is None.
inclusive : (bool, bool), optional
    A tuple consisting of the (lower flag, upper flag).  These flags
    determine whether values exactly equal to the lower or upper limits
    are included.  The default value is (True, True).
axis : int or None, optional
    Axis along which to operate. Default is 0. If None, compute over the
    whole array `a`.
ddof : int, optional
    Delta degrees of freedom.  Default is 1.

Returns
-------
tsem : float
    Trimmed standard error of the mean.

Notes
-----
`tsem` uses unbiased sample standard deviation, i.e. it uses a
correction factor ``n / (n - 1)``.

Examples
--------
>>> import numpy as np
>>> from scipy import stats
>>> x = np.arange(20)
>>> stats.tsem(x)
1.3228756555322954
>>> stats.tsem(x, (3,17))
1.1547005383792515
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l00847}{847}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l00522}{\+\_\+mask\+\_\+to\+\_\+limits()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a556a06abd7478c8d2c7500b2992c51c3}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!tstd@{tstd}}
\index{tstd@{tstd}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{tstd()}{tstd()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a556a06abd7478c8d2c7500b2992c51c3} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+tstd (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{limits}{ = {\ttfamily None}, }\item[{}]{inclusive}{ = {\ttfamily (\mbox{\hyperlink{classTrue}{True}},~\mbox{\hyperlink{classTrue}{True}})}, }\item[{}]{axis}{ = {\ttfamily 0}, }\item[{}]{ddof}{ = {\ttfamily 1}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Compute the trimmed sample standard deviation.

This function finds the sample standard deviation of given values,
ignoring values outside the given `limits`.

Parameters
----------
a : array_like
    Array of values.
limits : None or (lower limit, upper limit), optional
    Values in the input array less than the lower limit or greater than the
    upper limit will be ignored. When limits is None, then all values are
    used. Either of the limit values in the tuple can also be None
    representing a half-open interval.  The default value is None.
inclusive : (bool, bool), optional
    A tuple consisting of the (lower flag, upper flag).  These flags
    determine whether values exactly equal to the lower or upper limits
    are included.  The default value is (True, True).
axis : int or None, optional
    Axis along which to operate. Default is 0. If None, compute over the
    whole array `a`.
ddof : int, optional
    Delta degrees of freedom.  Default is 1.

Returns
-------
tstd : float
    Trimmed sample standard deviation.

Notes
-----
`tstd` computes the unbiased sample standard deviation, i.e. it uses a
correction factor ``n / (n - 1)``.

Examples
--------
>>> import numpy as np
>>> from scipy import stats
>>> x = np.arange(20)
>>> stats.tstd(x)
5.9160797830996161
>>> stats.tstd(x, (3,17))
4.4721359549995796
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l00798}{798}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l00618}{tvar()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a0b40cba4553c34430dc8745cc657d107}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!ttest\_1samp@{ttest\_1samp}}
\index{ttest\_1samp@{ttest\_1samp}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{ttest\_1samp()}{ttest\_1samp()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a0b40cba4553c34430dc8745cc657d107} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+ttest\+\_\+1samp (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{popmean}{, }\item[{}]{axis}{ = {\ttfamily 0}, }\item[{}]{nan\+\_\+policy}{ = {\ttfamily \textquotesingle{}propagate\textquotesingle{}}, }\item[{}]{alternative}{ = {\ttfamily "{}two-\/sided"{}}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Calculate the T-test for the mean of ONE group of scores.

This is a test for the null hypothesis that the expected value
(mean) of a sample of independent observations `a` is equal to the given
population mean, `popmean`.

Parameters
----------
a : array_like
    Sample observation.
popmean : float or array_like
    Expected value in null hypothesis. If array_like, then its length along
    `axis` must equal 1, and it must otherwise be broadcastable with `a`.
axis : int or None, optional
    Axis along which to compute test; default is 0. If None, compute over
    the whole array `a`.
nan_policy : {'propagate', 'raise', 'omit'}, optional
    Defines how to handle when input contains nan.
    The following options are available (default is 'propagate'):

      * 'propagate': returns nan
      * 'raise': throws an error
      * 'omit': performs the calculations ignoring nan values

alternative : {'two-sided', 'less', 'greater'}, optional
    Defines the alternative hypothesis.
    The following options are available (default is 'two-sided'):

    * 'two-sided': the mean of the underlying distribution of the sample
      is different than the given population mean (`popmean`)
    * 'less': the mean of the underlying distribution of the sample is
      less than the given population mean (`popmean`)
    * 'greater': the mean of the underlying distribution of the sample is
      greater than the given population mean (`popmean`)

Returns
-------
result : `~scipy.stats._result_classes.TtestResult`
    An object with the following attributes:

    statistic : float or array
        The t-statistic.
    pvalue : float or array
        The p-value associated with the given alternative.
    df : float or array
        The number of degrees of freedom used in calculation of the
        t-statistic; this is one less than the size of the sample
        (``a.shape[axis]``).

        .. versionadded:: 1.10.0

    The object also has the following method:

    confidence_interval(confidence_level=0.95)
        Computes a confidence interval around the population
        mean for the given confidence level.
        The confidence interval is returned in a ``namedtuple`` with
        fields `low` and `high`.

        .. versionadded:: 1.10.0

Notes
-----
The statistic is calculated as ``(np.mean(a) - popmean)/se``, where
``se`` is the standard error. Therefore, the statistic will be positive
when the sample mean is greater than the population mean and negative when
the sample mean is less than the population mean.

Examples
--------
Suppose we wish to test the null hypothesis that the mean of a population
is equal to 0.5. We choose a confidence level of 99%; that is, we will
reject the null hypothesis in favor of the alternative if the p-value is
less than 0.01.

When testing random variates from the standard uniform distribution, which
has a mean of 0.5, we expect the data to be consistent with the null
hypothesis most of the time.

>>> import numpy as np
>>> from scipy import stats
>>> rng = np.random.default_rng()
>>> rvs = stats.uniform.rvs(size=50, random_state=rng)
>>> stats.ttest_1samp(rvs, popmean=0.5)
TtestResult(statistic=2.456308468440, pvalue=0.017628209047638, df=49)

As expected, the p-value of 0.017 is not below our threshold of 0.01, so
we cannot reject the null hypothesis.

When testing data from the standard *normal* distribution, which has a mean
of 0, we would expect the null hypothesis to be rejected.

>>> rvs = stats.norm.rvs(size=50, random_state=rng)
>>> stats.ttest_1samp(rvs, popmean=0.5)
TtestResult(statistic=-7.433605518875, pvalue=1.416760157221e-09, df=49)

Indeed, the p-value is lower than our threshold of 0.01, so we reject the
null hypothesis in favor of the default "two-sided" alternative: the mean
of the population is *not* equal to 0.5.

However, suppose we were to test the null hypothesis against the
one-sided alternative that the mean of the population is *greater* than
0.5. Since the mean of the standard normal is less than 0.5, we would not
expect the null hypothesis to be rejected.

>>> stats.ttest_1samp(rvs, popmean=0.5, alternative='greater')
TtestResult(statistic=-7.433605518875, pvalue=0.99999999929, df=49)

Unsurprisingly, with a p-value greater than our threshold, we would not
reject the null hypothesis.

Note that when working with a confidence level of 99%, a true null
hypothesis will be rejected approximately 1% of the time.

>>> rvs = stats.uniform.rvs(size=(100, 50), random_state=rng)
>>> res = stats.ttest_1samp(rvs, popmean=0.5, axis=1)
>>> np.sum(res.pvalue < 0.01)
1

Indeed, even though all 100 samples above were drawn from the standard
uniform distribution, which *does* have a population mean of 0.5, we would
mistakenly reject the null hypothesis for one of them.

`ttest_1samp` can also compute a confidence interval around the population
mean.

>>> rvs = stats.norm.rvs(size=50, random_state=rng)
>>> res = stats.ttest_1samp(rvs, popmean=0)
>>> ci = res.confidence_interval(confidence_level=0.95)
>>> ci
ConfidenceInterval(low=-0.3193887540880017, high=0.2898583388980972)

The bounds of the 95% confidence interval are the
minimum and maximum values of the parameter `popmean` for which the
p-value of the test would be 0.05.

>>> res = stats.ttest_1samp(rvs, popmean=ci.low)
>>> np.testing.assert_allclose(res.pvalue, 0.05)
>>> res = stats.ttest_1samp(rvs, popmean=ci.high)
>>> np.testing.assert_allclose(res.pvalue, 0.05)

Under certain assumptions about the population from which a sample
is drawn, the confidence interval with confidence level 95% is expected
to contain the true population mean in 95% of sample replications.

>>> rvs = stats.norm.rvs(size=(50, 1000), loc=1, random_state=rng)
>>> res = stats.ttest_1samp(rvs, popmean=0)
>>> ci = res.confidence_interval()
>>> contains_pop_mean = (ci.low < 1) & (ci.high > 1)
>>> contains_pop_mean.sum()
953
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l06765}{6765}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l06972}{\+\_\+ttest\+\_\+finish()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a1021af752b49ceb9383493994d38eaef}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!ttest\_ind@{ttest\_ind}}
\index{ttest\_ind@{ttest\_ind}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{ttest\_ind()}{ttest\_ind()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a1021af752b49ceb9383493994d38eaef} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+ttest\+\_\+ind (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{b}{, }\item[{}]{axis}{ = {\ttfamily 0}, }\item[{}]{equal\+\_\+var}{ = {\ttfamily \mbox{\hyperlink{classTrue}{True}}}, }\item[{}]{nan\+\_\+policy}{ = {\ttfamily \textquotesingle{}propagate\textquotesingle{}}, }\item[{}]{permutations}{ = {\ttfamily None}, }\item[{}]{random\+\_\+state}{ = {\ttfamily None}, }\item[{}]{alternative}{ = {\ttfamily "{}two-\/sided"{}}, }\item[{}]{trim}{ = {\ttfamily 0}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Calculate the T-test for the means of *two independent* samples of scores.

This is a test for the null hypothesis that 2 independent samples
have identical average (expected) values. This test assumes that the
populations have identical variances by default.

Parameters
----------
a, b : array_like
    The arrays must have the same shape, except in the dimension
    corresponding to `axis` (the first, by default).
axis : int or None, optional
    Axis along which to compute test. If None, compute over the whole
    arrays, `a`, and `b`.
equal_var : bool, optional
    If True (default), perform a standard independent 2 sample test
    that assumes equal population variances [1]_.
    If False, perform Welch's t-test, which does not assume equal
    population variance [2]_.

    .. versionadded:: 0.11.0

nan_policy : {'propagate', 'raise', 'omit'}, optional
    Defines how to handle when input contains nan.
    The following options are available (default is 'propagate'):

      * 'propagate': returns nan
      * 'raise': throws an error
      * 'omit': performs the calculations ignoring nan values

    The 'omit' option is not currently available for permutation tests or
    one-sided asympyotic tests.

permutations : non-negative int, np.inf, or None (default), optional
    If 0 or None (default), use the t-distribution to calculate p-values.
    Otherwise, `permutations` is  the number of random permutations that
    will be used to estimate p-values using a permutation test. If
    `permutations` equals or exceeds the number of distinct partitions of
    the pooled data, an exact test is performed instead (i.e. each
    distinct partition is used exactly once). See Notes for details.

    .. versionadded:: 1.7.0

random_state : {None, int, `numpy.random.Generator`,
        `numpy.random.RandomState`}, optional

    If `seed` is None (or `np.random`), the `numpy.random.RandomState`
    singleton is used.
    If `seed` is an int, a new ``RandomState`` instance is used,
    seeded with `seed`.
    If `seed` is already a ``Generator`` or ``RandomState`` instance then
    that instance is used.

    Pseudorandom number generator state used to generate permutations
    (used only when `permutations` is not None).

    .. versionadded:: 1.7.0

alternative : {'two-sided', 'less', 'greater'}, optional
    Defines the alternative hypothesis.
    The following options are available (default is 'two-sided'):

    * 'two-sided': the means of the distributions underlying the samples
      are unequal.
    * 'less': the mean of the distribution underlying the first sample
      is less than the mean of the distribution underlying the second
      sample.
    * 'greater': the mean of the distribution underlying the first
      sample is greater than the mean of the distribution underlying
      the second sample.

    .. versionadded:: 1.6.0

trim : float, optional
    If nonzero, performs a trimmed (Yuen's) t-test.
    Defines the fraction of elements to be trimmed from each end of the
    input samples. If 0 (default), no elements will be trimmed from either
    side. The number of trimmed elements from each tail is the floor of the
    trim times the number of elements. Valid range is [0, .5).

    .. versionadded:: 1.7

Returns
-------
result : `~scipy.stats._result_classes.TtestResult`
    An object with the following attributes:

    statistic : float or ndarray
        The t-statistic.
    pvalue : float or ndarray
        The p-value associated with the given alternative.
    df : float or ndarray
        The number of degrees of freedom used in calculation of the
        t-statistic. This is always NaN for a permutation t-test.

        .. versionadded:: 1.11.0

    The object also has the following method:

    confidence_interval(confidence_level=0.95)
        Computes a confidence interval around the difference in
        population means for the given confidence level.
        The confidence interval is returned in a ``namedtuple`` with
        fields ``low`` and ``high``.
        When a permutation t-test is performed, the confidence interval
        is not computed, and fields ``low`` and ``high`` contain NaN.

        .. versionadded:: 1.11.0

Notes
-----
Suppose we observe two independent samples, e.g. flower petal lengths, and
we are considering whether the two samples were drawn from the same
population (e.g. the same species of flower or two species with similar
petal characteristics) or two different populations.

The t-test quantifies the difference between the arithmetic means
of the two samples. The p-value quantifies the probability of observing
as or more extreme values assuming the null hypothesis, that the
samples are drawn from populations with the same population means, is true.
A p-value larger than a chosen threshold (e.g. 5% or 1%) indicates that
our observation is not so unlikely to have occurred by chance. Therefore,
we do not reject the null hypothesis of equal population means.
If the p-value is smaller than our threshold, then we have evidence
against the null hypothesis of equal population means.

By default, the p-value is determined by comparing the t-statistic of the
observed data against a theoretical t-distribution.
When ``1 < permutations < binom(n, k)``, where

* ``k`` is the number of observations in `a`,
* ``n`` is the total number of observations in `a` and `b`, and
* ``binom(n, k)`` is the binomial coefficient (``n`` choose ``k``),

the data are pooled (concatenated), randomly assigned to either group `a`
or `b`, and the t-statistic is calculated. This process is performed
repeatedly (`permutation` times), generating a distribution of the
t-statistic under the null hypothesis, and the t-statistic of the observed
data is compared to this distribution to determine the p-value.
Specifically, the p-value reported is the "achieved significance level"
(ASL) as defined in 4.4 of [3]_. Note that there are other ways of
estimating p-values using randomized permutation tests; for other
options, see the more general `permutation_test`.

When ``permutations >= binom(n, k)``, an exact test is performed: the data
are partitioned between the groups in each distinct way exactly once.

The permutation test can be computationally expensive and not necessarily
more accurate than the analytical test, but it does not make strong
assumptions about the shape of the underlying distribution.

Use of trimming is commonly referred to as the trimmed t-test. At times
called Yuen's t-test, this is an extension of Welch's t-test, with the
difference being the use of winsorized means in calculation of the variance
and the trimmed sample size in calculation of the statistic. Trimming is
recommended if the underlying distribution is long-tailed or contaminated
with outliers [4]_.

The statistic is calculated as ``(np.mean(a) - np.mean(b))/se``, where
``se`` is the standard error. Therefore, the statistic will be positive
when the sample mean of `a` is greater than the sample mean of `b` and
negative when the sample mean of `a` is less than the sample mean of
`b`.

References
----------
.. [1] https://en.wikipedia.org/wiki/T-test#Independent_two-sample_t-test

.. [2] https://en.wikipedia.org/wiki/Welch%27s_t-test

.. [3] B. Efron and T. Hastie. Computer Age Statistical Inference. (2016).

.. [4] Yuen, Karen K. "The Two-Sample Trimmed t for Unequal Population
       Variances." Biometrika, vol. 61, no. 1, 1974, pp. 165-170. JSTOR,
       www.jstor.org/stable/2334299. Accessed 30 Mar. 2021.

.. [5] Yuen, Karen K., and W. J. Dixon. "The Approximate Behaviour and
       Performance of the Two-Sample Trimmed t." Biometrika, vol. 60,
       no. 2, 1973, pp. 369-374. JSTOR, www.jstor.org/stable/2334550.
       Accessed 30 Mar. 2021.

Examples
--------
>>> import numpy as np
>>> from scipy import stats
>>> rng = np.random.default_rng()

Test with sample with identical means:

>>> rvs1 = stats.norm.rvs(loc=5, scale=10, size=500, random_state=rng)
>>> rvs2 = stats.norm.rvs(loc=5, scale=10, size=500, random_state=rng)
>>> stats.ttest_ind(rvs1, rvs2)
Ttest_indResult(statistic=-0.4390847099199348, pvalue=0.6606952038870015)
>>> stats.ttest_ind(rvs1, rvs2, equal_var=False)
Ttest_indResult(statistic=-0.4390847099199348, pvalue=0.6606952553131064)

`ttest_ind` underestimates p for unequal variances:

>>> rvs3 = stats.norm.rvs(loc=5, scale=20, size=500, random_state=rng)
>>> stats.ttest_ind(rvs1, rvs3)
Ttest_indResult(statistic=-1.6370984482905417, pvalue=0.1019251574705033)
>>> stats.ttest_ind(rvs1, rvs3, equal_var=False)
Ttest_indResult(statistic=-1.637098448290542, pvalue=0.10202110497954867)

When ``n1 != n2``, the equal variance t-statistic is no longer equal to the
unequal variance t-statistic:

>>> rvs4 = stats.norm.rvs(loc=5, scale=20, size=100, random_state=rng)
>>> stats.ttest_ind(rvs1, rvs4)
Ttest_indResult(statistic=-1.9481646859513422, pvalue=0.05186270935842703)
>>> stats.ttest_ind(rvs1, rvs4, equal_var=False)
Ttest_indResult(statistic=-1.3146566100751664, pvalue=0.1913495266513811)

T-test with different means, variance, and n:

>>> rvs5 = stats.norm.rvs(loc=8, scale=20, size=100, random_state=rng)
>>> stats.ttest_ind(rvs1, rvs5)
Ttest_indResult(statistic=-2.8415950600298774, pvalue=0.0046418707568707885)
>>> stats.ttest_ind(rvs1, rvs5, equal_var=False)
Ttest_indResult(statistic=-1.8686598649188084, pvalue=0.06434714193919686)

When performing a permutation test, more permutations typically yields
more accurate results. Use a ``np.random.Generator`` to ensure
reproducibility:

>>> stats.ttest_ind(rvs1, rvs5, permutations=10000,
...                 random_state=rng)
Ttest_indResult(statistic=-2.8415950600298774, pvalue=0.0052994700529947)

Take these two samples, one of which has an extreme tail.

>>> a = (56, 128.6, 12, 123.8, 64.34, 78, 763.3)
>>> b = (1.1, 2.9, 4.2)

Use the `trim` keyword to perform a trimmed (Yuen) t-test. For example,
using 20% trimming, ``trim=.2``, the test will reduce the impact of one
(``np.floor(trim*len(a))``) element from each tail of sample `a`. It will
have no effect on sample `b` because ``np.floor(trim*len(b))`` is 0.

>>> stats.ttest_ind(a, b, trim=.2)
Ttest_indResult(statistic=3.4463884028073513,
                pvalue=0.01369338726499547)
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l07176}{7176}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l07598}{\+\_\+permutation\+\_\+ttest()}}, and \mbox{\hyperlink{__stats__py_8py_source_l07482}{\+\_\+ttest\+\_\+trim\+\_\+var\+\_\+mean\+\_\+len()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a3b98b9c4e131fb2b6d72da0ec90092f9}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!ttest\_ind\_from\_stats@{ttest\_ind\_from\_stats}}
\index{ttest\_ind\_from\_stats@{ttest\_ind\_from\_stats}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{ttest\_ind\_from\_stats()}{ttest\_ind\_from\_stats()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a3b98b9c4e131fb2b6d72da0ec90092f9} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+ttest\+\_\+ind\+\_\+from\+\_\+stats (\begin{DoxyParamCaption}\item[{}]{mean1}{, }\item[{}]{std1}{, }\item[{}]{nobs1}{, }\item[{}]{mean2}{, }\item[{}]{std2}{, }\item[{}]{nobs2}{, }\item[{}]{equal\+\_\+var}{ = {\ttfamily \mbox{\hyperlink{classTrue}{True}}}, }\item[{}]{alternative}{ = {\ttfamily "{}two-\/sided"{}}}\end{DoxyParamCaption})}

\begin{DoxyVerb}T-test for means of two independent samples from descriptive statistics.

This is a test for the null hypothesis that two independent
samples have identical average (expected) values.

Parameters
----------
mean1 : array_like
    The mean(s) of sample 1.
std1 : array_like
    The corrected sample standard deviation of sample 1 (i.e. ``ddof=1``).
nobs1 : array_like
    The number(s) of observations of sample 1.
mean2 : array_like
    The mean(s) of sample 2.
std2 : array_like
    The corrected sample standard deviation of sample 2 (i.e. ``ddof=1``).
nobs2 : array_like
    The number(s) of observations of sample 2.
equal_var : bool, optional
    If True (default), perform a standard independent 2 sample test
    that assumes equal population variances [1]_.
    If False, perform Welch's t-test, which does not assume equal
    population variance [2]_.
alternative : {'two-sided', 'less', 'greater'}, optional
    Defines the alternative hypothesis.
    The following options are available (default is 'two-sided'):

    * 'two-sided': the means of the distributions are unequal.
    * 'less': the mean of the first distribution is less than the
      mean of the second distribution.
    * 'greater': the mean of the first distribution is greater than the
      mean of the second distribution.

    .. versionadded:: 1.6.0

Returns
-------
statistic : float or array
    The calculated t-statistics.
pvalue : float or array
    The two-tailed p-value.

See Also
--------
scipy.stats.ttest_ind

Notes
-----
The statistic is calculated as ``(mean1 - mean2)/se``, where ``se`` is the
standard error. Therefore, the statistic will be positive when `mean1` is
greater than `mean2` and negative when `mean1` is less than `mean2`.

This method does not check whether any of the elements of `std1` or `std2`
are negative. If any elements of the `std1` or `std2` parameters are
negative in a call to this method, this method will return the same result
as if it were passed ``numpy.abs(std1)`` and ``numpy.abs(std2)``,
respectively, instead; no exceptions or warnings will be emitted.

References
----------
.. [1] https://en.wikipedia.org/wiki/T-test#Independent_two-sample_t-test

.. [2] https://en.wikipedia.org/wiki/Welch%27s_t-test

Examples
--------
Suppose we have the summary data for two samples, as follows (with the
Sample Variance being the corrected sample variance)::

                     Sample   Sample
               Size   Mean   Variance
    Sample 1    13    15.0     87.5
    Sample 2    11    12.0     39.0

Apply the t-test to this data (with the assumption that the population
variances are equal):

>>> import numpy as np
>>> from scipy.stats import ttest_ind_from_stats
>>> ttest_ind_from_stats(mean1=15.0, std1=np.sqrt(87.5), nobs1=13,
...                      mean2=12.0, std2=np.sqrt(39.0), nobs2=11)
Ttest_indResult(statistic=0.9051358093310269, pvalue=0.3751996797581487)

For comparison, here is the data from which those summary statistics
were taken.  With this data, we can compute the same result using
`scipy.stats.ttest_ind`:

>>> a = np.array([1, 3, 4, 6, 11, 13, 15, 19, 22, 24, 25, 26, 26])
>>> b = np.array([2, 4, 6, 9, 11, 13, 14, 15, 18, 19, 21])
>>> from scipy.stats import ttest_ind
>>> ttest_ind(a, b)
Ttest_indResult(statistic=0.905135809331027, pvalue=0.3751996797581486)

Suppose we instead have binary data and would like to apply a t-test to
compare the proportion of 1s in two independent groups::

                      Number of    Sample     Sample
                Size    ones        Mean     Variance
    Sample 1    150      30         0.2        0.161073
    Sample 2    200      45         0.225      0.175251

The sample mean :math:`\hat{p}` is the proportion of ones in the sample
and the variance for a binary observation is estimated by
:math:`\hat{p}(1-\hat{p})`.

>>> ttest_ind_from_stats(mean1=0.2, std1=np.sqrt(0.161073), nobs1=150,
...                      mean2=0.225, std2=np.sqrt(0.175251), nobs2=200)
Ttest_indResult(statistic=-0.5627187905196761, pvalue=0.5739887114209541)

For comparison, we could compute the t statistic and p-value using
arrays of 0s and 1s and `scipy.stat.ttest_ind`, as above.

>>> group1 = np.array([1]*30 + [0]*(150-30))
>>> group2 = np.array([1]*45 + [0]*(200-45))
>>> ttest_ind(group1, group2)
Ttest_indResult(statistic=-0.5627179589855622, pvalue=0.573989277115258)
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l07038}{7038}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a40e1ede0be1d6f44d3aa61aa9e1be55f}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!ttest\_rel@{ttest\_rel}}
\index{ttest\_rel@{ttest\_rel}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{ttest\_rel()}{ttest\_rel()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a40e1ede0be1d6f44d3aa61aa9e1be55f} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+ttest\+\_\+rel (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{b}{, }\item[{}]{axis}{ = {\ttfamily 0}, }\item[{}]{nan\+\_\+policy}{ = {\ttfamily \textquotesingle{}propagate\textquotesingle{}}, }\item[{}]{alternative}{ = {\ttfamily "{}two-\/sided"{}}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Calculate the t-test on TWO RELATED samples of scores, a and b.

This is a test for the null hypothesis that two related or
repeated samples have identical average (expected) values.

Parameters
----------
a, b : array_like
    The arrays must have the same shape.
axis : int or None, optional
    Axis along which to compute test. If None, compute over the whole
    arrays, `a`, and `b`.
nan_policy : {'propagate', 'raise', 'omit'}, optional
    Defines how to handle when input contains nan.
    The following options are available (default is 'propagate'):

      * 'propagate': returns nan
      * 'raise': throws an error
      * 'omit': performs the calculations ignoring nan values
alternative : {'two-sided', 'less', 'greater'}, optional
    Defines the alternative hypothesis.
    The following options are available (default is 'two-sided'):

    * 'two-sided': the means of the distributions underlying the samples
      are unequal.
    * 'less': the mean of the distribution underlying the first sample
      is less than the mean of the distribution underlying the second
      sample.
    * 'greater': the mean of the distribution underlying the first
      sample is greater than the mean of the distribution underlying
      the second sample.

    .. versionadded:: 1.6.0

Returns
-------
result : `~scipy.stats._result_classes.TtestResult`
    An object with the following attributes:

    statistic : float or array
        The t-statistic.
    pvalue : float or array
        The p-value associated with the given alternative.
    df : float or array
        The number of degrees of freedom used in calculation of the
        t-statistic; this is one less than the size of the sample
        (``a.shape[axis]``).

        .. versionadded:: 1.10.0

    The object also has the following method:

    confidence_interval(confidence_level=0.95)
        Computes a confidence interval around the difference in
        population means for the given confidence level.
        The confidence interval is returned in a ``namedtuple`` with
        fields `low` and `high`.

        .. versionadded:: 1.10.0

Notes
-----
Examples for use are scores of the same set of student in
different exams, or repeated sampling from the same units. The
test measures whether the average score differs significantly
across samples (e.g. exams). If we observe a large p-value, for
example greater than 0.05 or 0.1 then we cannot reject the null
hypothesis of identical average scores. If the p-value is smaller
than the threshold, e.g. 1%, 5% or 10%, then we reject the null
hypothesis of equal averages. Small p-values are associated with
large t-statistics.

The t-statistic is calculated as ``np.mean(a - b)/se``, where ``se`` is the
standard error. Therefore, the t-statistic will be positive when the sample
mean of ``a - b`` is greater than zero and negative when the sample mean of
``a - b`` is less than zero.

References
----------
https://en.wikipedia.org/wiki/T-test#Dependent_t-test_for_paired_samples

Examples
--------
>>> import numpy as np
>>> from scipy import stats
>>> rng = np.random.default_rng()

>>> rvs1 = stats.norm.rvs(loc=5, scale=10, size=500, random_state=rng)
>>> rvs2 = (stats.norm.rvs(loc=5, scale=10, size=500, random_state=rng)
...         + stats.norm.rvs(scale=0.2, size=500, random_state=rng))
>>> stats.ttest_rel(rvs1, rvs2)
TtestResult(statistic=-0.4549717054410304, pvalue=0.6493274702088672, df=499)  # noqa
>>> rvs3 = (stats.norm.rvs(loc=8, scale=10, size=500, random_state=rng)
...         + stats.norm.rvs(scale=0.2, size=500, random_state=rng))
>>> stats.ttest_rel(rvs1, rvs3)
TtestResult(statistic=-5.879467544540889, pvalue=7.540777129099917e-09, df=499)  # noqa
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l07682}{7682}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l06972}{\+\_\+ttest\+\_\+finish()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a26ec33f46d65449e3c4e90f85e6f3928}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!tvar@{tvar}}
\index{tvar@{tvar}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{tvar()}{tvar()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a26ec33f46d65449e3c4e90f85e6f3928} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+tvar (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{limits}{ = {\ttfamily None}, }\item[{}]{inclusive}{ = {\ttfamily (\mbox{\hyperlink{classTrue}{True}},~\mbox{\hyperlink{classTrue}{True}})}, }\item[{}]{axis}{ = {\ttfamily 0}, }\item[{}]{ddof}{ = {\ttfamily 1}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Compute the trimmed variance.

This function computes the sample variance of an array of values,
while ignoring values which are outside of given `limits`.

Parameters
----------
a : array_like
    Array of values.
limits : None or (lower limit, upper limit), optional
    Values in the input array less than the lower limit or greater than the
    upper limit will be ignored. When limits is None, then all values are
    used. Either of the limit values in the tuple can also be None
    representing a half-open interval.  The default value is None.
inclusive : (bool, bool), optional
    A tuple consisting of the (lower flag, upper flag).  These flags
    determine whether values exactly equal to the lower or upper limits
    are included.  The default value is (True, True).
axis : int or None, optional
    Axis along which to operate. Default is 0. If None, compute over the
    whole array `a`.
ddof : int, optional
    Delta degrees of freedom.  Default is 1.

Returns
-------
tvar : float
    Trimmed variance.

Notes
-----
`tvar` computes the unbiased sample variance, i.e. it uses a correction
factor ``n / (n - 1)``.

Examples
--------
>>> import numpy as np
>>> from scipy import stats
>>> x = np.arange(20)
>>> stats.tvar(x)
35.0
>>> stats.tvar(x, (3,17))
20.0
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l00618}{618}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l00522}{\+\_\+mask\+\_\+to\+\_\+limits()}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l00798}{tstd()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a0baf8c2dbcc1526d7272efa817c2331c}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!unpack\_TtestResult@{unpack\_TtestResult}}
\index{unpack\_TtestResult@{unpack\_TtestResult}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{unpack\_TtestResult()}{unpack\_TtestResult()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a0baf8c2dbcc1526d7272efa817c2331c} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+unpack\+\_\+\+Ttest\+Result (\begin{DoxyParamCaption}\item[{}]{res}{}\end{DoxyParamCaption})}



Definition at line \mbox{\hyperlink{__stats__py_8py_source_l06758}{6758}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a030f7a94530750520b7f11137b0f88c6}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!wasserstein\_distance@{wasserstein\_distance}}
\index{wasserstein\_distance@{wasserstein\_distance}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{wasserstein\_distance()}{wasserstein\_distance()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a030f7a94530750520b7f11137b0f88c6} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+wasserstein\+\_\+distance (\begin{DoxyParamCaption}\item[{}]{u\+\_\+values}{, }\item[{}]{v\+\_\+values}{, }\item[{}]{u\+\_\+weights}{ = {\ttfamily None}, }\item[{}]{v\+\_\+weights}{ = {\ttfamily None}}\end{DoxyParamCaption})}



STATISTICAL DISTANCES \#. 

\begin{DoxyVerb}    Compute the first Wasserstein distance between two 1D distributions.

    This distance is also known as the earth mover's distance, since it can be
    seen as the minimum amount of "work" required to transform :math:`u` into
    :math:`v`, where "work" is measured as the amount of distribution weight
    that must be moved, multiplied by the distance it has to be moved.

    .. versionadded:: 1.0.0

    Parameters
    ----------
    u_values, v_values : array_like
        Values observed in the (empirical) distribution.
    u_weights, v_weights : array_like, optional
        Weight for each value. If unspecified, each value is assigned the same
        weight.
        `u_weights` (resp. `v_weights`) must have the same length as
        `u_values` (resp. `v_values`). If the weight sum differs from 1, it
        must still be positive and finite so that the weights can be normalized
        to sum to 1.

    Returns
    -------
    distance : float
        The computed distance between the distributions.

    Notes
    -----
    The first Wasserstein distance between the distributions :math:`u` and
    :math:`v` is:

    .. math::

        l_1 (u, v) = \inf_{\pi \in \Gamma (u, v)} \int_{\mathbb{R} \times
        \mathbb{R}} |x-y| \mathrm{d} \pi (x, y)

    where :math:`\Gamma (u, v)` is the set of (probability) distributions on
    :math:`\mathbb{R} \times \mathbb{R}` whose marginals are :math:`u` and
    :math:`v` on the first and second factors respectively.

    If :math:`U` and :math:`V` are the respective CDFs of :math:`u` and
    :math:`v`, this distance also equals to:

    .. math::

        l_1(u, v) = \int_{-\infty}^{+\infty} |U-V|

    See [2]_ for a proof of the equivalence of both definitions.

    The input distributions can be empirical, therefore coming from samples
    whose values are effectively inputs of the function, or they can be seen as
    generalized functions, in which case they are weighted sums of Dirac delta
    functions located at the specified values.

    References
    ----------
    .. [1] "Wasserstein metric", https://en.wikipedia.org/wiki/Wasserstein_metric
    .. [2] Ramdas, Garcia, Cuturi "On Wasserstein Two Sample Testing and Related
           Families of Nonparametric Tests" (2015). :arXiv:`1509.02237`.

    Examples
    --------
    >>> from scipy.stats import wasserstein_distance
    >>> wasserstein_distance([0, 1, 3], [5, 6, 8])
    5.0
    >>> wasserstein_distance([0, 1], [0, 1], [3, 1], [2, 2])
    0.25
    >>> wasserstein_distance([3.4, 3.9, 7.5, 7.8], [4.5, 1.4],
    ...                      [1.4, 0.9, 3.1, 7.2], [3.2, 3.5])
    4.0781331438047861
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l09733}{9733}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l09896}{\+\_\+cdf\+\_\+distance()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a6ba4662b53e915f8ac0f275bb2199d4f}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!weightedtau@{weightedtau}}
\index{weightedtau@{weightedtau}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{weightedtau()}{weightedtau()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a6ba4662b53e915f8ac0f275bb2199d4f} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+weightedtau (\begin{DoxyParamCaption}\item[{}]{x}{, }\item[{}]{y}{, }\item[{}]{rank}{ = {\ttfamily \mbox{\hyperlink{classTrue}{True}}}, }\item[{}]{weigher}{ = {\ttfamily None}, }\item[{}]{additive}{ = {\ttfamily \mbox{\hyperlink{classTrue}{True}}}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Compute a weighted version of Kendall's :math:`\tau`.

The weighted :math:`\tau` is a weighted version of Kendall's
:math:`\tau` in which exchanges of high weight are more influential than
exchanges of low weight. The default parameters compute the additive
hyperbolic version of the index, :math:`\tau_\mathrm h`, which has
been shown to provide the best balance between important and
unimportant elements [1]_.

The weighting is defined by means of a rank array, which assigns a
nonnegative rank to each element (higher importance ranks being
associated with smaller values, e.g., 0 is the highest possible rank),
and a weigher function, which assigns a weight based on the rank to
each element. The weight of an exchange is then the sum or the product
of the weights of the ranks of the exchanged elements. The default
parameters compute :math:`\tau_\mathrm h`: an exchange between
elements with rank :math:`r` and :math:`s` (starting from zero) has
weight :math:`1/(r+1) + 1/(s+1)`.

Specifying a rank array is meaningful only if you have in mind an
external criterion of importance. If, as it usually happens, you do
not have in mind a specific rank, the weighted :math:`\tau` is
defined by averaging the values obtained using the decreasing
lexicographical rank by (`x`, `y`) and by (`y`, `x`). This is the
behavior with default parameters. Note that the convention used
here for ranking (lower values imply higher importance) is opposite
to that used by other SciPy statistical functions.

Parameters
----------
x, y : array_like
    Arrays of scores, of the same shape. If arrays are not 1-D, they will
    be flattened to 1-D.
rank : array_like of ints or bool, optional
    A nonnegative rank assigned to each element. If it is None, the
    decreasing lexicographical rank by (`x`, `y`) will be used: elements of
    higher rank will be those with larger `x`-values, using `y`-values to
    break ties (in particular, swapping `x` and `y` will give a different
    result). If it is False, the element indices will be used
    directly as ranks. The default is True, in which case this
    function returns the average of the values obtained using the
    decreasing lexicographical rank by (`x`, `y`) and by (`y`, `x`).
weigher : callable, optional
    The weigher function. Must map nonnegative integers (zero
    representing the most important element) to a nonnegative weight.
    The default, None, provides hyperbolic weighing, that is,
    rank :math:`r` is mapped to weight :math:`1/(r+1)`.
additive : bool, optional
    If True, the weight of an exchange is computed by adding the
    weights of the ranks of the exchanged elements; otherwise, the weights
    are multiplied. The default is True.

Returns
-------
res: SignificanceResult
    An object containing attributes:

    statistic : float
       The weighted :math:`\tau` correlation index.
    pvalue : float
       Presently ``np.nan``, as the null distribution of the statistic is
       unknown (even in the additive hyperbolic case).

See Also
--------
kendalltau : Calculates Kendall's tau.
spearmanr : Calculates a Spearman rank-order correlation coefficient.
theilslopes : Computes the Theil-Sen estimator for a set of points (x, y).

Notes
-----
This function uses an :math:`O(n \log n)`, mergesort-based algorithm
[1]_ that is a weighted extension of Knight's algorithm for Kendall's
:math:`\tau` [2]_. It can compute Shieh's weighted :math:`\tau` [3]_
between rankings without ties (i.e., permutations) by setting
`additive` and `rank` to False, as the definition given in [1]_ is a
generalization of Shieh's.

NaNs are considered the smallest possible score.

.. versionadded:: 0.19.0

References
----------
.. [1] Sebastiano Vigna, "A weighted correlation index for rankings with
       ties", Proceedings of the 24th international conference on World
       Wide Web, pp. 1166-1176, ACM, 2015.
.. [2] W.R. Knight, "A Computer Method for Calculating Kendall's Tau with
       Ungrouped Data", Journal of the American Statistical Association,
       Vol. 61, No. 314, Part 1, pp. 436-439, 1966.
.. [3] Grace S. Shieh. "A weighted Kendall's tau statistic", Statistics &
       Probability Letters, Vol. 39, No. 1, pp. 17-24, 1998.

Examples
--------
>>> import numpy as np
>>> from scipy import stats
>>> x = [12, 2, 1, 12, 2]
>>> y = [1, 4, 7, 1, 0]
>>> res = stats.weightedtau(x, y)
>>> res.statistic
-0.56694968153682723
>>> res.pvalue
nan
>>> res = stats.weightedtau(x, y, additive=False)
>>> res.statistic
-0.62205716951801038

NaNs are considered the smallest possible score:

>>> x = [12, 2, 1, 12, 2]
>>> y = [1, 4, 7, 1, np.nan]
>>> res = stats.weightedtau(x, y)
>>> res.statistic
-0.56694968153682723

This is exactly Kendall's tau:

>>> x = [12, 2, 1, 12, 2]
>>> y = [1, 4, 7, 1, 0]
>>> res = stats.weightedtau(x, y, weigher=lambda x: 1)
>>> res.statistic
-0.47140452079103173

>>> x = [12, 2, 1, 12, 2]
>>> y = [1, 4, 7, 1, 0]
>>> stats.weightedtau(x, y, rank=None)
SignificanceResult(statistic=-0.4157652301037516, pvalue=nan)
>>> stats.weightedtau(y, x, rank=None)
SignificanceResult(statistic=-0.7181341329699028, pvalue=nan)
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l05952}{5952}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_abecb96afa85e562e14f2ff4f301a0ef9}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!zmap@{zmap}}
\index{zmap@{zmap}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{zmap()}{zmap()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_abecb96afa85e562e14f2ff4f301a0ef9} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+zmap (\begin{DoxyParamCaption}\item[{}]{scores}{, }\item[{}]{compare}{, }\item[{}]{axis}{ = {\ttfamily 0}, }\item[{}]{ddof}{ = {\ttfamily 0}, }\item[{}]{nan\+\_\+policy}{ = {\ttfamily \textquotesingle{}propagate\textquotesingle{}}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Calculate the relative z-scores.

Return an array of z-scores, i.e., scores that are standardized to
zero mean and unit variance, where mean and variance are calculated
from the comparison array.

Parameters
----------
scores : array_like
    The input for which z-scores are calculated.
compare : array_like
    The input from which the mean and standard deviation of the
    normalization are taken; assumed to have the same dimension as
    `scores`.
axis : int or None, optional
    Axis over which mean and variance of `compare` are calculated.
    Default is 0. If None, compute over the whole array `scores`.
ddof : int, optional
    Degrees of freedom correction in the calculation of the
    standard deviation. Default is 0.
nan_policy : {'propagate', 'raise', 'omit'}, optional
    Defines how to handle the occurrence of nans in `compare`.
    'propagate' returns nan, 'raise' raises an exception, 'omit'
    performs the calculations ignoring nan values. Default is
    'propagate'. Note that when the value is 'omit', nans in `scores`
    also propagate to the output, but they do not affect the z-scores
    computed for the non-nan values.

Returns
-------
zscore : array_like
    Z-scores, in the same shape as `scores`.

Notes
-----
This function preserves ndarray subclasses, and works also with
matrices and masked arrays (it uses `asanyarray` instead of
`asarray` for parameters).

Examples
--------
>>> from scipy.stats import zmap
>>> a = [0.5, 2.0, 2.5, 3]
>>> b = [0, 1, 2, 3, 4]
>>> zmap(a, b)
array([-1.06066017,  0.        ,  0.35355339,  0.70710678])
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l03070}{3070}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l03980}{\+\_\+first()}}, \mbox{\hyperlink{__stats__py_8py_source_l02847}{\+\_\+isconst()}}, \mbox{\hyperlink{__stats__py_8py_source_l02863}{\+\_\+quiet\+\_\+nanmean()}}, and \mbox{\hyperlink{__stats__py_8py_source_l02877}{\+\_\+quiet\+\_\+nanstd()}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l02891}{zscore()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a40557c1c2fd14506e84aa782a7382f68}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!zscore@{zscore}}
\index{zscore@{zscore}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{zscore()}{zscore()}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a40557c1c2fd14506e84aa782a7382f68} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+zscore (\begin{DoxyParamCaption}\item[{}]{a}{, }\item[{}]{axis}{ = {\ttfamily 0}, }\item[{}]{ddof}{ = {\ttfamily 0}, }\item[{}]{nan\+\_\+policy}{ = {\ttfamily \textquotesingle{}propagate\textquotesingle{}}}\end{DoxyParamCaption})}

\begin{DoxyVerb}Compute the z score.

Compute the z score of each value in the sample, relative to the
sample mean and standard deviation.

Parameters
----------
a : array_like
    An array like object containing the sample data.
axis : int or None, optional
    Axis along which to operate. Default is 0. If None, compute over
    the whole array `a`.
ddof : int, optional
    Degrees of freedom correction in the calculation of the
    standard deviation. Default is 0.
nan_policy : {'propagate', 'raise', 'omit'}, optional
    Defines how to handle when input contains nan. 'propagate' returns nan,
    'raise' throws an error, 'omit' performs the calculations ignoring nan
    values. Default is 'propagate'.  Note that when the value is 'omit',
    nans in the input also propagate to the output, but they do not affect
    the z-scores computed for the non-nan values.

Returns
-------
zscore : array_like
    The z-scores, standardized by mean and standard deviation of
    input array `a`.

See Also
--------
numpy.mean : Arithmetic average
numpy.std : Arithmetic standard deviation
scipy.stats.gzscore : Geometric standard score

Notes
-----
This function preserves ndarray subclasses, and works also with
matrices and masked arrays (it uses `asanyarray` instead of
`asarray` for parameters).

References
----------
.. [1] "Standard score", *Wikipedia*,
       https://en.wikipedia.org/wiki/Standard_score.
.. [2] Huck, S. W., Cross, T. L., Clark, S. B, "Overcoming misconceptions
       about Z-scores", Teaching Statistics, vol. 8, pp. 38-40, 1986

Examples
--------
>>> import numpy as np
>>> a = np.array([ 0.7972,  0.0767,  0.4383,  0.7866,  0.8091,
...                0.1954,  0.6307,  0.6599,  0.1065,  0.0508])
>>> from scipy import stats
>>> stats.zscore(a)
array([ 1.1273, -1.247 , -0.0552,  1.0923,  1.1664, -0.8559,  0.5786,
        0.6748, -1.1488, -1.3324])

Computing along a specified axis, using n-1 degrees of freedom
(``ddof=1``) to calculate the standard deviation:

>>> b = np.array([[ 0.3148,  0.0478,  0.6243,  0.4608],
...               [ 0.7149,  0.0775,  0.6072,  0.9656],
...               [ 0.6341,  0.1403,  0.9759,  0.4064],
...               [ 0.5918,  0.6948,  0.904 ,  0.3721],
...               [ 0.0921,  0.2481,  0.1188,  0.1366]])
>>> stats.zscore(b, axis=1, ddof=1)
array([[-0.19264823, -1.28415119,  1.07259584,  0.40420358],
       [ 0.33048416, -1.37380874,  0.04251374,  1.00081084],
       [ 0.26796377, -1.12598418,  1.23283094, -0.37481053],
       [-0.22095197,  0.24468594,  1.19042819, -1.21416216],
       [-0.82780366,  1.4457416 , -0.43867764, -0.1792603 ]])

An example with `nan_policy='omit'`:

>>> x = np.array([[25.11, 30.10, np.nan, 32.02, 43.15],
...               [14.95, 16.06, 121.25, 94.35, 29.81]])
>>> stats.zscore(x, axis=1, nan_policy='omit')
array([[-1.13490897, -0.37830299,         nan, -0.08718406,  1.60039602],
       [-0.91611681, -0.89090508,  1.4983032 ,  0.88731639, -0.5785977 ]])
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{__stats__py_8py_source_l02891}{2891}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



References \mbox{\hyperlink{__stats__py_8py_source_l03070}{zmap()}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l02976}{gzscore()}}.



\doxysubsection{Variable Documentation}
\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_ab762f81902847b88a6822c3eca3bf77c}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_\_all\_\_@{\_\_all\_\_}}
\index{\_\_all\_\_@{\_\_all\_\_}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_\_all\_\_}{\_\_all\_\_}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_ab762f81902847b88a6822c3eca3bf77c} 
list scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+\+\_\+all\+\_\+\+\_\+\hspace{0.3cm}{\ttfamily [private]}}

{\bfseries Initial value\+:}
\begin{DoxyCode}{0}
\DoxyCodeLine{00001\ =\ \ [\textcolor{stringliteral}{'find\_repeats'},\ \textcolor{stringliteral}{'gmean'},\ \textcolor{stringliteral}{'hmean'},\ \textcolor{stringliteral}{'pmean'},\ \textcolor{stringliteral}{'mode'},\ \textcolor{stringliteral}{'tmean'},\ \textcolor{stringliteral}{'tvar'},}
\DoxyCodeLine{00002\ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'tmin'},\ \textcolor{stringliteral}{'tmax'},\ \textcolor{stringliteral}{'tstd'},\ \textcolor{stringliteral}{'tsem'},\ \textcolor{stringliteral}{'moment'},}
\DoxyCodeLine{00003\ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'skew'},\ \textcolor{stringliteral}{'kurtosis'},\ \textcolor{stringliteral}{'describe'},\ \textcolor{stringliteral}{'skewtest'},\ \textcolor{stringliteral}{'kurtosistest'},}
\DoxyCodeLine{00004\ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'normaltest'},\ \textcolor{stringliteral}{'jarque\_bera'},}
\DoxyCodeLine{00005\ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'scoreatpercentile'},\ \textcolor{stringliteral}{'percentileofscore'},}
\DoxyCodeLine{00006\ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'cumfreq'},\ \textcolor{stringliteral}{'relfreq'},\ \textcolor{stringliteral}{'obrientransform'},}
\DoxyCodeLine{00007\ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'sem'},\ \textcolor{stringliteral}{'zmap'},\ \textcolor{stringliteral}{'zscore'},\ \textcolor{stringliteral}{'gzscore'},\ \textcolor{stringliteral}{'iqr'},\ \textcolor{stringliteral}{'gstd'},}
\DoxyCodeLine{00008\ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'median\_abs\_deviation'},}
\DoxyCodeLine{00009\ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'sigmaclip'},\ \textcolor{stringliteral}{'trimboth'},\ \textcolor{stringliteral}{'trim1'},\ \textcolor{stringliteral}{'trim\_mean'},}
\DoxyCodeLine{00010\ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'f\_oneway'},\ \textcolor{stringliteral}{'pearsonr'},\ \textcolor{stringliteral}{'fisher\_exact'},}
\DoxyCodeLine{00011\ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'spearmanr'},\ \textcolor{stringliteral}{'pointbiserialr'},}
\DoxyCodeLine{00012\ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'kendalltau'},\ \textcolor{stringliteral}{'weightedtau'},\ \textcolor{stringliteral}{'multiscale\_graphcorr'},}
\DoxyCodeLine{00013\ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'linregress'},\ \textcolor{stringliteral}{'siegelslopes'},\ \textcolor{stringliteral}{'theilslopes'},\ \textcolor{stringliteral}{'ttest\_1samp'},}
\DoxyCodeLine{00014\ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'ttest\_ind'},\ \textcolor{stringliteral}{'ttest\_ind\_from\_stats'},\ \textcolor{stringliteral}{'ttest\_rel'},}
\DoxyCodeLine{00015\ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'kstest'},\ \textcolor{stringliteral}{'ks\_1samp'},\ \textcolor{stringliteral}{'ks\_2samp'},}
\DoxyCodeLine{00016\ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'chisquare'},\ \textcolor{stringliteral}{'power\_divergence'},}
\DoxyCodeLine{00017\ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'tiecorrect'},\ \textcolor{stringliteral}{'ranksums'},\ \textcolor{stringliteral}{'kruskal'},\ \textcolor{stringliteral}{'friedmanchisquare'},}
\DoxyCodeLine{00018\ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'rankdata'},}
\DoxyCodeLine{00019\ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'combine\_pvalues'},\ \textcolor{stringliteral}{'wasserstein\_distance'},\ \textcolor{stringliteral}{'energy\_distance'},}
\DoxyCodeLine{00020\ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'brunnermunzel'},\ \textcolor{stringliteral}{'alexandergovern'},}
\DoxyCodeLine{00021\ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'expectile'},\ ]}

\end{DoxyCode}


Definition at line \mbox{\hyperlink{__stats__py_8py_source_l00067}{67}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a8a1eac21b5e82f9848b2523b6b90995e}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_power\_div\_lambda\_names@{\_power\_div\_lambda\_names}}
\index{\_power\_div\_lambda\_names@{\_power\_div\_lambda\_names}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_power\_div\_lambda\_names}{\_power\_div\_lambda\_names}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a8a1eac21b5e82f9848b2523b6b90995e} 
dict scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+power\+\_\+div\+\_\+lambda\+\_\+names\hspace{0.3cm}{\ttfamily [protected]}}

{\bfseries Initial value\+:}
\begin{DoxyCode}{0}
\DoxyCodeLine{00001\ =\ \ \{}
\DoxyCodeLine{00002\ \ \ \ \ \textcolor{stringliteral}{"{}pearson"{}}:\ 1,}
\DoxyCodeLine{00003\ \ \ \ \ \textcolor{stringliteral}{"{}log-\/likelihood"{}}:\ 0,}
\DoxyCodeLine{00004\ \ \ \ \ \textcolor{stringliteral}{"{}freeman-\/tukey"{}}:\ -\/0.5,}
\DoxyCodeLine{00005\ \ \ \ \ \textcolor{stringliteral}{"{}mod-\/log-\/likelihood"{}}:\ -\/1,}
\DoxyCodeLine{00006\ \ \ \ \ \textcolor{stringliteral}{"{}neyman"{}}:\ -\/2,}
\DoxyCodeLine{00007\ \ \ \ \ \textcolor{stringliteral}{"{}cressie-\/read"{}}:\ 2/3,}
\DoxyCodeLine{00008\ \}}

\end{DoxyCode}


Definition at line \mbox{\hyperlink{__stats__py_8py_source_l07816}{7816}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a1d8ab9655add9064cf78ddbfc338cbb2}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!\_scale\_conversions@{\_scale\_conversions}}
\index{\_scale\_conversions@{\_scale\_conversions}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{\_scale\_conversions}{\_scale\_conversions}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a1d8ab9655add9064cf78ddbfc338cbb2} 
dict scipy.\+stats.\+\_\+stats\+\_\+py.\+\_\+scale\+\_\+conversions\hspace{0.3cm}{\ttfamily [protected]}}

{\bfseries Initial value\+:}
\begin{DoxyCode}{0}
\DoxyCodeLine{00001\ =\ \ \{\textcolor{stringliteral}{'raw'}:\ 1.0,}
\DoxyCodeLine{00002\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'normal'}:\ special.erfinv(0.5)\ *\ 2.0\ *\ math.sqrt(2.0)\}}

\end{DoxyCode}


Definition at line \mbox{\hyperlink{__stats__py_8py_source_l03298}{3298}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a084547535521fc8a4e2ca13da7ccf236}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!BrunnerMunzelResult@{BrunnerMunzelResult}}
\index{BrunnerMunzelResult@{BrunnerMunzelResult}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{BrunnerMunzelResult}{BrunnerMunzelResult}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a084547535521fc8a4e2ca13da7ccf236} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+Brunner\+Munzel\+Result}

{\bfseries Initial value\+:}
\begin{DoxyCode}{0}
\DoxyCodeLine{00001\ =\ \ \mbox{\hyperlink{classnamedtuple}{namedtuple}}(\textcolor{stringliteral}{'BrunnerMunzelResult'},}
\DoxyCodeLine{00002\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ (\textcolor{stringliteral}{'statistic'},\ \textcolor{stringliteral}{'pvalue'}))}

\end{DoxyCode}


Definition at line \mbox{\hyperlink{__stats__py_8py_source_l09432}{9432}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_ae4b1ccd5e913ab32dee4d40b27b90992}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!ConfidenceInterval@{ConfidenceInterval}}
\index{ConfidenceInterval@{ConfidenceInterval}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{ConfidenceInterval}{ConfidenceInterval}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_ae4b1ccd5e913ab32dee4d40b27b90992} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+Confidence\+Interval = \mbox{\hyperlink{classnamedtuple}{namedtuple}}(\textquotesingle{}Confidence\+Interval\textquotesingle{}, \mbox{[}\textquotesingle{}low\textquotesingle{}, \textquotesingle{}high\textquotesingle{}\mbox{]})}



Definition at line \mbox{\hyperlink{__stats__py_8py_source_l04442}{4442}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a1a75604ce103a4f3a8661da14b41ba21}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!CumfreqResult@{CumfreqResult}}
\index{CumfreqResult@{CumfreqResult}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{CumfreqResult}{CumfreqResult}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a1a75604ce103a4f3a8661da14b41ba21} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+Cumfreq\+Result}

{\bfseries Initial value\+:}
\begin{DoxyCode}{0}
\DoxyCodeLine{00001\ =\ \ \mbox{\hyperlink{classnamedtuple}{namedtuple}}(\textcolor{stringliteral}{'CumfreqResult'},}
\DoxyCodeLine{00002\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ (\textcolor{stringliteral}{'cumcount'},\ \textcolor{stringliteral}{'lowerlimit'},\ \textcolor{stringliteral}{'binsize'},}
\DoxyCodeLine{00003\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'extrapoints'}))}

\end{DoxyCode}


Definition at line \mbox{\hyperlink{__stats__py_8py_source_l02537}{2537}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_ad17cbb60a06b3b961b8d3dec3f75f571}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!DescribeResult@{DescribeResult}}
\index{DescribeResult@{DescribeResult}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{DescribeResult}{DescribeResult}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_ad17cbb60a06b3b961b8d3dec3f75f571} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+Describe\+Result}

{\bfseries Initial value\+:}
\begin{DoxyCode}{0}
\DoxyCodeLine{00001\ =\ \ \mbox{\hyperlink{classnamedtuple}{namedtuple}}(\textcolor{stringliteral}{'DescribeResult'},}
\DoxyCodeLine{00002\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ (\textcolor{stringliteral}{'nobs'},\ \textcolor{stringliteral}{'minmax'},\ \textcolor{stringliteral}{'mean'},\ \textcolor{stringliteral}{'variance'},\ \textcolor{stringliteral}{'skewness'},}
\DoxyCodeLine{00003\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'kurtosis'}))}

\end{DoxyCode}


Definition at line \mbox{\hyperlink{__stats__py_8py_source_l01321}{1321}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a55dd91ad97a133a4f1c4014a850145ea}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!F\_onewayResult@{F\_onewayResult}}
\index{F\_onewayResult@{F\_onewayResult}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{F\_onewayResult}{F\_onewayResult}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a55dd91ad97a133a4f1c4014a850145ea} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+F\+\_\+oneway\+Result = \mbox{\hyperlink{classnamedtuple}{namedtuple}}(\textquotesingle{}F\+\_\+oneway\+Result\textquotesingle{}, (\textquotesingle{}statistic\textquotesingle{}, \textquotesingle{}pvalue\textquotesingle{}))}



Definition at line \mbox{\hyperlink{__stats__py_8py_source_l03960}{3960}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_acf31b33400d148f6ef0e0f3b66f5d2be}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!FriedmanchisquareResult@{FriedmanchisquareResult}}
\index{FriedmanchisquareResult@{FriedmanchisquareResult}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{FriedmanchisquareResult}{FriedmanchisquareResult}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_acf31b33400d148f6ef0e0f3b66f5d2be} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+Friedmanchisquare\+Result}

{\bfseries Initial value\+:}
\begin{DoxyCode}{0}
\DoxyCodeLine{00001\ =\ \ \mbox{\hyperlink{classnamedtuple}{namedtuple}}(\textcolor{stringliteral}{'FriedmanchisquareResult'},}
\DoxyCodeLine{00002\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ (\textcolor{stringliteral}{'statistic'},\ \textcolor{stringliteral}{'pvalue'}))}

\end{DoxyCode}


Definition at line \mbox{\hyperlink{__stats__py_8py_source_l09334}{9334}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a159d3500af93794f6cb4a9f959924bb5}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!HistogramResult@{HistogramResult}}
\index{HistogramResult@{HistogramResult}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{HistogramResult}{HistogramResult}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a159d3500af93794f6cb4a9f959924bb5} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+Histogram\+Result}

{\bfseries Initial value\+:}
\begin{DoxyCode}{0}
\DoxyCodeLine{00001\ =\ \ \mbox{\hyperlink{classnamedtuple}{namedtuple}}(\textcolor{stringliteral}{'HistogramResult'},}
\DoxyCodeLine{00002\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ (\textcolor{stringliteral}{'count'},\ \textcolor{stringliteral}{'lowerlimit'},\ \textcolor{stringliteral}{'binsize'},\ \textcolor{stringliteral}{'extrapoints'}))}

\end{DoxyCode}


Definition at line \mbox{\hyperlink{__stats__py_8py_source_l02455}{2455}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_ac9cba6ca5b344ceda11181bcd5f6adea}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!KruskalResult@{KruskalResult}}
\index{KruskalResult@{KruskalResult}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{KruskalResult}{KruskalResult}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_ac9cba6ca5b344ceda11181bcd5f6adea} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+Kruskal\+Result = \mbox{\hyperlink{classnamedtuple}{namedtuple}}(\textquotesingle{}Kruskal\+Result\textquotesingle{}, (\textquotesingle{}statistic\textquotesingle{}, \textquotesingle{}pvalue\textquotesingle{}))}



Definition at line \mbox{\hyperlink{__stats__py_8py_source_l09211}{9211}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a4c3e4d7762108939709c279fff05dcb4}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!Ks\_2sampResult@{Ks\_2sampResult}}
\index{Ks\_2sampResult@{Ks\_2sampResult}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{Ks\_2sampResult}{Ks\_2sampResult}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a4c3e4d7762108939709c279fff05dcb4} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+Ks\+\_\+2samp\+Result = Kstest\+Result}



Definition at line \mbox{\hyperlink{__stats__py_8py_source_l08481}{8481}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_af6c2563e8e2c6e7d3e19fa32131a6796}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!KstestResult@{KstestResult}}
\index{KstestResult@{KstestResult}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{KstestResult}{KstestResult}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_af6c2563e8e2c6e7d3e19fa32131a6796} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+Kstest\+Result}

{\bfseries Initial value\+:}
\begin{DoxyCode}{0}
\DoxyCodeLine{00001\ =\ \ \_make\_tuple\_bunch(\textcolor{stringliteral}{'KstestResult'},\ [\textcolor{stringliteral}{'statistic'},\ \textcolor{stringliteral}{'pvalue'}],}
\DoxyCodeLine{00002\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ [\textcolor{stringliteral}{'statistic\_location'},\ \textcolor{stringliteral}{'statistic\_sign'}])}

\end{DoxyCode}


Definition at line \mbox{\hyperlink{__stats__py_8py_source_l08258}{8258}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a4db485cb945393c294a5defd286e49bb}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!KurtosistestResult@{KurtosistestResult}}
\index{KurtosistestResult@{KurtosistestResult}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{KurtosistestResult}{KurtosistestResult}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a4db485cb945393c294a5defd286e49bb} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+Kurtosistest\+Result = \mbox{\hyperlink{classnamedtuple}{namedtuple}}(\textquotesingle{}Kurtosistest\+Result\textquotesingle{}, (\textquotesingle{}statistic\textquotesingle{}, \textquotesingle{}pvalue\textquotesingle{}))}



Definition at line \mbox{\hyperlink{__stats__py_8py_source_l01624}{1624}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a79b67a68f52e541cf43ed09574e13108}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!MGCResult@{MGCResult}}
\index{MGCResult@{MGCResult}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{MGCResult}{MGCResult}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a79b67a68f52e541cf43ed09574e13108} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+MGCResult}

{\bfseries Initial value\+:}
\begin{DoxyCode}{0}
\DoxyCodeLine{00001\ =\ \ \_make\_tuple\_bunch(\textcolor{stringliteral}{'MGCResult'},}
\DoxyCodeLine{00002\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ [\textcolor{stringliteral}{'statistic'},\ \textcolor{stringliteral}{'pvalue'},\ \textcolor{stringliteral}{'mgc\_dict'}],\ [])}

\end{DoxyCode}


Definition at line \mbox{\hyperlink{__stats__py_8py_source_l06220}{6220}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a8276a1473f84d29df1cfb05a798c6c86}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!ModeResult@{ModeResult}}
\index{ModeResult@{ModeResult}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{ModeResult}{ModeResult}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a8276a1473f84d29df1cfb05a798c6c86} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+Mode\+Result = \mbox{\hyperlink{classnamedtuple}{namedtuple}}(\textquotesingle{}Mode\+Result\textquotesingle{}, (\textquotesingle{}\mbox{\hyperlink{namespacescipy_1_1stats_1_1__stats__py_a8c1c3534a15de7f93b96291982b74ac2}{mode}}\textquotesingle{}, \textquotesingle{}count\textquotesingle{}))}



Definition at line \mbox{\hyperlink{__stats__py_8py_source_l00425}{425}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a7ecccad49a9bf1d4db7baeba60ed6c54}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!NormaltestResult@{NormaltestResult}}
\index{NormaltestResult@{NormaltestResult}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{NormaltestResult}{NormaltestResult}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a7ecccad49a9bf1d4db7baeba60ed6c54} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+Normaltest\+Result = \mbox{\hyperlink{classnamedtuple}{namedtuple}}(\textquotesingle{}Normaltest\+Result\textquotesingle{}, (\textquotesingle{}statistic\textquotesingle{}, \textquotesingle{}pvalue\textquotesingle{}))}



Definition at line \mbox{\hyperlink{__stats__py_8py_source_l01833}{1833}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_ad063c82a38b3f71d713793c4565ca702}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!PearsonRResultBase@{PearsonRResultBase}}
\index{PearsonRResultBase@{PearsonRResultBase}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{PearsonRResultBase}{PearsonRResultBase}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_ad063c82a38b3f71d713793c4565ca702} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+Pearson\+RResult\+Base}

{\bfseries Initial value\+:}
\begin{DoxyCode}{0}
\DoxyCodeLine{00001\ =\ \ \_make\_tuple\_bunch(\textcolor{stringliteral}{'PearsonRResultBase'},}
\DoxyCodeLine{00002\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ [\textcolor{stringliteral}{'statistic'},\ \textcolor{stringliteral}{'pvalue'}],\ [])}

\end{DoxyCode}


Definition at line \mbox{\hyperlink{__stats__py_8py_source_l04444}{4444}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a0b74f644917a36fcd23aa95f48810815}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!Power\_divergenceResult@{Power\_divergenceResult}}
\index{Power\_divergenceResult@{Power\_divergenceResult}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{Power\_divergenceResult}{Power\_divergenceResult}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a0b74f644917a36fcd23aa95f48810815} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+Power\+\_\+divergence\+Result}

{\bfseries Initial value\+:}
\begin{DoxyCode}{0}
\DoxyCodeLine{00001\ =\ \ \mbox{\hyperlink{classnamedtuple}{namedtuple}}(\textcolor{stringliteral}{'Power\_divergenceResult'},}
\DoxyCodeLine{00002\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ (\textcolor{stringliteral}{'statistic'},\ \textcolor{stringliteral}{'pvalue'}))}

\end{DoxyCode}


Definition at line \mbox{\hyperlink{__stats__py_8py_source_l07853}{7853}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a13929c3e11b3aea010483ae368fdeeea}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!RanksumsResult@{RanksumsResult}}
\index{RanksumsResult@{RanksumsResult}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{RanksumsResult}{RanksumsResult}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a13929c3e11b3aea010483ae368fdeeea} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+Ranksums\+Result = \mbox{\hyperlink{classnamedtuple}{namedtuple}}(\textquotesingle{}Ranksums\+Result\textquotesingle{}, (\textquotesingle{}statistic\textquotesingle{}, \textquotesingle{}pvalue\textquotesingle{}))}



Definition at line \mbox{\hyperlink{__stats__py_8py_source_l09129}{9129}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a0739c0c0192b367b27d6d1fc6b076469}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!RelfreqResult@{RelfreqResult}}
\index{RelfreqResult@{RelfreqResult}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{RelfreqResult}{RelfreqResult}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a0739c0c0192b367b27d6d1fc6b076469} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+Relfreq\+Result}

{\bfseries Initial value\+:}
\begin{DoxyCode}{0}
\DoxyCodeLine{00001\ =\ \ \mbox{\hyperlink{classnamedtuple}{namedtuple}}(\textcolor{stringliteral}{'RelfreqResult'},}
\DoxyCodeLine{00002\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ (\textcolor{stringliteral}{'frequency'},\ \textcolor{stringliteral}{'lowerlimit'},\ \textcolor{stringliteral}{'binsize'},}
\DoxyCodeLine{00003\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'extrapoints'}))}

\end{DoxyCode}


Definition at line \mbox{\hyperlink{__stats__py_8py_source_l02619}{2619}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a5abc71d3f4e3d43499eea3985b7d2bd9}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!RepeatedResults@{RepeatedResults}}
\index{RepeatedResults@{RepeatedResults}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{RepeatedResults}{RepeatedResults}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a5abc71d3f4e3d43499eea3985b7d2bd9} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+Repeated\+Results = \mbox{\hyperlink{classnamedtuple}{namedtuple}}(\textquotesingle{}Repeated\+Results\textquotesingle{}, (\textquotesingle{}values\textquotesingle{}, \textquotesingle{}counts\textquotesingle{}))}



SUPPORT FUNCTIONS \#. 



Definition at line \mbox{\hyperlink{__stats__py_8py_source_l10030}{10030}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l10033}{find\+\_\+repeats()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_ade3bb71bac397106f36f0ebe2f76ff8e}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!SigmaclipResult@{SigmaclipResult}}
\index{SigmaclipResult@{SigmaclipResult}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{SigmaclipResult}{SigmaclipResult}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_ade3bb71bac397106f36f0ebe2f76ff8e} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+Sigmaclip\+Result = \mbox{\hyperlink{classnamedtuple}{namedtuple}}(\textquotesingle{}Sigmaclip\+Result\textquotesingle{}, (\textquotesingle{}clipped\textquotesingle{}, \textquotesingle{}lower\textquotesingle{}, \textquotesingle{}upper\textquotesingle{}))}



TRIMMING FUNCTIONS \#. 



Definition at line \mbox{\hyperlink{__stats__py_8py_source_l03645}{3645}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.



Referenced by \mbox{\hyperlink{__stats__py_8py_source_l03648}{sigmaclip()}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a99e444e3ed4cb89551065e638d8e25ba}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!SignificanceResult@{SignificanceResult}}
\index{SignificanceResult@{SignificanceResult}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{SignificanceResult}{SignificanceResult}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a99e444e3ed4cb89551065e638d8e25ba} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+Significance\+Result}

{\bfseries Initial value\+:}
\begin{DoxyCode}{0}
\DoxyCodeLine{00001\ =\ \ \_make\_tuple\_bunch(\textcolor{stringliteral}{'SignificanceResult'},}
\DoxyCodeLine{00002\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ [\textcolor{stringliteral}{'statistic'},\ \textcolor{stringliteral}{'pvalue'}],\ [])}

\end{DoxyCode}


Definition at line \mbox{\hyperlink{__stats__py_8py_source_l00122}{122}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_a517833de500c51cf8468fe2ccee2b20d}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!SkewtestResult@{SkewtestResult}}
\index{SkewtestResult@{SkewtestResult}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{SkewtestResult}{SkewtestResult}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_a517833de500c51cf8468fe2ccee2b20d} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+Skewtest\+Result = \mbox{\hyperlink{classnamedtuple}{namedtuple}}(\textquotesingle{}Skewtest\+Result\textquotesingle{}, (\textquotesingle{}statistic\textquotesingle{}, \textquotesingle{}pvalue\textquotesingle{}))}



Definition at line \mbox{\hyperlink{__stats__py_8py_source_l01433}{1433}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_ab6b754f522c03e7fccd72c806de40407}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!Ttest\_indResult@{Ttest\_indResult}}
\index{Ttest\_indResult@{Ttest\_indResult}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{Ttest\_indResult}{Ttest\_indResult}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_ab6b754f522c03e7fccd72c806de40407} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+Ttest\+\_\+ind\+Result = \mbox{\hyperlink{classnamedtuple}{namedtuple}}(\textquotesingle{}Ttest\+\_\+ind\+Result\textquotesingle{}, (\textquotesingle{}statistic\textquotesingle{}, \textquotesingle{}pvalue\textquotesingle{}))}



Definition at line \mbox{\hyperlink{__stats__py_8py_source_l07035}{7035}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

\Hypertarget{namespacescipy_1_1stats_1_1__stats__py_acf46192a3335b4aa0f1a5fdc412ecc64}\index{scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}!TtestResultBase@{TtestResultBase}}
\index{TtestResultBase@{TtestResultBase}!scipy.stats.\_stats\_py@{scipy.stats.\_stats\_py}}
\doxysubsubsection{\texorpdfstring{TtestResultBase}{TtestResultBase}}
{\footnotesize\ttfamily \label{namespacescipy_1_1stats_1_1__stats__py_acf46192a3335b4aa0f1a5fdc412ecc64} 
scipy.\+stats.\+\_\+stats\+\_\+py.\+Ttest\+Result\+Base}

{\bfseries Initial value\+:}
\begin{DoxyCode}{0}
\DoxyCodeLine{00001\ =\ \ \_make\_tuple\_bunch(\textcolor{stringliteral}{'TtestResultBase'},}
\DoxyCodeLine{00002\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ [\textcolor{stringliteral}{'statistic'},\ \textcolor{stringliteral}{'pvalue'}],\ [\textcolor{stringliteral}{'df'}])}

\end{DoxyCode}


INFERENTIAL STATISTICS \#. 



Definition at line \mbox{\hyperlink{__stats__py_8py_source_l06685}{6685}} of file \mbox{\hyperlink{__stats__py_8py_source}{\+\_\+stats\+\_\+py.\+py}}.

